// rdf-namespaces/dist/arg.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * opposition
 * 
 * 
 *
 * http://www.w3.org/ns/pim/arg#opposition
 */
export const opposition: Property = 'http://www.w3.org/ns/pim/arg#opposition';

/**
 * requires
 * 
 * 
 *
 * http://www.w3.org/ns/pim/arg#requires
 */
export const requires: Property = 'http://www.w3.org/ns/pim/arg#requires';

/**
 * support
 * 
 * 
 *
 * http://www.w3.org/ns/pim/arg#support
 */
export const support: Property = 'http://www.w3.org/ns/pim/arg#support';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/pim/arg#Conjunction
 */
export const Conjunction: Class = 'http://www.w3.org/ns/pim/arg#Conjunction';

/**
 * position
 * 
 * 
 *
 * http://www.w3.org/ns/pim/arg#Position
 */
export const Position: Class = 'http://www.w3.org/ns/pim/arg#Position';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/pim/arg#Resolution
 */
export const Resolution: Class = 'http://www.w3.org/ns/pim/arg#Resolution';
