// rdf-namespaces/dist/cal-13a8c1b1.js
/**
 * VALARM
 *
 * Provide a grouping of component properties that define an alarm.
 *
 * http://www.w3.org/2002/12/cal/ical#Valarm
 */
var Valarm = 'http://www.w3.org/2002/12/cal/ical#Valarm';
/**
 *
 *
 *
 *
 * http://www.w3.org/2002/12/cal/ical#Value_DATE
 */
var Value_DATE = 'http://www.w3.org/2002/12/cal/ical#Value_DATE';
/**
 *
 *
 *
 *
 * http://www.w3.org/2002/12/cal/ical#Value_DURATION
 */
var Value_DURATION = 'http://www.w3.org/2002/12/cal/ical#Value_DURATION';
/**
 *
 *
 *
 *
 * http://www.w3.org/2002/12/cal/ical#Value_PERIOD
 */
var Value_PERIOD = 'http://www.w3.org/2002/12/cal/ical#Value_PERIOD';
/**
 *
 *
 *
 *
 * http://www.w3.org/2002/12/cal/ical#Value_RECUR
 */
var Value_RECUR = 'http://www.w3.org/2002/12/cal/ical#Value_RECUR';
/**
 * VEVENT
 *
 * Provide a grouping of component properties that describe an event.
 *
 * http://www.w3.org/2002/12/cal/ical#Vevent
 */
var Vevent = 'http://www.w3.org/2002/12/cal/ical#Vevent';
/**
 * VFREEBUSY
 *
 * Provide a grouping of component properties that describe either a request for free/busy time, describe a response to a request for free/busy time or describe a published set of busy time.
 *
 * http://www.w3.org/2002/12/cal/ical#Vfreebusy
 */
var Vfreebusy = 'http://www.w3.org/2002/12/cal/ical#Vfreebusy';
/**
 * VJOURNAL
 *
 * Provide a grouping of component properties that describe a journal entry.
 *
 * http://www.w3.org/2002/12/cal/ical#Vjournal
 */
var Vjournal = 'http://www.w3.org/2002/12/cal/ical#Vjournal';
/**
 * VTIMEZONE
 *
 * Provide a grouping of component properties that defines a time zone.
 *
 * http://www.w3.org/2002/12/cal/ical#Vtimezone
 */
var Vtimezone = 'http://www.w3.org/2002/12/cal/ical#Vtimezone';
/**
 * VTODO
 *
 * Provide a grouping of calendar properties that describe a to-do.
 *
 * http://www.w3.org/2002/12/cal/ical#Vtodo
 */
var Vtodo = 'http://www.w3.org/2002/12/cal/ical#Vtodo';
/**
 * ATTACH
 *
 *
 * 	    default value type: URI
 *
 * http://www.w3.org/2002/12/cal/ical#attach
 */
var attach = 'http://www.w3.org/2002/12/cal/ical#attach';
/**
 * ATTENDEE
 *
 *
 * 	    value type: CAL-ADDRESS
 *
 * http://www.w3.org/2002/12/cal/ical#attendee
 */
var attendee = 'http://www.w3.org/2002/12/cal/ical#attendee';
/**
 * COMPLETED
 *
 *
 * 	    value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#completed
 */
var completed = 'http://www.w3.org/2002/12/cal/ical#completed';
/**
 * CREATED
 *
 *
 * 	    value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#created
 */
var created = 'http://www.w3.org/2002/12/cal/ical#created';
/**
 * DTEND
 *
 *
 * 	    default value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#dtend
 */
var dtend = 'http://www.w3.org/2002/12/cal/ical#dtend';
/**
 * DTSTAMP
 *
 *
 * 	    value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#dtstamp
 */
var dtstamp = 'http://www.w3.org/2002/12/cal/ical#dtstamp';
/**
 * DTSTART
 *
 *
 * 	    default value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#dtstart
 */
var dtstart = 'http://www.w3.org/2002/12/cal/ical#dtstart';
/**
 * DUE
 *
 *
 * 	    default value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#due
 */
var due = 'http://www.w3.org/2002/12/cal/ical#due';
/**
 * DURATION
 *
 *
 * 	    value type: DURATION
 *
 * http://www.w3.org/2002/12/cal/ical#duration
 */
var duration = 'http://www.w3.org/2002/12/cal/ical#duration';
/**
 * EXDATE
 *
 *
 * 	    default value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#exdate
 */
var exdate = 'http://www.w3.org/2002/12/cal/ical#exdate';
/**
 * EXRULE
 *
 *
 * 	    value type: RECUR
 *
 * http://www.w3.org/2002/12/cal/ical#exrule
 */
var exrule = 'http://www.w3.org/2002/12/cal/ical#exrule';
/**
 * FREEBUSY
 *
 *
 * 	    value type: PERIOD
 *
 * http://www.w3.org/2002/12/cal/ical#freebusy
 */
var freebusy = 'http://www.w3.org/2002/12/cal/ical#freebusy';
/**
 * GEO
 *
 * This property specifies information related to the global position for the activity specified by a calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#geo
 */
var geo = 'http://www.w3.org/2002/12/cal/ical#geo';
/**
 * LAST-MODIFIED
 *
 *
 * 	    value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#lastModified
 */
var lastModified = 'http://www.w3.org/2002/12/cal/ical#lastModified';
/**
 * ORGANIZER
 *
 *
 * 	    value type: CAL-ADDRESS
 *
 * http://www.w3.org/2002/12/cal/ical#organizer
 */
var organizer = 'http://www.w3.org/2002/12/cal/ical#organizer';
/**
 * RDATE
 *
 *
 * 	    default value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#rdate
 */
var rdate = 'http://www.w3.org/2002/12/cal/ical#rdate';
/**
 * RECURRENCE-ID
 *
 *
 * 	    default value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#recurrenceId
 */
var recurrenceId = 'http://www.w3.org/2002/12/cal/ical#recurrenceId';
/**
 * RRULE
 *
 *
 * 	    value type: RECUR
 *
 * http://www.w3.org/2002/12/cal/ical#rrule
 */
var rrule = 'http://www.w3.org/2002/12/cal/ical#rrule';
/**
 * TRIGGER
 *
 *
 * 	    default value type: DURATION
 *
 * http://www.w3.org/2002/12/cal/ical#trigger
 */
var trigger = 'http://www.w3.org/2002/12/cal/ical#trigger';
/**
 * TZURL
 *
 *
 * 	    value type: URI
 *
 * http://www.w3.org/2002/12/cal/ical#tzurl
 */
var tzurl = 'http://www.w3.org/2002/12/cal/ical#tzurl';
/**
 * URL
 *
 *
 * 	    value type: URI
 *
 * http://www.w3.org/2002/12/cal/ical#url
 */
var url = 'http://www.w3.org/2002/12/cal/ical#url';
/**
 * ACTION
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#action
 */
var action = 'http://www.w3.org/2002/12/cal/ical#action';
/**
 * CALSCALE
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#calscale
 */
var calscale = 'http://www.w3.org/2002/12/cal/ical#calscale';
/**
 * CATEGORIES
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#categories
 */
var categories = 'http://www.w3.org/2002/12/cal/ical#categories';
/**
 * CLASS
 *
 *
 * 	    value type: TEXT
 * Note that `class` is a reserved Javascript keyword, and is therefore suffixed by `__workaround`.
 * For a list of reserved keywords, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords.
 *
 * http://www.w3.org/2002/12/cal/ical#class
 */
var class__workaround = 'http://www.w3.org/2002/12/cal/ical#class';
/**
 * COMMENT
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#comment
 */
var comment = 'http://www.w3.org/2002/12/cal/ical#comment';
/**
 * CONTACT
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#contact
 */
var contact = 'http://www.w3.org/2002/12/cal/ical#contact';
/**
 * DESCRIPTION
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#description
 */
var description = 'http://www.w3.org/2002/12/cal/ical#description';
/**
 * LOCATION
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#location
 */
var location = 'http://www.w3.org/2002/12/cal/ical#location';
/**
 * METHOD
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#method
 */
var method = 'http://www.w3.org/2002/12/cal/ical#method';
/**
 * PERCENT-COMPLETE
 *
 *
 * 	    value type: INTEGER
 *
 * http://www.w3.org/2002/12/cal/ical#percentComplete
 */
var percentComplete = 'http://www.w3.org/2002/12/cal/ical#percentComplete';
/**
 * PRIORITY
 *
 *
 * 	    value type: INTEGER
 *
 * http://www.w3.org/2002/12/cal/ical#priority
 */
var priority = 'http://www.w3.org/2002/12/cal/ical#priority';
/**
 * PRODID
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#prodid
 */
var prodid = 'http://www.w3.org/2002/12/cal/ical#prodid';
/**
 * RELATED-TO
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#relatedTo
 */
var relatedTo = 'http://www.w3.org/2002/12/cal/ical#relatedTo';
/**
 * REPEAT
 *
 *
 * 	    value type: INTEGER
 *
 * http://www.w3.org/2002/12/cal/ical#repeat
 */
var repeat = 'http://www.w3.org/2002/12/cal/ical#repeat';
/**
 * REQUEST-STATUS
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#requestStatus
 */
var requestStatus = 'http://www.w3.org/2002/12/cal/ical#requestStatus';
/**
 * RESOURCES
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#resources
 */
var resources = 'http://www.w3.org/2002/12/cal/ical#resources';
/**
 * SEQUENCE
 *
 *
 * 	    value type: integer
 *
 * http://www.w3.org/2002/12/cal/ical#sequence
 */
var sequence = 'http://www.w3.org/2002/12/cal/ical#sequence';
/**
 * STATUS
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#status
 */
var status = 'http://www.w3.org/2002/12/cal/ical#status';
/**
 * SUMMARY
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#summary
 */
var summary = 'http://www.w3.org/2002/12/cal/ical#summary';
/**
 * TRANSP
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#transp
 */
var transp = 'http://www.w3.org/2002/12/cal/ical#transp';
/**
 * TZID
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#tzid
 */
var tzid = 'http://www.w3.org/2002/12/cal/ical#tzid';
/**
 * TZNAME
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#tzname
 */
var tzname = 'http://www.w3.org/2002/12/cal/ical#tzname';
/**
 * TZOFFSETFROM
 *
 *
 * 	    value type: UTC-OFFSET
 *
 * http://www.w3.org/2002/12/cal/ical#tzoffsetfrom
 */
var tzoffsetfrom = 'http://www.w3.org/2002/12/cal/ical#tzoffsetfrom';
/**
 * TZOFFSETTO
 *
 *
 * 	    value type: UTC-OFFSET
 *
 * http://www.w3.org/2002/12/cal/ical#tzoffsetto
 */
var tzoffsetto = 'http://www.w3.org/2002/12/cal/ical#tzoffsetto';
/**
 * UID
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#uid
 */
var uid = 'http://www.w3.org/2002/12/cal/ical#uid';
/**
 * VERSION
 *
 *
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#version
 */
var version = 'http://www.w3.org/2002/12/cal/ical#version';

var calImport = /*#__PURE__*/Object.freeze({
	Valarm: Valarm,
	Value_DATE: Value_DATE,
	Value_DURATION: Value_DURATION,
	Value_PERIOD: Value_PERIOD,
	Value_RECUR: Value_RECUR,
	Vevent: Vevent,
	Vfreebusy: Vfreebusy,
	Vjournal: Vjournal,
	Vtimezone: Vtimezone,
	Vtodo: Vtodo,
	attach: attach,
	attendee: attendee,
	completed: completed,
	created: created,
	dtend: dtend,
	dtstamp: dtstamp,
	dtstart: dtstart,
	due: due,
	duration: duration,
	exdate: exdate,
	exrule: exrule,
	freebusy: freebusy,
	geo: geo,
	lastModified: lastModified,
	organizer: organizer,
	rdate: rdate,
	recurrenceId: recurrenceId,
	rrule: rrule,
	trigger: trigger,
	tzurl: tzurl,
	url: url,
	action: action,
	calscale: calscale,
	categories: categories,
	class__workaround: class__workaround,
	comment: comment,
	contact: contact,
	description: description,
	location: location,
	method: method,
	percentComplete: percentComplete,
	priority: priority,
	prodid: prodid,
	relatedTo: relatedTo,
	repeat: repeat,
	requestStatus: requestStatus,
	resources: resources,
	sequence: sequence,
	status: status,
	summary: summary,
	transp: transp,
	tzid: tzid,
	tzname: tzname,
	tzoffsetfrom: tzoffsetfrom,
	tzoffsetto: tzoffsetto,
	uid: uid,
	version: version
});

export { tzoffsetfrom as $, rrule as A, trigger as B, tzurl as C, url as D, action as E, calscale as F, categories as G, class__workaround as H, comment as I, contact as J, description as K, location as L, method as M, percentComplete as N, priority as O, prodid as P, relatedTo as Q, repeat as R, requestStatus as S, resources as T, sequence as U, Valarm as V, status as W, summary as X, transp as Y, tzid as Z, tzname as _, Value_DATE as a, tzoffsetto as a0, uid as a1, version as a2, calImport as a3, Value_DURATION as b, Value_PERIOD as c, Value_RECUR as d, Vevent as e, Vfreebusy as f, Vjournal as g, Vtimezone as h, Vtodo as i, attach as j, attendee as k, completed as l, created as m, dtend as n, dtstamp as o, dtstart as p, due as q, duration as r, exdate as s, exrule as t, freebusy as u, geo as v, lastModified as w, organizer as x, rdate as y, recurrenceId as z };
