// rdf-namespaces/dist/cal.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * VALARM
 * 
 * Provide a grouping of component properties that define an alarm.
 *
 * http://www.w3.org/2002/12/cal/ical#Valarm
 */
export const Valarm: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Valarm';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2002/12/cal/ical#Value_DATE
 */
export const Value_DATE: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Value_DATE';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2002/12/cal/ical#Value_DURATION
 */
export const Value_DURATION: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Value_DURATION';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2002/12/cal/ical#Value_PERIOD
 */
export const Value_PERIOD: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Value_PERIOD';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2002/12/cal/ical#Value_RECUR
 */
export const Value_RECUR: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Value_RECUR';

/**
 * VEVENT
 * 
 * Provide a grouping of component properties that describe an event.
 *
 * http://www.w3.org/2002/12/cal/ical#Vevent
 */
export const Vevent: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Vevent';

/**
 * VFREEBUSY
 * 
 * Provide a grouping of component properties that describe either a request for free/busy time, describe a response to a request for free/busy time or describe a published set of busy time.
 *
 * http://www.w3.org/2002/12/cal/ical#Vfreebusy
 */
export const Vfreebusy: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Vfreebusy';

/**
 * VJOURNAL
 * 
 * Provide a grouping of component properties that describe a journal entry.
 *
 * http://www.w3.org/2002/12/cal/ical#Vjournal
 */
export const Vjournal: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Vjournal';

/**
 * VTIMEZONE
 * 
 * Provide a grouping of component properties that defines a time zone.
 *
 * http://www.w3.org/2002/12/cal/ical#Vtimezone
 */
export const Vtimezone: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Vtimezone';

/**
 * VTODO
 * 
 * Provide a grouping of calendar properties that describe a to-do.
 *
 * http://www.w3.org/2002/12/cal/ical#Vtodo
 */
export const Vtodo: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Vtodo';

/**
 * ATTACH
 * 
 * 
 * 	    default value type: URI
 *
 * http://www.w3.org/2002/12/cal/ical#attach
 */
export const attach: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#attach';

/**
 * ATTENDEE
 * 
 * 
 * 	    value type: CAL-ADDRESS
 *
 * http://www.w3.org/2002/12/cal/ical#attendee
 */
export const attendee: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#attendee';

/**
 * COMPLETED
 * 
 * 
 * 	    value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#completed
 */
export const completed: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#completed';

/**
 * CREATED
 * 
 * 
 * 	    value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#created
 */
export const created: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#created';

/**
 * DTEND
 * 
 * 
 * 	    default value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#dtend
 */
export const dtend: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#dtend';

/**
 * DTSTAMP
 * 
 * 
 * 	    value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#dtstamp
 */
export const dtstamp: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#dtstamp';

/**
 * DTSTART
 * 
 * 
 * 	    default value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#dtstart
 */
export const dtstart: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#dtstart';

/**
 * DUE
 * 
 * 
 * 	    default value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#due
 */
export const due: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#due';

/**
 * DURATION
 * 
 * 
 * 	    value type: DURATION
 *
 * http://www.w3.org/2002/12/cal/ical#duration
 */
export const duration: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#duration';

/**
 * EXDATE
 * 
 * 
 * 	    default value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#exdate
 */
export const exdate: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#exdate';

/**
 * EXRULE
 * 
 * 
 * 	    value type: RECUR
 *
 * http://www.w3.org/2002/12/cal/ical#exrule
 */
export const exrule: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#exrule';

/**
 * FREEBUSY
 * 
 * 
 * 	    value type: PERIOD
 *
 * http://www.w3.org/2002/12/cal/ical#freebusy
 */
export const freebusy: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#freebusy';

/**
 * GEO
 * 
 * This property specifies information related to the global position for the activity specified by a calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#geo
 */
export const geo: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#geo';

/**
 * LAST-MODIFIED
 * 
 * 
 * 	    value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#lastModified
 */
export const lastModified: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#lastModified';

/**
 * ORGANIZER
 * 
 * 
 * 	    value type: CAL-ADDRESS
 *
 * http://www.w3.org/2002/12/cal/ical#organizer
 */
export const organizer: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#organizer';

/**
 * RDATE
 * 
 * 
 * 	    default value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#rdate
 */
export const rdate: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#rdate';

/**
 * RECURRENCE-ID
 * 
 * 
 * 	    default value type: DATE-TIME
 *
 * http://www.w3.org/2002/12/cal/ical#recurrenceId
 */
export const recurrenceId: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#recurrenceId';

/**
 * RRULE
 * 
 * 
 * 	    value type: RECUR
 *
 * http://www.w3.org/2002/12/cal/ical#rrule
 */
export const rrule: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#rrule';

/**
 * TRIGGER
 * 
 * 
 * 	    default value type: DURATION
 *
 * http://www.w3.org/2002/12/cal/ical#trigger
 */
export const trigger: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#trigger';

/**
 * TZURL
 * 
 * 
 * 	    value type: URI
 *
 * http://www.w3.org/2002/12/cal/ical#tzurl
 */
export const tzurl: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#tzurl';

/**
 * URL
 * 
 * 
 * 	    value type: URI
 *
 * http://www.w3.org/2002/12/cal/ical#url
 */
export const url: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#url';

/**
 * ACTION
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#action
 */
export const action: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#action';

/**
 * CALSCALE
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#calscale
 */
export const calscale: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#calscale';

/**
 * CATEGORIES
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#categories
 */
export const categories: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#categories';

/**
 * CLASS
 * 
 * 
 * 	    value type: TEXT
 * Note that `class` is a reserved Javascript keyword, and is therefore suffixed by `__workaround`.
 * For a list of reserved keywords, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords.
 *
 * http://www.w3.org/2002/12/cal/ical#class
 */
export const class__workaround: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#class';

/**
 * COMMENT
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#comment
 */
export const comment: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#comment';

/**
 * CONTACT
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#contact
 */
export const contact: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#contact';

/**
 * DESCRIPTION
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#description
 */
export const description: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#description';

/**
 * LOCATION
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#location
 */
export const location: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#location';

/**
 * METHOD
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#method
 */
export const method: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#method';

/**
 * PERCENT-COMPLETE
 * 
 * 
 * 	    value type: INTEGER
 *
 * http://www.w3.org/2002/12/cal/ical#percentComplete
 */
export const percentComplete: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#percentComplete';

/**
 * PRIORITY
 * 
 * 
 * 	    value type: INTEGER
 *
 * http://www.w3.org/2002/12/cal/ical#priority
 */
export const priority: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#priority';

/**
 * PRODID
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#prodid
 */
export const prodid: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#prodid';

/**
 * RELATED-TO
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#relatedTo
 */
export const relatedTo: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#relatedTo';

/**
 * REPEAT
 * 
 * 
 * 	    value type: INTEGER
 *
 * http://www.w3.org/2002/12/cal/ical#repeat
 */
export const repeat: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#repeat';

/**
 * REQUEST-STATUS
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#requestStatus
 */
export const requestStatus: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#requestStatus';

/**
 * RESOURCES
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#resources
 */
export const resources: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#resources';

/**
 * SEQUENCE
 * 
 * 
 * 	    value type: integer
 *
 * http://www.w3.org/2002/12/cal/ical#sequence
 */
export const sequence: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#sequence';

/**
 * STATUS
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#status
 */
export const status: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#status';

/**
 * SUMMARY
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#summary
 */
export const summary: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#summary';

/**
 * TRANSP
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#transp
 */
export const transp: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#transp';

/**
 * TZID
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#tzid
 */
export const tzid: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#tzid';

/**
 * TZNAME
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#tzname
 */
export const tzname: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#tzname';

/**
 * TZOFFSETFROM
 * 
 * 
 * 	    value type: UTC-OFFSET
 *
 * http://www.w3.org/2002/12/cal/ical#tzoffsetfrom
 */
export const tzoffsetfrom: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#tzoffsetfrom';

/**
 * TZOFFSETTO
 * 
 * 
 * 	    value type: UTC-OFFSET
 *
 * http://www.w3.org/2002/12/cal/ical#tzoffsetto
 */
export const tzoffsetto: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#tzoffsetto';

/**
 * UID
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#uid
 */
export const uid: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#uid';

/**
 * VERSION
 * 
 * 
 * 	    value type: TEXT
 *
 * http://www.w3.org/2002/12/cal/ical#version
 */
export const version: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#version';
