// rdf-namespaces/dist/cal.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var cal$1 = require('./cal-4a4b4cc5.js');



exports.Valarm = cal$1.Valarm;
exports.Value_DATE = cal$1.Value_DATE;
exports.Value_DURATION = cal$1.Value_DURATION;
exports.Value_PERIOD = cal$1.Value_PERIOD;
exports.Value_RECUR = cal$1.Value_RECUR;
exports.Vevent = cal$1.Vevent;
exports.Vfreebusy = cal$1.Vfreebusy;
exports.Vjournal = cal$1.Vjournal;
exports.Vtimezone = cal$1.Vtimezone;
exports.Vtodo = cal$1.Vtodo;
exports.action = cal$1.action;
exports.attach = cal$1.attach;
exports.attendee = cal$1.attendee;
exports.calscale = cal$1.calscale;
exports.categories = cal$1.categories;
exports.class__workaround = cal$1.class__workaround;
exports.comment = cal$1.comment;
exports.completed = cal$1.completed;
exports.contact = cal$1.contact;
exports.created = cal$1.created;
exports.description = cal$1.description;
exports.dtend = cal$1.dtend;
exports.dtstamp = cal$1.dtstamp;
exports.dtstart = cal$1.dtstart;
exports.due = cal$1.due;
exports.duration = cal$1.duration;
exports.exdate = cal$1.exdate;
exports.exrule = cal$1.exrule;
exports.freebusy = cal$1.freebusy;
exports.geo = cal$1.geo;
exports.lastModified = cal$1.lastModified;
exports.location = cal$1.location;
exports.method = cal$1.method;
exports.organizer = cal$1.organizer;
exports.percentComplete = cal$1.percentComplete;
exports.priority = cal$1.priority;
exports.prodid = cal$1.prodid;
exports.rdate = cal$1.rdate;
exports.recurrenceId = cal$1.recurrenceId;
exports.relatedTo = cal$1.relatedTo;
exports.repeat = cal$1.repeat;
exports.requestStatus = cal$1.requestStatus;
exports.resources = cal$1.resources;
exports.rrule = cal$1.rrule;
exports.sequence = cal$1.sequence;
exports.status = cal$1.status;
exports.summary = cal$1.summary;
exports.transp = cal$1.transp;
exports.trigger = cal$1.trigger;
exports.tzid = cal$1.tzid;
exports.tzname = cal$1.tzname;
exports.tzoffsetfrom = cal$1.tzoffsetfrom;
exports.tzoffsetto = cal$1.tzoffsetto;
exports.tzurl = cal$1.tzurl;
exports.uid = cal$1.uid;
exports.url = cal$1.url;
exports.version = cal$1.version;
