// rdf-namespaces/dist/contact.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * full name
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#fullName
 */
export const fullName: Property = 'http://www.w3.org/2000/10/swap/pim/contact#fullName';

/**
 * sort name
 * 
 * re-arranged for lexicographic ordering; ala Doe, John
 *
 * http://www.w3.org/2000/10/swap/pim/contact#sortName
 */
export const sortName: Property = 'http://www.w3.org/2000/10/swap/pim/contact#sortName';

/**
 * nearest airport
 * 
 * ?X nearestAirport ?Y locates ?X in an international
 *    context; for example, for the purpose of organizing a face-to-face
 *    meeting of a W3C working group. This property is intended to mitigate
 *    privacy risks of giving out detailed contact info.
 *
 * http://www.w3.org/2000/10/swap/pim/contact#nearestAirport
 */
export const nearestAirport: Property = 'http://www.w3.org/2000/10/swap/pim/contact#nearestAirport';

/**
 * name
 * 
 * A person may be known as various strings.
 * 	For example, an email friendly name string.
 * 	If you have an email from someone using a string as the human-readable phrase,
 * 	then it is reasonable to assume that there are :knownAs that.
 *
 * http://www.w3.org/2000/10/swap/pim/contact#knownAs
 */
export const knownAs: Property = 'http://www.w3.org/2000/10/swap/pim/contact#knownAs';

/**
 * assistant
 * 
 * A person (or other agent) who is an assistant to the subject.
 *
 * http://www.w3.org/2000/10/swap/pim/contact#assistant
 */
export const assistant: Property = 'http://www.w3.org/2000/10/swap/pim/contact#assistant';

/**
 * participant
 * 
 * A person (or other agent) who  particpates in an event, meeting, etc.
 *
 * http://www.w3.org/2000/10/swap/pim/contact#participant
 */
export const participant: Property = 'http://www.w3.org/2000/10/swap/pim/contact#participant';

/**
 * preferred
 * 
 * A string which is the URI a person, organization, etc, prefers that people use for them.
 *
 * http://www.w3.org/2000/10/swap/pim/contact#preferredURI
 */
export const preferredURI: Property = 'http://www.w3.org/2000/10/swap/pim/contact#preferredURI';

/**
 * work
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#office
 */
export const office: Property = 'http://www.w3.org/2000/10/swap/pim/contact#office';

/**
 * home
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#home
 */
export const home: Property = 'http://www.w3.org/2000/10/swap/pim/contact#home';

/**
 * vacation
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#vacationHome
 */
export const vacationHome: Property = 'http://www.w3.org/2000/10/swap/pim/contact#vacationHome';

/**
 * mobile
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#mobile
 */
export const mobile: Property = 'http://www.w3.org/2000/10/swap/pim/contact#mobile';

/**
 * emergency only
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#emergency
 */
export const emergency: Property = 'http://www.w3.org/2000/10/swap/pim/contact#emergency';

/**
 * address
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#address
 */
export const address: Property = 'http://www.w3.org/2000/10/swap/pim/contact#address';

/**
 * region
 * 
 * The sub-national unit such as state (us),
 * province (ca), county (uk), département (fr), and so on
 *
 * http://www.w3.org/2000/10/swap/pim/contact#region
 */
export const region: Property = 'http://www.w3.org/2000/10/swap/pim/contact#region';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#phone
 */
export const phone: Property = 'http://www.w3.org/2000/10/swap/pim/contact#phone';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#fax
 */
export const fax: Property = 'http://www.w3.org/2000/10/swap/pim/contact#fax';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#birthday
 */
export const birthday: Property = 'http://www.w3.org/2000/10/swap/pim/contact#birthday';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#child
 */
export const child: Property = 'http://www.w3.org/2000/10/swap/pim/contact#child';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#_addressProperty
 */
export const _addressProperty: Property = 'http://www.w3.org/2000/10/swap/pim/contact#_addressProperty';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#motherTongue
 */
export const motherTongue: Property = 'http://www.w3.org/2000/10/swap/pim/contact#motherTongue';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#description
 */
export const description: Property = 'http://www.w3.org/2000/10/swap/pim/contact#description';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#partner
 */
export const partner: Property = 'http://www.w3.org/2000/10/swap/pim/contact#partner';

/**
 * web page
 * 
 * A related web page
 *
 * http://www.w3.org/2000/10/swap/pim/contact#webPage
 */
export const webPage: Property = 'http://www.w3.org/2000/10/swap/pim/contact#webPage';

/**
 * 
 * 
 * The sort of thing which can have a phone number.
 *     Typically a person or an incorporated company, or unincorporated group.
 *
 * http://www.w3.org/2000/10/swap/pim/contact#SocialEntity
 */
export const SocialEntity: Class = 'http://www.w3.org/2000/10/swap/pim/contact#SocialEntity';

/**
 * contact point
 * 
 * A place, or mobile situation, with address, phone number, fax, etc. 
 *     Related to a person by home, office, etc.
 *     Note one person's workplace may be another person's home.
 *     A person may have more than one home and more than one workplace.
 *     (In practice it sometimes maybe useful with restriucted datasets to
 *     assume that this is not the case, when extracting data from other ontologies
 *     with no concept of ContactLocation).  Strongly related to a person: 
 *     in some ways a role that a person can be in.
 *
 * http://www.w3.org/2000/10/swap/pim/contact#ContactLocation
 */
export const ContactLocation: Class = 'http://www.w3.org/2000/10/swap/pim/contact#ContactLocation';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#Address
 */
export const Address: Class = 'http://www.w3.org/2000/10/swap/pim/contact#Address';

/**
 * phone
 * 
 * 
 *     An end-point in the public swiitched telephone system.
 *     Anything identified by a URI with tel: scheme is in this class.
 *     
 *
 * http://www.w3.org/2000/10/swap/pim/contact#Phone
 */
export const Phone: Class = 'http://www.w3.org/2000/10/swap/pim/contact#Phone';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#Male
 */
export const Male: Class = 'http://www.w3.org/2000/10/swap/pim/contact#Male';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#Female
 */
export const Female: Class = 'http://www.w3.org/2000/10/swap/pim/contact#Female';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/contact#LanguageCode
 */
export const LanguageCode: Class = 'http://www.w3.org/2000/10/swap/pim/contact#LanguageCode';
