// rdf-namespaces/dist/contact-c3e04d20.js
'use strict';

/**
 * full name
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#fullName
 */
var fullName = 'http://www.w3.org/2000/10/swap/pim/contact#fullName';
/**
 * sort name
 *
 * re-arranged for lexicographic ordering; ala Doe, John
 *
 * http://www.w3.org/2000/10/swap/pim/contact#sortName
 */
var sortName = 'http://www.w3.org/2000/10/swap/pim/contact#sortName';
/**
 * nearest airport
 *
 * ?X nearestAirport ?Y locates ?X in an international
 *    context; for example, for the purpose of organizing a face-to-face
 *    meeting of a W3C working group. This property is intended to mitigate
 *    privacy risks of giving out detailed contact info.
 *
 * http://www.w3.org/2000/10/swap/pim/contact#nearestAirport
 */
var nearestAirport = 'http://www.w3.org/2000/10/swap/pim/contact#nearestAirport';
/**
 * name
 *
 * A person may be known as various strings.
 * 	For example, an email friendly name string.
 * 	If you have an email from someone using a string as the human-readable phrase,
 * 	then it is reasonable to assume that there are :knownAs that.
 *
 * http://www.w3.org/2000/10/swap/pim/contact#knownAs
 */
var knownAs = 'http://www.w3.org/2000/10/swap/pim/contact#knownAs';
/**
 * assistant
 *
 * A person (or other agent) who is an assistant to the subject.
 *
 * http://www.w3.org/2000/10/swap/pim/contact#assistant
 */
var assistant = 'http://www.w3.org/2000/10/swap/pim/contact#assistant';
/**
 * participant
 *
 * A person (or other agent) who  particpates in an event, meeting, etc.
 *
 * http://www.w3.org/2000/10/swap/pim/contact#participant
 */
var participant = 'http://www.w3.org/2000/10/swap/pim/contact#participant';
/**
 * preferred
 *
 * A string which is the URI a person, organization, etc, prefers that people use for them.
 *
 * http://www.w3.org/2000/10/swap/pim/contact#preferredURI
 */
var preferredURI = 'http://www.w3.org/2000/10/swap/pim/contact#preferredURI';
/**
 * work
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#office
 */
var office = 'http://www.w3.org/2000/10/swap/pim/contact#office';
/**
 * home
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#home
 */
var home = 'http://www.w3.org/2000/10/swap/pim/contact#home';
/**
 * vacation
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#vacationHome
 */
var vacationHome = 'http://www.w3.org/2000/10/swap/pim/contact#vacationHome';
/**
 * mobile
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#mobile
 */
var mobile = 'http://www.w3.org/2000/10/swap/pim/contact#mobile';
/**
 * emergency only
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#emergency
 */
var emergency = 'http://www.w3.org/2000/10/swap/pim/contact#emergency';
/**
 * address
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#address
 */
var address = 'http://www.w3.org/2000/10/swap/pim/contact#address';
/**
 * region
 *
 * The sub-national unit such as state (us),
 * province (ca), county (uk), département (fr), and so on
 *
 * http://www.w3.org/2000/10/swap/pim/contact#region
 */
var region = 'http://www.w3.org/2000/10/swap/pim/contact#region';
/**
 *
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#phone
 */
var phone = 'http://www.w3.org/2000/10/swap/pim/contact#phone';
/**
 *
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#fax
 */
var fax = 'http://www.w3.org/2000/10/swap/pim/contact#fax';
/**
 *
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#birthday
 */
var birthday = 'http://www.w3.org/2000/10/swap/pim/contact#birthday';
/**
 *
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#child
 */
var child = 'http://www.w3.org/2000/10/swap/pim/contact#child';
/**
 *
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#_addressProperty
 */
var _addressProperty = 'http://www.w3.org/2000/10/swap/pim/contact#_addressProperty';
/**
 *
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#motherTongue
 */
var motherTongue = 'http://www.w3.org/2000/10/swap/pim/contact#motherTongue';
/**
 *
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#description
 */
var description = 'http://www.w3.org/2000/10/swap/pim/contact#description';
/**
 *
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#partner
 */
var partner = 'http://www.w3.org/2000/10/swap/pim/contact#partner';
/**
 * web page
 *
 * A related web page
 *
 * http://www.w3.org/2000/10/swap/pim/contact#webPage
 */
var webPage = 'http://www.w3.org/2000/10/swap/pim/contact#webPage';
/**
 *
 *
 * The sort of thing which can have a phone number.
 *     Typically a person or an incorporated company, or unincorporated group.
 *
 * http://www.w3.org/2000/10/swap/pim/contact#SocialEntity
 */
var SocialEntity = 'http://www.w3.org/2000/10/swap/pim/contact#SocialEntity';
/**
 * contact point
 *
 * A place, or mobile situation, with address, phone number, fax, etc.
 *     Related to a person by home, office, etc.
 *     Note one person's workplace may be another person's home.
 *     A person may have more than one home and more than one workplace.
 *     (In practice it sometimes maybe useful with restriucted datasets to
 *     assume that this is not the case, when extracting data from other ontologies
 *     with no concept of ContactLocation).  Strongly related to a person:
 *     in some ways a role that a person can be in.
 *
 * http://www.w3.org/2000/10/swap/pim/contact#ContactLocation
 */
var ContactLocation = 'http://www.w3.org/2000/10/swap/pim/contact#ContactLocation';
/**
 *
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#Address
 */
var Address = 'http://www.w3.org/2000/10/swap/pim/contact#Address';
/**
 * phone
 *
 *
 *     An end-point in the public swiitched telephone system.
 *     Anything identified by a URI with tel: scheme is in this class.
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#Phone
 */
var Phone = 'http://www.w3.org/2000/10/swap/pim/contact#Phone';
/**
 *
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#Male
 */
var Male = 'http://www.w3.org/2000/10/swap/pim/contact#Male';
/**
 *
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#Female
 */
var Female = 'http://www.w3.org/2000/10/swap/pim/contact#Female';
/**
 *
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/contact#LanguageCode
 */
var LanguageCode = 'http://www.w3.org/2000/10/swap/pim/contact#LanguageCode';

var contactImport = /*#__PURE__*/Object.freeze({
	fullName: fullName,
	sortName: sortName,
	nearestAirport: nearestAirport,
	knownAs: knownAs,
	assistant: assistant,
	participant: participant,
	preferredURI: preferredURI,
	office: office,
	home: home,
	vacationHome: vacationHome,
	mobile: mobile,
	emergency: emergency,
	address: address,
	region: region,
	phone: phone,
	fax: fax,
	birthday: birthday,
	child: child,
	_addressProperty: _addressProperty,
	motherTongue: motherTongue,
	description: description,
	partner: partner,
	webPage: webPage,
	SocialEntity: SocialEntity,
	ContactLocation: ContactLocation,
	Address: Address,
	Phone: Phone,
	Male: Male,
	Female: Female,
	LanguageCode: LanguageCode
});

exports.Address = Address;
exports.ContactLocation = ContactLocation;
exports.Female = Female;
exports.LanguageCode = LanguageCode;
exports.Male = Male;
exports.Phone = Phone;
exports.SocialEntity = SocialEntity;
exports._addressProperty = _addressProperty;
exports.address = address;
exports.assistant = assistant;
exports.birthday = birthday;
exports.child = child;
exports.contactImport = contactImport;
exports.description = description;
exports.emergency = emergency;
exports.fax = fax;
exports.fullName = fullName;
exports.home = home;
exports.knownAs = knownAs;
exports.mobile = mobile;
exports.motherTongue = motherTongue;
exports.nearestAirport = nearestAirport;
exports.office = office;
exports.participant = participant;
exports.partner = partner;
exports.phone = phone;
exports.preferredURI = preferredURI;
exports.region = region;
exports.sortName = sortName;
exports.vacationHome = vacationHome;
exports.webPage = webPage;
