// rdf-namespaces/dist/contact.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var contact$1 = require('./contact-c3e04d20.js');



exports.Address = contact$1.Address;
exports.ContactLocation = contact$1.ContactLocation;
exports.Female = contact$1.Female;
exports.LanguageCode = contact$1.LanguageCode;
exports.Male = contact$1.Male;
exports.Phone = contact$1.Phone;
exports.SocialEntity = contact$1.SocialEntity;
exports._addressProperty = contact$1._addressProperty;
exports.address = contact$1.address;
exports.assistant = contact$1.assistant;
exports.birthday = contact$1.birthday;
exports.child = contact$1.child;
exports.description = contact$1.description;
exports.emergency = contact$1.emergency;
exports.fax = contact$1.fax;
exports.fullName = contact$1.fullName;
exports.home = contact$1.home;
exports.knownAs = contact$1.knownAs;
exports.mobile = contact$1.mobile;
exports.motherTongue = contact$1.motherTongue;
exports.nearestAirport = contact$1.nearestAirport;
exports.office = contact$1.office;
exports.participant = contact$1.participant;
exports.partner = contact$1.partner;
exports.phone = contact$1.phone;
exports.preferredURI = contact$1.preferredURI;
exports.region = contact$1.region;
exports.sortName = contact$1.sortName;
exports.vacationHome = contact$1.vacationHome;
exports.webPage = contact$1.webPage;
