// rdf-namespaces/dist/dc-e8eba9f8.js
'use strict';

/**
 * Contributor
 *
 * An entity responsible for making contributions to the resource.
 *
 * http://purl.org/dc/elements/1.1/contributor
 */
var contributor = 'http://purl.org/dc/elements/1.1/contributor';
/**
 * Coverage
 *
 * The spatial or temporal topic of the resource, the spatial applicability of the resource, or the jurisdiction under which the resource is relevant.
 *
 * http://purl.org/dc/elements/1.1/coverage
 */
var coverage = 'http://purl.org/dc/elements/1.1/coverage';
/**
 * Creator
 *
 * An entity primarily responsible for making the resource.
 *
 * http://purl.org/dc/elements/1.1/creator
 */
var creator = 'http://purl.org/dc/elements/1.1/creator';
/**
 * Date
 *
 * A point or period of time associated with an event in the lifecycle of the resource.
 *
 * http://purl.org/dc/elements/1.1/date
 */
var date = 'http://purl.org/dc/elements/1.1/date';
/**
 * Description
 *
 * An account of the resource.
 *
 * http://purl.org/dc/elements/1.1/description
 */
var description = 'http://purl.org/dc/elements/1.1/description';
/**
 * Format
 *
 * The file format, physical medium, or dimensions of the resource.
 *
 * http://purl.org/dc/elements/1.1/format
 */
var format = 'http://purl.org/dc/elements/1.1/format';
/**
 * Identifier
 *
 * An unambiguous reference to the resource within a given context.
 *
 * http://purl.org/dc/elements/1.1/identifier
 */
var identifier = 'http://purl.org/dc/elements/1.1/identifier';
/**
 * Language
 *
 * A language of the resource.
 *
 * http://purl.org/dc/elements/1.1/language
 */
var language = 'http://purl.org/dc/elements/1.1/language';
/**
 * Publisher
 *
 * An entity responsible for making the resource available.
 *
 * http://purl.org/dc/elements/1.1/publisher
 */
var publisher = 'http://purl.org/dc/elements/1.1/publisher';
/**
 * Relation
 *
 * A related resource.
 *
 * http://purl.org/dc/elements/1.1/relation
 */
var relation = 'http://purl.org/dc/elements/1.1/relation';
/**
 * Rights
 *
 * Information about rights held in and over the resource.
 *
 * http://purl.org/dc/elements/1.1/rights
 */
var rights = 'http://purl.org/dc/elements/1.1/rights';
/**
 * Source
 *
 * A related resource from which the described resource is derived.
 *
 * http://purl.org/dc/elements/1.1/source
 */
var source = 'http://purl.org/dc/elements/1.1/source';
/**
 * Subject
 *
 * The topic of the resource.
 *
 * http://purl.org/dc/elements/1.1/subject
 */
var subject = 'http://purl.org/dc/elements/1.1/subject';
/**
 * Title
 *
 * A name given to the resource.
 *
 * http://purl.org/dc/elements/1.1/title
 */
var title = 'http://purl.org/dc/elements/1.1/title';
/**
 * Type
 *
 * The nature or genre of the resource.
 *
 * http://purl.org/dc/elements/1.1/type
 */
var type = 'http://purl.org/dc/elements/1.1/type';

var dcImport = /*#__PURE__*/Object.freeze({
	contributor: contributor,
	coverage: coverage,
	creator: creator,
	date: date,
	description: description,
	format: format,
	identifier: identifier,
	language: language,
	publisher: publisher,
	relation: relation,
	rights: rights,
	source: source,
	subject: subject,
	title: title,
	type: type
});

exports.contributor = contributor;
exports.coverage = coverage;
exports.creator = creator;
exports.date = date;
exports.dcImport = dcImport;
exports.description = description;
exports.format = format;
exports.identifier = identifier;
exports.language = language;
exports.publisher = publisher;
exports.relation = relation;
exports.rights = rights;
exports.source = source;
exports.subject = subject;
exports.title = title;
exports.type = type;
