// rdf-namespaces/dist/dc.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * Contributor
 * 
 * An entity responsible for making contributions to the resource.
 *
 * http://purl.org/dc/elements/1.1/contributor
 */
export const contributor: Property = 'http://purl.org/dc/elements/1.1/contributor';

/**
 * Coverage
 * 
 * The spatial or temporal topic of the resource, the spatial applicability of the resource, or the jurisdiction under which the resource is relevant.
 *
 * http://purl.org/dc/elements/1.1/coverage
 */
export const coverage: Property = 'http://purl.org/dc/elements/1.1/coverage';

/**
 * Creator
 * 
 * An entity primarily responsible for making the resource.
 *
 * http://purl.org/dc/elements/1.1/creator
 */
export const creator: Property = 'http://purl.org/dc/elements/1.1/creator';

/**
 * Date
 * 
 * A point or period of time associated with an event in the lifecycle of the resource.
 *
 * http://purl.org/dc/elements/1.1/date
 */
export const date: Property = 'http://purl.org/dc/elements/1.1/date';

/**
 * Description
 * 
 * An account of the resource.
 *
 * http://purl.org/dc/elements/1.1/description
 */
export const description: Property = 'http://purl.org/dc/elements/1.1/description';

/**
 * Format
 * 
 * The file format, physical medium, or dimensions of the resource.
 *
 * http://purl.org/dc/elements/1.1/format
 */
export const format: Property = 'http://purl.org/dc/elements/1.1/format';

/**
 * Identifier
 * 
 * An unambiguous reference to the resource within a given context.
 *
 * http://purl.org/dc/elements/1.1/identifier
 */
export const identifier: Property = 'http://purl.org/dc/elements/1.1/identifier';

/**
 * Language
 * 
 * A language of the resource.
 *
 * http://purl.org/dc/elements/1.1/language
 */
export const language: Property = 'http://purl.org/dc/elements/1.1/language';

/**
 * Publisher
 * 
 * An entity responsible for making the resource available.
 *
 * http://purl.org/dc/elements/1.1/publisher
 */
export const publisher: Property = 'http://purl.org/dc/elements/1.1/publisher';

/**
 * Relation
 * 
 * A related resource.
 *
 * http://purl.org/dc/elements/1.1/relation
 */
export const relation: Property = 'http://purl.org/dc/elements/1.1/relation';

/**
 * Rights
 * 
 * Information about rights held in and over the resource.
 *
 * http://purl.org/dc/elements/1.1/rights
 */
export const rights: Property = 'http://purl.org/dc/elements/1.1/rights';

/**
 * Source
 * 
 * A related resource from which the described resource is derived.
 *
 * http://purl.org/dc/elements/1.1/source
 */
export const source: Property = 'http://purl.org/dc/elements/1.1/source';

/**
 * Subject
 * 
 * The topic of the resource.
 *
 * http://purl.org/dc/elements/1.1/subject
 */
export const subject: Property = 'http://purl.org/dc/elements/1.1/subject';

/**
 * Title
 * 
 * A name given to the resource.
 *
 * http://purl.org/dc/elements/1.1/title
 */
export const title: Property = 'http://purl.org/dc/elements/1.1/title';

/**
 * Type
 * 
 * The nature or genre of the resource.
 *
 * http://purl.org/dc/elements/1.1/type
 */
export const type: Property = 'http://purl.org/dc/elements/1.1/type';
