// rdf-namespaces/dist/dct.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var dct$1 = require('./dct-f8aebeba.js');



exports.Agent = dct$1.Agent;
exports.AgentClass = dct$1.AgentClass;
exports.BibliographicResource = dct$1.BibliographicResource;
exports.Box = dct$1.Box;
exports.FileFormat = dct$1.FileFormat;
exports.Frequency = dct$1.Frequency;
exports.ISO3166 = dct$1.ISO3166;
exports.Jurisdiction = dct$1.Jurisdiction;
exports.LicenseDocument = dct$1.LicenseDocument;
exports.LinguisticSystem = dct$1.LinguisticSystem;
exports.Location = dct$1.Location;
exports.LocationPeriodOrJurisdiction = dct$1.LocationPeriodOrJurisdiction;
exports.MediaType = dct$1.MediaType;
exports.MediaTypeOrExtent = dct$1.MediaTypeOrExtent;
exports.MethodOfAccrual = dct$1.MethodOfAccrual;
exports.MethodOfInstruction = dct$1.MethodOfInstruction;
exports.Period = dct$1.Period;
exports.PeriodOfTime = dct$1.PeriodOfTime;
exports.PhysicalMedium = dct$1.PhysicalMedium;
exports.PhysicalResource = dct$1.PhysicalResource;
exports.Point = dct$1.Point;
exports.Policy = dct$1.Policy;
exports.ProvenanceStatement = dct$1.ProvenanceStatement;
exports.RFC1766 = dct$1.RFC1766;
exports.RFC3066 = dct$1.RFC3066;
exports.RFC4646 = dct$1.RFC4646;
exports.RFC5646 = dct$1.RFC5646;
exports.RightsStatement = dct$1.RightsStatement;
exports.SizeOrDuration = dct$1.SizeOrDuration;
exports.Standard = dct$1.Standard;
exports.URI = dct$1.URI;
exports.W3CDTF = dct$1.W3CDTF;
exports.abstract = dct$1.abstract;
exports.accessRights = dct$1.accessRights;
exports.accrualMethod = dct$1.accrualMethod;
exports.accrualPeriodicity = dct$1.accrualPeriodicity;
exports.accrualPolicy = dct$1.accrualPolicy;
exports.alternative = dct$1.alternative;
exports.audience = dct$1.audience;
exports.available = dct$1.available;
exports.bibliographicCitation = dct$1.bibliographicCitation;
exports.conformsTo = dct$1.conformsTo;
exports.contributor = dct$1.contributor;
exports.coverage = dct$1.coverage;
exports.created = dct$1.created;
exports.creator = dct$1.creator;
exports.date = dct$1.date;
exports.dateAccepted = dct$1.dateAccepted;
exports.dateCopyrighted = dct$1.dateCopyrighted;
exports.dateSubmitted = dct$1.dateSubmitted;
exports.description = dct$1.description;
exports.educationLevel = dct$1.educationLevel;
exports.extent = dct$1.extent;
exports.format = dct$1.format;
exports.hasFormat = dct$1.hasFormat;
exports.hasPart = dct$1.hasPart;
exports.hasVersion = dct$1.hasVersion;
exports.identifier = dct$1.identifier;
exports.instructionalMethod = dct$1.instructionalMethod;
exports.isFormatOf = dct$1.isFormatOf;
exports.isPartOf = dct$1.isPartOf;
exports.isReferencedBy = dct$1.isReferencedBy;
exports.isReplacedBy = dct$1.isReplacedBy;
exports.isRequiredBy = dct$1.isRequiredBy;
exports.isVersionOf = dct$1.isVersionOf;
exports.issued = dct$1.issued;
exports.language = dct$1.language;
exports.license = dct$1.license;
exports.mediator = dct$1.mediator;
exports.medium = dct$1.medium;
exports.modified = dct$1.modified;
exports.provenance = dct$1.provenance;
exports.publisher = dct$1.publisher;
exports.references = dct$1.references;
exports.relation = dct$1.relation;
exports.replaces = dct$1.replaces;
exports.requires = dct$1.requires;
exports.rights = dct$1.rights;
exports.rightsHolder = dct$1.rightsHolder;
exports.source = dct$1.source;
exports.spatial = dct$1.spatial;
exports.subject = dct$1.subject;
exports.tableOfContents = dct$1.tableOfContents;
exports.temporal = dct$1.temporal;
exports.title = dct$1.title;
exports.type = dct$1.type;
exports.valid = dct$1.valid;
