// rdf-namespaces/dist/dct.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * Abstract
 * 
 * A summary of the resource.
 *
 * http://purl.org/dc/terms/abstract
 */
export const abstract: Property = 'http://purl.org/dc/terms/abstract';

/**
 * Access Rights
 * 
 * Information about who can access the resource or an indication of its security status.
 *
 * http://purl.org/dc/terms/accessRights
 */
export const accessRights: Property = 'http://purl.org/dc/terms/accessRights';

/**
 * Accrual Method
 * 
 * The method by which items are added to a collection.
 *
 * http://purl.org/dc/terms/accrualMethod
 */
export const accrualMethod: Property = 'http://purl.org/dc/terms/accrualMethod';

/**
 * Accrual Periodicity
 * 
 * The frequency with which items are added to a collection.
 *
 * http://purl.org/dc/terms/accrualPeriodicity
 */
export const accrualPeriodicity: Property = 'http://purl.org/dc/terms/accrualPeriodicity';

/**
 * Accrual Policy
 * 
 * The policy governing the addition of items to a collection.
 *
 * http://purl.org/dc/terms/accrualPolicy
 */
export const accrualPolicy: Property = 'http://purl.org/dc/terms/accrualPolicy';

/**
 * Alternative Title
 * 
 * An alternative name for the resource.
 *
 * http://purl.org/dc/terms/alternative
 */
export const alternative: Property = 'http://purl.org/dc/terms/alternative';

/**
 * Audience
 * 
 * A class of entity for whom the resource is intended or useful.
 *
 * http://purl.org/dc/terms/audience
 */
export const audience: Property = 'http://purl.org/dc/terms/audience';

/**
 * Date Available
 * 
 * Date (often a range) that the resource became or will become available.
 *
 * http://purl.org/dc/terms/available
 */
export const available: Property = 'http://purl.org/dc/terms/available';

/**
 * Bibliographic Citation
 * 
 * A bibliographic reference for the resource.
 *
 * http://purl.org/dc/terms/bibliographicCitation
 */
export const bibliographicCitation: Property = 'http://purl.org/dc/terms/bibliographicCitation';

/**
 * Conforms To
 * 
 * An established standard to which the described resource conforms.
 *
 * http://purl.org/dc/terms/conformsTo
 */
export const conformsTo: Property = 'http://purl.org/dc/terms/conformsTo';

/**
 * Contributor
 * 
 * An entity responsible for making contributions to the resource.
 *
 * http://purl.org/dc/terms/contributor
 */
export const contributor: Property = 'http://purl.org/dc/terms/contributor';

/**
 * Coverage
 * 
 * The spatial or temporal topic of the resource, the spatial applicability of the resource, or the jurisdiction under which the resource is relevant.
 *
 * http://purl.org/dc/terms/coverage
 */
export const coverage: Property = 'http://purl.org/dc/terms/coverage';

/**
 * Date Created
 * 
 * Date of creation of the resource.
 *
 * http://purl.org/dc/terms/created
 */
export const created: Property = 'http://purl.org/dc/terms/created';

/**
 * Creator
 * 
 * An entity primarily responsible for making the resource.
 *
 * http://purl.org/dc/terms/creator
 */
export const creator: Property = 'http://purl.org/dc/terms/creator';

/**
 * Date
 * 
 * A point or period of time associated with an event in the lifecycle of the resource.
 *
 * http://purl.org/dc/terms/date
 */
export const date: Property = 'http://purl.org/dc/terms/date';

/**
 * Date Accepted
 * 
 * Date of acceptance of the resource.
 *
 * http://purl.org/dc/terms/dateAccepted
 */
export const dateAccepted: Property = 'http://purl.org/dc/terms/dateAccepted';

/**
 * Date Copyrighted
 * 
 * Date of copyright.
 *
 * http://purl.org/dc/terms/dateCopyrighted
 */
export const dateCopyrighted: Property = 'http://purl.org/dc/terms/dateCopyrighted';

/**
 * Date Submitted
 * 
 * Date of submission of the resource.
 *
 * http://purl.org/dc/terms/dateSubmitted
 */
export const dateSubmitted: Property = 'http://purl.org/dc/terms/dateSubmitted';

/**
 * Description
 * 
 * An account of the resource.
 *
 * http://purl.org/dc/terms/description
 */
export const description: Property = 'http://purl.org/dc/terms/description';

/**
 * Audience Education Level
 * 
 * A class of entity, defined in terms of progression through an educational or training context, for which the described resource is intended.
 *
 * http://purl.org/dc/terms/educationLevel
 */
export const educationLevel: Property = 'http://purl.org/dc/terms/educationLevel';

/**
 * Extent
 * 
 * The size or duration of the resource.
 *
 * http://purl.org/dc/terms/extent
 */
export const extent: Property = 'http://purl.org/dc/terms/extent';

/**
 * Format
 * 
 * The file format, physical medium, or dimensions of the resource.
 *
 * http://purl.org/dc/terms/format
 */
export const format: Property = 'http://purl.org/dc/terms/format';

/**
 * Has Format
 * 
 * A related resource that is substantially the same as the pre-existing described resource, but in another format.
 *
 * http://purl.org/dc/terms/hasFormat
 */
export const hasFormat: Property = 'http://purl.org/dc/terms/hasFormat';

/**
 * Has Part
 * 
 * A related resource that is included either physically or logically in the described resource.
 *
 * http://purl.org/dc/terms/hasPart
 */
export const hasPart: Property = 'http://purl.org/dc/terms/hasPart';

/**
 * Has Version
 * 
 * A related resource that is a version, edition, or adaptation of the described resource.
 *
 * http://purl.org/dc/terms/hasVersion
 */
export const hasVersion: Property = 'http://purl.org/dc/terms/hasVersion';

/**
 * Identifier
 * 
 * An unambiguous reference to the resource within a given context.
 *
 * http://purl.org/dc/terms/identifier
 */
export const identifier: Property = 'http://purl.org/dc/terms/identifier';

/**
 * Instructional Method
 * 
 * A process, used to engender knowledge, attitudes and skills, that the described resource is designed to support.
 *
 * http://purl.org/dc/terms/instructionalMethod
 */
export const instructionalMethod: Property = 'http://purl.org/dc/terms/instructionalMethod';

/**
 * Is Format Of
 * 
 * A related resource that is substantially the same as the described resource, but in another format.
 *
 * http://purl.org/dc/terms/isFormatOf
 */
export const isFormatOf: Property = 'http://purl.org/dc/terms/isFormatOf';

/**
 * Is Part Of
 * 
 * A related resource in which the described resource is physically or logically included.
 *
 * http://purl.org/dc/terms/isPartOf
 */
export const isPartOf: Property = 'http://purl.org/dc/terms/isPartOf';

/**
 * Is Referenced By
 * 
 * A related resource that references, cites, or otherwise points to the described resource.
 *
 * http://purl.org/dc/terms/isReferencedBy
 */
export const isReferencedBy: Property = 'http://purl.org/dc/terms/isReferencedBy';

/**
 * Is Replaced By
 * 
 * A related resource that supplants, displaces, or supersedes the described resource.
 *
 * http://purl.org/dc/terms/isReplacedBy
 */
export const isReplacedBy: Property = 'http://purl.org/dc/terms/isReplacedBy';

/**
 * Is Required By
 * 
 * A related resource that requires the described resource to support its function, delivery, or coherence.
 *
 * http://purl.org/dc/terms/isRequiredBy
 */
export const isRequiredBy: Property = 'http://purl.org/dc/terms/isRequiredBy';

/**
 * Is Version Of
 * 
 * A related resource of which the described resource is a version, edition, or adaptation.
 *
 * http://purl.org/dc/terms/isVersionOf
 */
export const isVersionOf: Property = 'http://purl.org/dc/terms/isVersionOf';

/**
 * Date Issued
 * 
 * Date of formal issuance (e.g., publication) of the resource.
 *
 * http://purl.org/dc/terms/issued
 */
export const issued: Property = 'http://purl.org/dc/terms/issued';

/**
 * Language
 * 
 * A language of the resource.
 *
 * http://purl.org/dc/terms/language
 */
export const language: Property = 'http://purl.org/dc/terms/language';

/**
 * License
 * 
 * A legal document giving official permission to do something with the resource.
 *
 * http://purl.org/dc/terms/license
 */
export const license: Property = 'http://purl.org/dc/terms/license';

/**
 * Mediator
 * 
 * An entity that mediates access to the resource and for whom the resource is intended or useful.
 *
 * http://purl.org/dc/terms/mediator
 */
export const mediator: Property = 'http://purl.org/dc/terms/mediator';

/**
 * Medium
 * 
 * The material or physical carrier of the resource.
 *
 * http://purl.org/dc/terms/medium
 */
export const medium: Property = 'http://purl.org/dc/terms/medium';

/**
 * Date Modified
 * 
 * Date on which the resource was changed.
 *
 * http://purl.org/dc/terms/modified
 */
export const modified: Property = 'http://purl.org/dc/terms/modified';

/**
 * Provenance
 * 
 * A statement of any changes in ownership and custody of the resource since its creation that are significant for its authenticity, integrity, and interpretation.
 *
 * http://purl.org/dc/terms/provenance
 */
export const provenance: Property = 'http://purl.org/dc/terms/provenance';

/**
 * Publisher
 * 
 * An entity responsible for making the resource available.
 *
 * http://purl.org/dc/terms/publisher
 */
export const publisher: Property = 'http://purl.org/dc/terms/publisher';

/**
 * References
 * 
 * A related resource that is referenced, cited, or otherwise pointed to by the described resource.
 *
 * http://purl.org/dc/terms/references
 */
export const references: Property = 'http://purl.org/dc/terms/references';

/**
 * Relation
 * 
 * A related resource.
 *
 * http://purl.org/dc/terms/relation
 */
export const relation: Property = 'http://purl.org/dc/terms/relation';

/**
 * Replaces
 * 
 * A related resource that is supplanted, displaced, or superseded by the described resource.
 *
 * http://purl.org/dc/terms/replaces
 */
export const replaces: Property = 'http://purl.org/dc/terms/replaces';

/**
 * Requires
 * 
 * A related resource that is required by the described resource to support its function, delivery, or coherence.
 *
 * http://purl.org/dc/terms/requires
 */
export const requires: Property = 'http://purl.org/dc/terms/requires';

/**
 * Rights
 * 
 * Information about rights held in and over the resource.
 *
 * http://purl.org/dc/terms/rights
 */
export const rights: Property = 'http://purl.org/dc/terms/rights';

/**
 * Rights Holder
 * 
 * A person or organization owning or managing rights over the resource.
 *
 * http://purl.org/dc/terms/rightsHolder
 */
export const rightsHolder: Property = 'http://purl.org/dc/terms/rightsHolder';

/**
 * Source
 * 
 * A related resource from which the described resource is derived.
 *
 * http://purl.org/dc/terms/source
 */
export const source: Property = 'http://purl.org/dc/terms/source';

/**
 * Spatial Coverage
 * 
 * Spatial characteristics of the resource.
 *
 * http://purl.org/dc/terms/spatial
 */
export const spatial: Property = 'http://purl.org/dc/terms/spatial';

/**
 * Subject
 * 
 * The topic of the resource.
 *
 * http://purl.org/dc/terms/subject
 */
export const subject: Property = 'http://purl.org/dc/terms/subject';

/**
 * Table Of Contents
 * 
 * A list of subunits of the resource.
 *
 * http://purl.org/dc/terms/tableOfContents
 */
export const tableOfContents: Property = 'http://purl.org/dc/terms/tableOfContents';

/**
 * Temporal Coverage
 * 
 * Temporal characteristics of the resource.
 *
 * http://purl.org/dc/terms/temporal
 */
export const temporal: Property = 'http://purl.org/dc/terms/temporal';

/**
 * Title
 * 
 * A name given to the resource.
 *
 * http://purl.org/dc/terms/title
 */
export const title: Property = 'http://purl.org/dc/terms/title';

/**
 * Type
 * 
 * The nature or genre of the resource.
 *
 * http://purl.org/dc/terms/type
 */
export const type: Property = 'http://purl.org/dc/terms/type';

/**
 * Date Valid
 * 
 * Date (often a range) of validity of a resource.
 *
 * http://purl.org/dc/terms/valid
 */
export const valid: Property = 'http://purl.org/dc/terms/valid';

/**
 * Agent
 * 
 * A resource that acts or has the power to act.
 *
 * http://purl.org/dc/terms/Agent
 */
export const Agent: string = 'http://purl.org/dc/terms/Agent';

/**
 * Agent Class
 * 
 * A group of agents.
 *
 * http://purl.org/dc/terms/AgentClass
 */
export const AgentClass: Class = 'http://purl.org/dc/terms/AgentClass';

/**
 * Bibliographic Resource
 * 
 * A book, article, or other documentary resource.
 *
 * http://purl.org/dc/terms/BibliographicResource
 */
export const BibliographicResource: Class = 'http://purl.org/dc/terms/BibliographicResource';

/**
 * File Format
 * 
 * A digital resource format.
 *
 * http://purl.org/dc/terms/FileFormat
 */
export const FileFormat: Class = 'http://purl.org/dc/terms/FileFormat';

/**
 * Frequency
 * 
 * A rate at which something recurs.
 *
 * http://purl.org/dc/terms/Frequency
 */
export const Frequency: Class = 'http://purl.org/dc/terms/Frequency';

/**
 * Jurisdiction
 * 
 * The extent or range of judicial, law enforcement, or other authority.
 *
 * http://purl.org/dc/terms/Jurisdiction
 */
export const Jurisdiction: Class = 'http://purl.org/dc/terms/Jurisdiction';

/**
 * License Document
 * 
 * A legal document giving official permission to do something with a Resource.
 *
 * http://purl.org/dc/terms/LicenseDocument
 */
export const LicenseDocument: Class = 'http://purl.org/dc/terms/LicenseDocument';

/**
 * Linguistic System
 * 
 * A system of signs, symbols, sounds, gestures, or rules used in communication.
 *
 * http://purl.org/dc/terms/LinguisticSystem
 */
export const LinguisticSystem: Class = 'http://purl.org/dc/terms/LinguisticSystem';

/**
 * Location
 * 
 * A spatial region or named place.
 *
 * http://purl.org/dc/terms/Location
 */
export const Location: Class = 'http://purl.org/dc/terms/Location';

/**
 * Location, Period, or Jurisdiction
 * 
 * A location, period of time, or jurisdiction.
 *
 * http://purl.org/dc/terms/LocationPeriodOrJurisdiction
 */
export const LocationPeriodOrJurisdiction: Class = 'http://purl.org/dc/terms/LocationPeriodOrJurisdiction';

/**
 * Media Type
 * 
 * A file format or physical medium.
 *
 * http://purl.org/dc/terms/MediaType
 */
export const MediaType: Class = 'http://purl.org/dc/terms/MediaType';

/**
 * Media Type or Extent
 * 
 * A media type or extent.
 *
 * http://purl.org/dc/terms/MediaTypeOrExtent
 */
export const MediaTypeOrExtent: Class = 'http://purl.org/dc/terms/MediaTypeOrExtent';

/**
 * Method of Accrual
 * 
 * A method by which resources are added to a collection.
 *
 * http://purl.org/dc/terms/MethodOfAccrual
 */
export const MethodOfAccrual: Class = 'http://purl.org/dc/terms/MethodOfAccrual';

/**
 * Method of Instruction
 * 
 * A process that is used to engender knowledge, attitudes, and skills.
 *
 * http://purl.org/dc/terms/MethodOfInstruction
 */
export const MethodOfInstruction: Class = 'http://purl.org/dc/terms/MethodOfInstruction';

/**
 * Period of Time
 * 
 * An interval of time that is named or defined by its start and end dates.
 *
 * http://purl.org/dc/terms/PeriodOfTime
 */
export const PeriodOfTime: Class = 'http://purl.org/dc/terms/PeriodOfTime';

/**
 * Physical Medium
 * 
 * A physical material or carrier.
 *
 * http://purl.org/dc/terms/PhysicalMedium
 */
export const PhysicalMedium: Class = 'http://purl.org/dc/terms/PhysicalMedium';

/**
 * Physical Resource
 * 
 * A material thing.
 *
 * http://purl.org/dc/terms/PhysicalResource
 */
export const PhysicalResource: Class = 'http://purl.org/dc/terms/PhysicalResource';

/**
 * Policy
 * 
 * A plan or course of action by an authority, intended to influence and determine decisions, actions, and other matters.
 *
 * http://purl.org/dc/terms/Policy
 */
export const Policy: Class = 'http://purl.org/dc/terms/Policy';

/**
 * Provenance Statement
 * 
 * A statement of any changes in ownership and custody of a resource since its creation that are significant for its authenticity, integrity, and interpretation.
 *
 * http://purl.org/dc/terms/ProvenanceStatement
 */
export const ProvenanceStatement: Class = 'http://purl.org/dc/terms/ProvenanceStatement';

/**
 * Rights Statement
 * 
 * A statement about the intellectual property rights (IPR) held in or over a Resource, a legal document giving official permission to do something with a resource, or a statement about access rights.
 *
 * http://purl.org/dc/terms/RightsStatement
 */
export const RightsStatement: Class = 'http://purl.org/dc/terms/RightsStatement';

/**
 * Size or Duration
 * 
 * A dimension or extent, or a time taken to play or execute.
 *
 * http://purl.org/dc/terms/SizeOrDuration
 */
export const SizeOrDuration: Class = 'http://purl.org/dc/terms/SizeOrDuration';

/**
 * Standard
 * 
 * A basis for comparison; a reference point against which other things can be evaluated.
 *
 * http://purl.org/dc/terms/Standard
 */
export const Standard: Class = 'http://purl.org/dc/terms/Standard';

/**
 * DCMI Box
 * 
 * The set of regions in space defined by their geographic coordinates according to the DCMI Box Encoding Scheme.
 *
 * http://purl.org/dc/terms/Box
 */
export const Box: Datatype = 'http://purl.org/dc/terms/Box';

/**
 * ISO 3166
 * 
 * The set of codes listed in ISO 3166-1 for the representation of names of countries.
 *
 * http://purl.org/dc/terms/ISO3166
 */
export const ISO3166: Datatype = 'http://purl.org/dc/terms/ISO3166';

/**
 * DCMI Period
 * 
 * The set of time intervals defined by their limits according to the DCMI Period Encoding Scheme.
 *
 * http://purl.org/dc/terms/Period
 */
export const Period: Datatype = 'http://purl.org/dc/terms/Period';

/**
 * DCMI Point
 * 
 * The set of points in space defined by their geographic coordinates according to the DCMI Point Encoding Scheme.
 *
 * http://purl.org/dc/terms/Point
 */
export const Point: Datatype = 'http://purl.org/dc/terms/Point';

/**
 * RFC 1766
 * 
 * The set of tags, constructed according to RFC 1766, for the identification of languages.
 *
 * http://purl.org/dc/terms/RFC1766
 */
export const RFC1766: Datatype = 'http://purl.org/dc/terms/RFC1766';

/**
 * RFC 3066
 * 
 * The set of tags constructed according to RFC 3066 for the identification of languages.
 *
 * http://purl.org/dc/terms/RFC3066
 */
export const RFC3066: Datatype = 'http://purl.org/dc/terms/RFC3066';

/**
 * RFC 4646
 * 
 * The set of tags constructed according to RFC 4646 for the identification of languages.
 *
 * http://purl.org/dc/terms/RFC4646
 */
export const RFC4646: Datatype = 'http://purl.org/dc/terms/RFC4646';

/**
 * RFC 5646
 * 
 * The set of tags constructed according to RFC 5646 for the identification of languages.
 *
 * http://purl.org/dc/terms/RFC5646
 */
export const RFC5646: Datatype = 'http://purl.org/dc/terms/RFC5646';

/**
 * URI
 * 
 * The set of identifiers constructed according to the generic syntax for Uniform Resource Identifiers as specified by the Internet Engineering Task Force.
 *
 * http://purl.org/dc/terms/URI
 */
export const URI: Datatype = 'http://purl.org/dc/terms/URI';

/**
 * W3C-DTF
 * 
 * The set of dates and times constructed according to the W3C Date and Time Formats Specification.
 *
 * http://purl.org/dc/terms/W3CDTF
 */
export const W3CDTF: Datatype = 'http://purl.org/dc/terms/W3CDTF';
