// rdf-namespaces/dist/foaf.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var foaf$1 = require('./foaf-acbc1a2a.js');



exports.Agent = foaf$1.Agent;
exports.Document = foaf$1.Document;
exports.Group = foaf$1.Group;
exports.Image = foaf$1.Image;
exports.LabelProperty = foaf$1.LabelProperty;
exports.OnlineAccount = foaf$1.OnlineAccount;
exports.OnlineChatAccount = foaf$1.OnlineChatAccount;
exports.OnlineEcommerceAccount = foaf$1.OnlineEcommerceAccount;
exports.OnlineGamingAccount = foaf$1.OnlineGamingAccount;
exports.Organization = foaf$1.Organization;
exports.Person = foaf$1.Person;
exports.PersonalProfileDocument = foaf$1.PersonalProfileDocument;
exports.Project = foaf$1.Project;
exports.account = foaf$1.account;
exports.accountName = foaf$1.accountName;
exports.accountServiceHomepage = foaf$1.accountServiceHomepage;
exports.age = foaf$1.age;
exports.aimChatID = foaf$1.aimChatID;
exports.based_near = foaf$1.based_near;
exports.birthday = foaf$1.birthday;
exports.currentProject = foaf$1.currentProject;
exports.depiction = foaf$1.depiction;
exports.depicts = foaf$1.depicts;
exports.dnaChecksum = foaf$1.dnaChecksum;
exports.familyName = foaf$1.familyName;
exports.family_name = foaf$1.family_name;
exports.firstName = foaf$1.firstName;
exports.focus = foaf$1.focus;
exports.fundedBy = foaf$1.fundedBy;
exports.geekcode = foaf$1.geekcode;
exports.gender = foaf$1.gender;
exports.givenName = foaf$1.givenName;
exports.givenname = foaf$1.givenname;
exports.holdsAccount = foaf$1.holdsAccount;
exports.homepage = foaf$1.homepage;
exports.icqChatID = foaf$1.icqChatID;
exports.img = foaf$1.img;
exports.interest = foaf$1.interest;
exports.isPrimaryTopicOf = foaf$1.isPrimaryTopicOf;
exports.jabberID = foaf$1.jabberID;
exports.knows = foaf$1.knows;
exports.lastName = foaf$1.lastName;
exports.logo = foaf$1.logo;
exports.made = foaf$1.made;
exports.maker = foaf$1.maker;
exports.mbox = foaf$1.mbox;
exports.mbox_sha1sum = foaf$1.mbox_sha1sum;
exports.member = foaf$1.member;
exports.membershipClass = foaf$1.membershipClass;
exports.msnChatID = foaf$1.msnChatID;
exports.myersBriggs = foaf$1.myersBriggs;
exports.name = foaf$1.name;
exports.nick = foaf$1.nick;
exports.openid = foaf$1.openid;
exports.page = foaf$1.page;
exports.pastProject = foaf$1.pastProject;
exports.phone = foaf$1.phone;
exports.plan = foaf$1.plan;
exports.primaryTopic = foaf$1.primaryTopic;
exports.publications = foaf$1.publications;
exports.schoolHomepage = foaf$1.schoolHomepage;
exports.sha1 = foaf$1.sha1;
exports.skypeID = foaf$1.skypeID;
exports.status = foaf$1.status;
exports.surname = foaf$1.surname;
exports.theme = foaf$1.theme;
exports.thumbnail = foaf$1.thumbnail;
exports.tipjar = foaf$1.tipjar;
exports.title = foaf$1.title;
exports.topic = foaf$1.topic;
exports.topic_interest = foaf$1.topic_interest;
exports.weblog = foaf$1.weblog;
exports.workInfoHomepage = foaf$1.workInfoHomepage;
exports.workplaceHomepage = foaf$1.workplaceHomepage;
exports.yahooChatID = foaf$1.yahooChatID;
