// rdf-namespaces/dist/foaf-107261f2.js
/**
 * personal mailbox
 *
 * A
 * personal mailbox, ie. an Internet mailbox associated with exactly one owner, the first owner of this mailbox. This is a 'static inverse functional property', in that  there is (across time and change) at most one individual that ever has any particular value for foaf:mbox.
 *
 * http://xmlns.com/foaf/0.1/mbox
 */
var mbox = 'http://xmlns.com/foaf/0.1/mbox';
/**
 * sha1sum of a personal mailbox URI name
 *
 * The sha1sum of the URI of an Internet mailbox associated with exactly one owner, the  first owner of the mailbox.
 *
 * http://xmlns.com/foaf/0.1/mbox_sha1sum
 */
var mbox_sha1sum = 'http://xmlns.com/foaf/0.1/mbox_sha1sum';
/**
 * gender
 *
 * The gender of this Agent (typically but not necessarily 'male' or 'female').
 *
 * http://xmlns.com/foaf/0.1/gender
 */
var gender = 'http://xmlns.com/foaf/0.1/gender';
/**
 * geekcode
 *
 * A textual geekcode for this person, see http://www.geekcode.com/geek.html
 *
 * http://xmlns.com/foaf/0.1/geekcode
 */
var geekcode = 'http://xmlns.com/foaf/0.1/geekcode';
/**
 * DNA checksum
 *
 * A checksum for the DNA of some thing. Joke.
 *
 * http://xmlns.com/foaf/0.1/dnaChecksum
 */
var dnaChecksum = 'http://xmlns.com/foaf/0.1/dnaChecksum';
/**
 * sha1sum (hex)
 *
 * A sha1sum hash, in hex.
 *
 * http://xmlns.com/foaf/0.1/sha1
 */
var sha1 = 'http://xmlns.com/foaf/0.1/sha1';
/**
 * based near
 *
 * A location that something is based near, for some broadly human notion of near.
 *
 * http://xmlns.com/foaf/0.1/based_near
 */
var based_near = 'http://xmlns.com/foaf/0.1/based_near';
/**
 * title
 *
 * Title (Mr, Mrs, Ms, Dr. etc)
 *
 * http://xmlns.com/foaf/0.1/title
 */
var title = 'http://xmlns.com/foaf/0.1/title';
/**
 * nickname
 *
 * A short informal nickname characterising an agent (includes login identifiers, IRC and other chat nicknames).
 *
 * http://xmlns.com/foaf/0.1/nick
 */
var nick = 'http://xmlns.com/foaf/0.1/nick';
/**
 * jabber ID
 *
 * A jabber ID for something.
 *
 * http://xmlns.com/foaf/0.1/jabberID
 */
var jabberID = 'http://xmlns.com/foaf/0.1/jabberID';
/**
 * AIM chat ID
 *
 * An AIM chat ID
 *
 * http://xmlns.com/foaf/0.1/aimChatID
 */
var aimChatID = 'http://xmlns.com/foaf/0.1/aimChatID';
/**
 * Skype ID
 *
 * A Skype ID
 *
 * http://xmlns.com/foaf/0.1/skypeID
 */
var skypeID = 'http://xmlns.com/foaf/0.1/skypeID';
/**
 * ICQ chat ID
 *
 * An ICQ chat ID
 *
 * http://xmlns.com/foaf/0.1/icqChatID
 */
var icqChatID = 'http://xmlns.com/foaf/0.1/icqChatID';
/**
 * Yahoo chat ID
 *
 * A Yahoo chat ID
 *
 * http://xmlns.com/foaf/0.1/yahooChatID
 */
var yahooChatID = 'http://xmlns.com/foaf/0.1/yahooChatID';
/**
 * MSN chat ID
 *
 * An MSN chat ID
 *
 * http://xmlns.com/foaf/0.1/msnChatID
 */
var msnChatID = 'http://xmlns.com/foaf/0.1/msnChatID';
/**
 * name
 *
 * A name for some thing.
 *
 * http://xmlns.com/foaf/0.1/name
 */
var name = 'http://xmlns.com/foaf/0.1/name';
/**
 * firstName
 *
 * The first name of a person.
 *
 * http://xmlns.com/foaf/0.1/firstName
 */
var firstName = 'http://xmlns.com/foaf/0.1/firstName';
/**
 * lastName
 *
 * The last name of a person.
 *
 * http://xmlns.com/foaf/0.1/lastName
 */
var lastName = 'http://xmlns.com/foaf/0.1/lastName';
/**
 * Given name
 *
 * The given name of some person.
 *
 * http://xmlns.com/foaf/0.1/givenName
 */
var givenName = 'http://xmlns.com/foaf/0.1/givenName';
/**
 * Given name
 *
 * The given name of some person.
 *
 * http://xmlns.com/foaf/0.1/givenname
 */
var givenname = 'http://xmlns.com/foaf/0.1/givenname';
/**
 * Surname
 *
 * The surname of some person.
 *
 * http://xmlns.com/foaf/0.1/surname
 */
var surname = 'http://xmlns.com/foaf/0.1/surname';
/**
 * family_name
 *
 * The family name of some person.
 *
 * http://xmlns.com/foaf/0.1/family_name
 */
var family_name = 'http://xmlns.com/foaf/0.1/family_name';
/**
 * familyName
 *
 * The family name of some person.
 *
 * http://xmlns.com/foaf/0.1/familyName
 */
var familyName = 'http://xmlns.com/foaf/0.1/familyName';
/**
 * phone
 *
 * A phone,  specified using fully qualified tel: URI scheme (refs: http://www.w3.org/Addressing/schemes.html#tel).
 *
 * http://xmlns.com/foaf/0.1/phone
 */
var phone = 'http://xmlns.com/foaf/0.1/phone';
/**
 * homepage
 *
 * A homepage for some thing.
 *
 * http://xmlns.com/foaf/0.1/homepage
 */
var homepage = 'http://xmlns.com/foaf/0.1/homepage';
/**
 * weblog
 *
 * A weblog of some thing (whether person, group, company etc.).
 *
 * http://xmlns.com/foaf/0.1/weblog
 */
var weblog = 'http://xmlns.com/foaf/0.1/weblog';
/**
 * openid
 *
 * An OpenID for an Agent.
 *
 * http://xmlns.com/foaf/0.1/openid
 */
var openid = 'http://xmlns.com/foaf/0.1/openid';
/**
 * tipjar
 *
 * A tipjar document for this agent, describing means for payment and reward.
 *
 * http://xmlns.com/foaf/0.1/tipjar
 */
var tipjar = 'http://xmlns.com/foaf/0.1/tipjar';
/**
 * plan
 *
 * A .plan comment, in the tradition of finger and '.plan' files.
 *
 * http://xmlns.com/foaf/0.1/plan
 */
var plan = 'http://xmlns.com/foaf/0.1/plan';
/**
 * made
 *
 * Something that was made by this agent.
 *
 * http://xmlns.com/foaf/0.1/made
 */
var made = 'http://xmlns.com/foaf/0.1/made';
/**
 * maker
 *
 * An agent that
 * made this thing.
 *
 * http://xmlns.com/foaf/0.1/maker
 */
var maker = 'http://xmlns.com/foaf/0.1/maker';
/**
 * image
 *
 * An image that can be used to represent some thing (ie. those depictions which are particularly representative of something, eg. one's photo on a homepage).
 *
 * http://xmlns.com/foaf/0.1/img
 */
var img = 'http://xmlns.com/foaf/0.1/img';
/**
 * depiction
 *
 * A depiction of some thing.
 *
 * http://xmlns.com/foaf/0.1/depiction
 */
var depiction = 'http://xmlns.com/foaf/0.1/depiction';
/**
 * depicts
 *
 * A thing depicted in this representation.
 *
 * http://xmlns.com/foaf/0.1/depicts
 */
var depicts = 'http://xmlns.com/foaf/0.1/depicts';
/**
 * thumbnail
 *
 * A derived thumbnail image.
 *
 * http://xmlns.com/foaf/0.1/thumbnail
 */
var thumbnail = 'http://xmlns.com/foaf/0.1/thumbnail';
/**
 * myersBriggs
 *
 * A Myers Briggs (MBTI) personality classification.
 *
 * http://xmlns.com/foaf/0.1/myersBriggs
 */
var myersBriggs = 'http://xmlns.com/foaf/0.1/myersBriggs';
/**
 * workplace homepage
 *
 * A workplace homepage of some person; the homepage of an organization they work for.
 *
 * http://xmlns.com/foaf/0.1/workplaceHomepage
 */
var workplaceHomepage = 'http://xmlns.com/foaf/0.1/workplaceHomepage';
/**
 * work info homepage
 *
 * A work info homepage of some person; a page about their work for some organization.
 *
 * http://xmlns.com/foaf/0.1/workInfoHomepage
 */
var workInfoHomepage = 'http://xmlns.com/foaf/0.1/workInfoHomepage';
/**
 * schoolHomepage
 *
 * A homepage of a school attended by the person.
 *
 * http://xmlns.com/foaf/0.1/schoolHomepage
 */
var schoolHomepage = 'http://xmlns.com/foaf/0.1/schoolHomepage';
/**
 * knows
 *
 * A person known by this person (indicating some level of reciprocated interaction between the parties).
 *
 * http://xmlns.com/foaf/0.1/knows
 */
var knows = 'http://xmlns.com/foaf/0.1/knows';
/**
 * interest
 *
 * A page about a topic of interest to this person.
 *
 * http://xmlns.com/foaf/0.1/interest
 */
var interest = 'http://xmlns.com/foaf/0.1/interest';
/**
 * topic_interest
 *
 * A thing of interest to this person.
 *
 * http://xmlns.com/foaf/0.1/topic_interest
 */
var topic_interest = 'http://xmlns.com/foaf/0.1/topic_interest';
/**
 * publications
 *
 * A link to the publications of this person.
 *
 * http://xmlns.com/foaf/0.1/publications
 */
var publications = 'http://xmlns.com/foaf/0.1/publications';
/**
 * current project
 *
 * A current project this person works on.
 *
 * http://xmlns.com/foaf/0.1/currentProject
 */
var currentProject = 'http://xmlns.com/foaf/0.1/currentProject';
/**
 * past project
 *
 * A project this person has previously worked on.
 *
 * http://xmlns.com/foaf/0.1/pastProject
 */
var pastProject = 'http://xmlns.com/foaf/0.1/pastProject';
/**
 * funded by
 *
 * An organization funding a project or person.
 *
 * http://xmlns.com/foaf/0.1/fundedBy
 */
var fundedBy = 'http://xmlns.com/foaf/0.1/fundedBy';
/**
 * logo
 *
 * A logo representing some thing.
 *
 * http://xmlns.com/foaf/0.1/logo
 */
var logo = 'http://xmlns.com/foaf/0.1/logo';
/**
 * topic
 *
 * A topic of some page or document.
 *
 * http://xmlns.com/foaf/0.1/topic
 */
var topic = 'http://xmlns.com/foaf/0.1/topic';
/**
 * primary topic
 *
 * The primary topic of some page or document.
 *
 * http://xmlns.com/foaf/0.1/primaryTopic
 */
var primaryTopic = 'http://xmlns.com/foaf/0.1/primaryTopic';
/**
 * focus
 *
 * The underlying or 'focal' entity associated with some SKOS-described concept.
 *
 * http://xmlns.com/foaf/0.1/focus
 */
var focus = 'http://xmlns.com/foaf/0.1/focus';
/**
 * is primary topic of
 *
 * A document that this thing is the primary topic of.
 *
 * http://xmlns.com/foaf/0.1/isPrimaryTopicOf
 */
var isPrimaryTopicOf = 'http://xmlns.com/foaf/0.1/isPrimaryTopicOf';
/**
 * page
 *
 * A page or document about this thing.
 *
 * http://xmlns.com/foaf/0.1/page
 */
var page = 'http://xmlns.com/foaf/0.1/page';
/**
 * theme
 *
 * A theme.
 *
 * http://xmlns.com/foaf/0.1/theme
 */
var theme = 'http://xmlns.com/foaf/0.1/theme';
/**
 * account
 *
 * Indicates an account held by this agent.
 *
 * http://xmlns.com/foaf/0.1/account
 */
var account = 'http://xmlns.com/foaf/0.1/account';
/**
 * account
 *
 * Indicates an account held by this agent.
 *
 * http://xmlns.com/foaf/0.1/holdsAccount
 */
var holdsAccount = 'http://xmlns.com/foaf/0.1/holdsAccount';
/**
 * account service homepage
 *
 * Indicates a homepage of the service provide for this online account.
 *
 * http://xmlns.com/foaf/0.1/accountServiceHomepage
 */
var accountServiceHomepage = 'http://xmlns.com/foaf/0.1/accountServiceHomepage';
/**
 * account name
 *
 * Indicates the name (identifier) associated with this online account.
 *
 * http://xmlns.com/foaf/0.1/accountName
 */
var accountName = 'http://xmlns.com/foaf/0.1/accountName';
/**
 * member
 *
 * Indicates a member of a Group
 *
 * http://xmlns.com/foaf/0.1/member
 */
var member = 'http://xmlns.com/foaf/0.1/member';
/**
 * membershipClass
 *
 * Indicates the class of individuals that are a member of a Group
 *
 * http://xmlns.com/foaf/0.1/membershipClass
 */
var membershipClass = 'http://xmlns.com/foaf/0.1/membershipClass';
/**
 * birthday
 *
 * The birthday of this Agent, represented in mm-dd string form, eg. '12-31'.
 *
 * http://xmlns.com/foaf/0.1/birthday
 */
var birthday = 'http://xmlns.com/foaf/0.1/birthday';
/**
 * age
 *
 * The age in years of some agent.
 *
 * http://xmlns.com/foaf/0.1/age
 */
var age = 'http://xmlns.com/foaf/0.1/age';
/**
 * status
 *
 * A string expressing what the user is happy for the general public (normally) to know about their current activity.
 *
 * http://xmlns.com/foaf/0.1/status
 */
var status = 'http://xmlns.com/foaf/0.1/status';
/**
 * Label Property
 *
 * A foaf:LabelProperty is any RDF property with texual values that serve as labels.
 *
 * http://xmlns.com/foaf/0.1/LabelProperty
 */
var LabelProperty = 'http://xmlns.com/foaf/0.1/LabelProperty';
/**
 * Person
 *
 * A person.
 *
 * http://xmlns.com/foaf/0.1/Person
 */
var Person = 'http://xmlns.com/foaf/0.1/Person';
/**
 * Document
 *
 * A document.
 *
 * http://xmlns.com/foaf/0.1/Document
 */
var Document = 'http://xmlns.com/foaf/0.1/Document';
/**
 * Organization
 *
 * An organization.
 *
 * http://xmlns.com/foaf/0.1/Organization
 */
var Organization = 'http://xmlns.com/foaf/0.1/Organization';
/**
 * Group
 *
 * A class of Agents.
 *
 * http://xmlns.com/foaf/0.1/Group
 */
var Group = 'http://xmlns.com/foaf/0.1/Group';
/**
 * Agent
 *
 * An agent (eg. person, group, software or physical artifact).
 *
 * http://xmlns.com/foaf/0.1/Agent
 */
var Agent = 'http://xmlns.com/foaf/0.1/Agent';
/**
 * Project
 *
 * A project (a collective endeavour of some kind).
 *
 * http://xmlns.com/foaf/0.1/Project
 */
var Project = 'http://xmlns.com/foaf/0.1/Project';
/**
 * Image
 *
 * An image.
 *
 * http://xmlns.com/foaf/0.1/Image
 */
var Image = 'http://xmlns.com/foaf/0.1/Image';
/**
 * PersonalProfileDocument
 *
 * A personal profile RDF document.
 *
 * http://xmlns.com/foaf/0.1/PersonalProfileDocument
 */
var PersonalProfileDocument = 'http://xmlns.com/foaf/0.1/PersonalProfileDocument';
/**
 * Online Account
 *
 * An online account.
 *
 * http://xmlns.com/foaf/0.1/OnlineAccount
 */
var OnlineAccount = 'http://xmlns.com/foaf/0.1/OnlineAccount';
/**
 * Online Gaming Account
 *
 * An online gaming account.
 *
 * http://xmlns.com/foaf/0.1/OnlineGamingAccount
 */
var OnlineGamingAccount = 'http://xmlns.com/foaf/0.1/OnlineGamingAccount';
/**
 * Online E-commerce Account
 *
 * An online e-commerce account.
 *
 * http://xmlns.com/foaf/0.1/OnlineEcommerceAccount
 */
var OnlineEcommerceAccount = 'http://xmlns.com/foaf/0.1/OnlineEcommerceAccount';
/**
 * Online Chat Account
 *
 * An online chat account.
 *
 * http://xmlns.com/foaf/0.1/OnlineChatAccount
 */
var OnlineChatAccount = 'http://xmlns.com/foaf/0.1/OnlineChatAccount';

var foafImport = /*#__PURE__*/Object.freeze({
	mbox: mbox,
	mbox_sha1sum: mbox_sha1sum,
	gender: gender,
	geekcode: geekcode,
	dnaChecksum: dnaChecksum,
	sha1: sha1,
	based_near: based_near,
	title: title,
	nick: nick,
	jabberID: jabberID,
	aimChatID: aimChatID,
	skypeID: skypeID,
	icqChatID: icqChatID,
	yahooChatID: yahooChatID,
	msnChatID: msnChatID,
	name: name,
	firstName: firstName,
	lastName: lastName,
	givenName: givenName,
	givenname: givenname,
	surname: surname,
	family_name: family_name,
	familyName: familyName,
	phone: phone,
	homepage: homepage,
	weblog: weblog,
	openid: openid,
	tipjar: tipjar,
	plan: plan,
	made: made,
	maker: maker,
	img: img,
	depiction: depiction,
	depicts: depicts,
	thumbnail: thumbnail,
	myersBriggs: myersBriggs,
	workplaceHomepage: workplaceHomepage,
	workInfoHomepage: workInfoHomepage,
	schoolHomepage: schoolHomepage,
	knows: knows,
	interest: interest,
	topic_interest: topic_interest,
	publications: publications,
	currentProject: currentProject,
	pastProject: pastProject,
	fundedBy: fundedBy,
	logo: logo,
	topic: topic,
	primaryTopic: primaryTopic,
	focus: focus,
	isPrimaryTopicOf: isPrimaryTopicOf,
	page: page,
	theme: theme,
	account: account,
	holdsAccount: holdsAccount,
	accountServiceHomepage: accountServiceHomepage,
	accountName: accountName,
	member: member,
	membershipClass: membershipClass,
	birthday: birthday,
	age: age,
	status: status,
	LabelProperty: LabelProperty,
	Person: Person,
	Document: Document,
	Organization: Organization,
	Group: Group,
	Agent: Agent,
	Project: Project,
	Image: Image,
	PersonalProfileDocument: PersonalProfileDocument,
	OnlineAccount: OnlineAccount,
	OnlineGamingAccount: OnlineGamingAccount,
	OnlineEcommerceAccount: OnlineEcommerceAccount,
	OnlineChatAccount: OnlineChatAccount
});

export { account as $, openid as A, tipjar as B, plan as C, made as D, maker as E, img as F, depiction as G, depicts as H, thumbnail as I, myersBriggs as J, workplaceHomepage as K, workInfoHomepage as L, schoolHomepage as M, knows as N, interest as O, topic_interest as P, publications as Q, currentProject as R, pastProject as S, fundedBy as T, logo as U, topic as V, primaryTopic as W, focus as X, isPrimaryTopicOf as Y, page as Z, theme as _, mbox_sha1sum as a, holdsAccount as a0, accountServiceHomepage as a1, accountName as a2, member as a3, membershipClass as a4, birthday as a5, age as a6, status as a7, LabelProperty as a8, Person as a9, Document as aa, Organization as ab, Group as ac, Agent as ad, Project as ae, Image as af, PersonalProfileDocument as ag, OnlineAccount as ah, OnlineGamingAccount as ai, OnlineEcommerceAccount as aj, OnlineChatAccount as ak, foafImport as al, geekcode as b, based_near as c, dnaChecksum as d, aimChatID as e, skypeID as f, gender as g, msnChatID as h, icqChatID as i, jabberID as j, name as k, firstName as l, mbox as m, nick as n, lastName as o, givenName as p, givenname as q, surname as r, sha1 as s, title as t, family_name as u, familyName as v, phone as w, homepage as x, yahooChatID as y, weblog as z };
