// rdf-namespaces/dist/foaf.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * personal mailbox
 * 
 * A 
 * personal mailbox, ie. an Internet mailbox associated with exactly one owner, the first owner of this mailbox. This is a 'static inverse functional property', in that  there is (across time and change) at most one individual that ever has any particular value for foaf:mbox.
 *
 * http://xmlns.com/foaf/0.1/mbox
 */
export const mbox: Property = 'http://xmlns.com/foaf/0.1/mbox';

/**
 * sha1sum of a personal mailbox URI name
 * 
 * The sha1sum of the URI of an Internet mailbox associated with exactly one owner, the  first owner of the mailbox.
 *
 * http://xmlns.com/foaf/0.1/mbox_sha1sum
 */
export const mbox_sha1sum: Property = 'http://xmlns.com/foaf/0.1/mbox_sha1sum';

/**
 * gender
 * 
 * The gender of this Agent (typically but not necessarily 'male' or 'female').
 *
 * http://xmlns.com/foaf/0.1/gender
 */
export const gender: Property = 'http://xmlns.com/foaf/0.1/gender';

/**
 * geekcode
 * 
 * A textual geekcode for this person, see http://www.geekcode.com/geek.html
 *
 * http://xmlns.com/foaf/0.1/geekcode
 */
export const geekcode: Property = 'http://xmlns.com/foaf/0.1/geekcode';

/**
 * DNA checksum
 * 
 * A checksum for the DNA of some thing. Joke.
 *
 * http://xmlns.com/foaf/0.1/dnaChecksum
 */
export const dnaChecksum: Property = 'http://xmlns.com/foaf/0.1/dnaChecksum';

/**
 * sha1sum (hex)
 * 
 * A sha1sum hash, in hex.
 *
 * http://xmlns.com/foaf/0.1/sha1
 */
export const sha1: Property = 'http://xmlns.com/foaf/0.1/sha1';

/**
 * based near
 * 
 * A location that something is based near, for some broadly human notion of near.
 *
 * http://xmlns.com/foaf/0.1/based_near
 */
export const based_near: Property = 'http://xmlns.com/foaf/0.1/based_near';

/**
 * title
 * 
 * Title (Mr, Mrs, Ms, Dr. etc)
 *
 * http://xmlns.com/foaf/0.1/title
 */
export const title: Property = 'http://xmlns.com/foaf/0.1/title';

/**
 * nickname
 * 
 * A short informal nickname characterising an agent (includes login identifiers, IRC and other chat nicknames).
 *
 * http://xmlns.com/foaf/0.1/nick
 */
export const nick: Property = 'http://xmlns.com/foaf/0.1/nick';

/**
 * jabber ID
 * 
 * A jabber ID for something.
 *
 * http://xmlns.com/foaf/0.1/jabberID
 */
export const jabberID: Property = 'http://xmlns.com/foaf/0.1/jabberID';

/**
 * AIM chat ID
 * 
 * An AIM chat ID
 *
 * http://xmlns.com/foaf/0.1/aimChatID
 */
export const aimChatID: Property = 'http://xmlns.com/foaf/0.1/aimChatID';

/**
 * Skype ID
 * 
 * A Skype ID
 *
 * http://xmlns.com/foaf/0.1/skypeID
 */
export const skypeID: Property = 'http://xmlns.com/foaf/0.1/skypeID';

/**
 * ICQ chat ID
 * 
 * An ICQ chat ID
 *
 * http://xmlns.com/foaf/0.1/icqChatID
 */
export const icqChatID: Property = 'http://xmlns.com/foaf/0.1/icqChatID';

/**
 * Yahoo chat ID
 * 
 * A Yahoo chat ID
 *
 * http://xmlns.com/foaf/0.1/yahooChatID
 */
export const yahooChatID: Property = 'http://xmlns.com/foaf/0.1/yahooChatID';

/**
 * MSN chat ID
 * 
 * An MSN chat ID
 *
 * http://xmlns.com/foaf/0.1/msnChatID
 */
export const msnChatID: Property = 'http://xmlns.com/foaf/0.1/msnChatID';

/**
 * name
 * 
 * A name for some thing.
 *
 * http://xmlns.com/foaf/0.1/name
 */
export const name: Property = 'http://xmlns.com/foaf/0.1/name';

/**
 * firstName
 * 
 * The first name of a person.
 *
 * http://xmlns.com/foaf/0.1/firstName
 */
export const firstName: Property = 'http://xmlns.com/foaf/0.1/firstName';

/**
 * lastName
 * 
 * The last name of a person.
 *
 * http://xmlns.com/foaf/0.1/lastName
 */
export const lastName: Property = 'http://xmlns.com/foaf/0.1/lastName';

/**
 * Given name
 * 
 * The given name of some person.
 *
 * http://xmlns.com/foaf/0.1/givenName
 */
export const givenName: Property = 'http://xmlns.com/foaf/0.1/givenName';

/**
 * Given name
 * 
 * The given name of some person.
 *
 * http://xmlns.com/foaf/0.1/givenname
 */
export const givenname: Property = 'http://xmlns.com/foaf/0.1/givenname';

/**
 * Surname
 * 
 * The surname of some person.
 *
 * http://xmlns.com/foaf/0.1/surname
 */
export const surname: Property = 'http://xmlns.com/foaf/0.1/surname';

/**
 * family_name
 * 
 * The family name of some person.
 *
 * http://xmlns.com/foaf/0.1/family_name
 */
export const family_name: Property = 'http://xmlns.com/foaf/0.1/family_name';

/**
 * familyName
 * 
 * The family name of some person.
 *
 * http://xmlns.com/foaf/0.1/familyName
 */
export const familyName: Property = 'http://xmlns.com/foaf/0.1/familyName';

/**
 * phone
 * 
 * A phone,  specified using fully qualified tel: URI scheme (refs: http://www.w3.org/Addressing/schemes.html#tel).
 *
 * http://xmlns.com/foaf/0.1/phone
 */
export const phone: Property = 'http://xmlns.com/foaf/0.1/phone';

/**
 * homepage
 * 
 * A homepage for some thing.
 *
 * http://xmlns.com/foaf/0.1/homepage
 */
export const homepage: Property = 'http://xmlns.com/foaf/0.1/homepage';

/**
 * weblog
 * 
 * A weblog of some thing (whether person, group, company etc.).
 *
 * http://xmlns.com/foaf/0.1/weblog
 */
export const weblog: Property = 'http://xmlns.com/foaf/0.1/weblog';

/**
 * openid
 * 
 * An OpenID for an Agent.
 *
 * http://xmlns.com/foaf/0.1/openid
 */
export const openid: Property = 'http://xmlns.com/foaf/0.1/openid';

/**
 * tipjar
 * 
 * A tipjar document for this agent, describing means for payment and reward.
 *
 * http://xmlns.com/foaf/0.1/tipjar
 */
export const tipjar: Property = 'http://xmlns.com/foaf/0.1/tipjar';

/**
 * plan
 * 
 * A .plan comment, in the tradition of finger and '.plan' files.
 *
 * http://xmlns.com/foaf/0.1/plan
 */
export const plan: Property = 'http://xmlns.com/foaf/0.1/plan';

/**
 * made
 * 
 * Something that was made by this agent.
 *
 * http://xmlns.com/foaf/0.1/made
 */
export const made: Property = 'http://xmlns.com/foaf/0.1/made';

/**
 * maker
 * 
 * An agent that 
 * made this thing.
 *
 * http://xmlns.com/foaf/0.1/maker
 */
export const maker: Property = 'http://xmlns.com/foaf/0.1/maker';

/**
 * image
 * 
 * An image that can be used to represent some thing (ie. those depictions which are particularly representative of something, eg. one's photo on a homepage).
 *
 * http://xmlns.com/foaf/0.1/img
 */
export const img: Property = 'http://xmlns.com/foaf/0.1/img';

/**
 * depiction
 * 
 * A depiction of some thing.
 *
 * http://xmlns.com/foaf/0.1/depiction
 */
export const depiction: Property = 'http://xmlns.com/foaf/0.1/depiction';

/**
 * depicts
 * 
 * A thing depicted in this representation.
 *
 * http://xmlns.com/foaf/0.1/depicts
 */
export const depicts: Property = 'http://xmlns.com/foaf/0.1/depicts';

/**
 * thumbnail
 * 
 * A derived thumbnail image.
 *
 * http://xmlns.com/foaf/0.1/thumbnail
 */
export const thumbnail: Property = 'http://xmlns.com/foaf/0.1/thumbnail';

/**
 * myersBriggs
 * 
 * A Myers Briggs (MBTI) personality classification.
 *
 * http://xmlns.com/foaf/0.1/myersBriggs
 */
export const myersBriggs: Property = 'http://xmlns.com/foaf/0.1/myersBriggs';

/**
 * workplace homepage
 * 
 * A workplace homepage of some person; the homepage of an organization they work for.
 *
 * http://xmlns.com/foaf/0.1/workplaceHomepage
 */
export const workplaceHomepage: Property = 'http://xmlns.com/foaf/0.1/workplaceHomepage';

/**
 * work info homepage
 * 
 * A work info homepage of some person; a page about their work for some organization.
 *
 * http://xmlns.com/foaf/0.1/workInfoHomepage
 */
export const workInfoHomepage: Property = 'http://xmlns.com/foaf/0.1/workInfoHomepage';

/**
 * schoolHomepage
 * 
 * A homepage of a school attended by the person.
 *
 * http://xmlns.com/foaf/0.1/schoolHomepage
 */
export const schoolHomepage: Property = 'http://xmlns.com/foaf/0.1/schoolHomepage';

/**
 * knows
 * 
 * A person known by this person (indicating some level of reciprocated interaction between the parties).
 *
 * http://xmlns.com/foaf/0.1/knows
 */
export const knows: Property = 'http://xmlns.com/foaf/0.1/knows';

/**
 * interest
 * 
 * A page about a topic of interest to this person.
 *
 * http://xmlns.com/foaf/0.1/interest
 */
export const interest: Property = 'http://xmlns.com/foaf/0.1/interest';

/**
 * topic_interest
 * 
 * A thing of interest to this person.
 *
 * http://xmlns.com/foaf/0.1/topic_interest
 */
export const topic_interest: Property = 'http://xmlns.com/foaf/0.1/topic_interest';

/**
 * publications
 * 
 * A link to the publications of this person.
 *
 * http://xmlns.com/foaf/0.1/publications
 */
export const publications: Property = 'http://xmlns.com/foaf/0.1/publications';

/**
 * current project
 * 
 * A current project this person works on.
 *
 * http://xmlns.com/foaf/0.1/currentProject
 */
export const currentProject: Property = 'http://xmlns.com/foaf/0.1/currentProject';

/**
 * past project
 * 
 * A project this person has previously worked on.
 *
 * http://xmlns.com/foaf/0.1/pastProject
 */
export const pastProject: Property = 'http://xmlns.com/foaf/0.1/pastProject';

/**
 * funded by
 * 
 * An organization funding a project or person.
 *
 * http://xmlns.com/foaf/0.1/fundedBy
 */
export const fundedBy: Property = 'http://xmlns.com/foaf/0.1/fundedBy';

/**
 * logo
 * 
 * A logo representing some thing.
 *
 * http://xmlns.com/foaf/0.1/logo
 */
export const logo: Property = 'http://xmlns.com/foaf/0.1/logo';

/**
 * topic
 * 
 * A topic of some page or document.
 *
 * http://xmlns.com/foaf/0.1/topic
 */
export const topic: Property = 'http://xmlns.com/foaf/0.1/topic';

/**
 * primary topic
 * 
 * The primary topic of some page or document.
 *
 * http://xmlns.com/foaf/0.1/primaryTopic
 */
export const primaryTopic: Property = 'http://xmlns.com/foaf/0.1/primaryTopic';

/**
 * focus
 * 
 * The underlying or 'focal' entity associated with some SKOS-described concept.
 *
 * http://xmlns.com/foaf/0.1/focus
 */
export const focus: Property = 'http://xmlns.com/foaf/0.1/focus';

/**
 * is primary topic of
 * 
 * A document that this thing is the primary topic of.
 *
 * http://xmlns.com/foaf/0.1/isPrimaryTopicOf
 */
export const isPrimaryTopicOf: Property = 'http://xmlns.com/foaf/0.1/isPrimaryTopicOf';

/**
 * page
 * 
 * A page or document about this thing.
 *
 * http://xmlns.com/foaf/0.1/page
 */
export const page: Property = 'http://xmlns.com/foaf/0.1/page';

/**
 * theme
 * 
 * A theme.
 *
 * http://xmlns.com/foaf/0.1/theme
 */
export const theme: Property = 'http://xmlns.com/foaf/0.1/theme';

/**
 * account
 * 
 * Indicates an account held by this agent.
 *
 * http://xmlns.com/foaf/0.1/account
 */
export const account: Property = 'http://xmlns.com/foaf/0.1/account';

/**
 * account
 * 
 * Indicates an account held by this agent.
 *
 * http://xmlns.com/foaf/0.1/holdsAccount
 */
export const holdsAccount: Property = 'http://xmlns.com/foaf/0.1/holdsAccount';

/**
 * account service homepage
 * 
 * Indicates a homepage of the service provide for this online account.
 *
 * http://xmlns.com/foaf/0.1/accountServiceHomepage
 */
export const accountServiceHomepage: Property = 'http://xmlns.com/foaf/0.1/accountServiceHomepage';

/**
 * account name
 * 
 * Indicates the name (identifier) associated with this online account.
 *
 * http://xmlns.com/foaf/0.1/accountName
 */
export const accountName: Property = 'http://xmlns.com/foaf/0.1/accountName';

/**
 * member
 * 
 * Indicates a member of a Group
 *
 * http://xmlns.com/foaf/0.1/member
 */
export const member: Property = 'http://xmlns.com/foaf/0.1/member';

/**
 * membershipClass
 * 
 * Indicates the class of individuals that are a member of a Group
 *
 * http://xmlns.com/foaf/0.1/membershipClass
 */
export const membershipClass: Property = 'http://xmlns.com/foaf/0.1/membershipClass';

/**
 * birthday
 * 
 * The birthday of this Agent, represented in mm-dd string form, eg. '12-31'.
 *
 * http://xmlns.com/foaf/0.1/birthday
 */
export const birthday: Property = 'http://xmlns.com/foaf/0.1/birthday';

/**
 * age
 * 
 * The age in years of some agent.
 *
 * http://xmlns.com/foaf/0.1/age
 */
export const age: Property = 'http://xmlns.com/foaf/0.1/age';

/**
 * status
 * 
 * A string expressing what the user is happy for the general public (normally) to know about their current activity.
 *
 * http://xmlns.com/foaf/0.1/status
 */
export const status: Property = 'http://xmlns.com/foaf/0.1/status';

/**
 * Label Property
 * 
 * A foaf:LabelProperty is any RDF property with texual values that serve as labels.
 *
 * http://xmlns.com/foaf/0.1/LabelProperty
 */
export const LabelProperty: Class = 'http://xmlns.com/foaf/0.1/LabelProperty';

/**
 * Person
 * 
 * A person.
 *
 * http://xmlns.com/foaf/0.1/Person
 */
export const Person: Class = 'http://xmlns.com/foaf/0.1/Person';

/**
 * Document
 * 
 * A document.
 *
 * http://xmlns.com/foaf/0.1/Document
 */
export const Document: Class = 'http://xmlns.com/foaf/0.1/Document';

/**
 * Organization
 * 
 * An organization.
 *
 * http://xmlns.com/foaf/0.1/Organization
 */
export const Organization: Class = 'http://xmlns.com/foaf/0.1/Organization';

/**
 * Group
 * 
 * A class of Agents.
 *
 * http://xmlns.com/foaf/0.1/Group
 */
export const Group: Class = 'http://xmlns.com/foaf/0.1/Group';

/**
 * Agent
 * 
 * An agent (eg. person, group, software or physical artifact).
 *
 * http://xmlns.com/foaf/0.1/Agent
 */
export const Agent: OwlClass = 'http://xmlns.com/foaf/0.1/Agent';

/**
 * Project
 * 
 * A project (a collective endeavour of some kind).
 *
 * http://xmlns.com/foaf/0.1/Project
 */
export const Project: Class = 'http://xmlns.com/foaf/0.1/Project';

/**
 * Image
 * 
 * An image.
 *
 * http://xmlns.com/foaf/0.1/Image
 */
export const Image: Class = 'http://xmlns.com/foaf/0.1/Image';

/**
 * PersonalProfileDocument
 * 
 * A personal profile RDF document.
 *
 * http://xmlns.com/foaf/0.1/PersonalProfileDocument
 */
export const PersonalProfileDocument: Class = 'http://xmlns.com/foaf/0.1/PersonalProfileDocument';

/**
 * Online Account
 * 
 * An online account.
 *
 * http://xmlns.com/foaf/0.1/OnlineAccount
 */
export const OnlineAccount: Class = 'http://xmlns.com/foaf/0.1/OnlineAccount';

/**
 * Online Gaming Account
 * 
 * An online gaming account.
 *
 * http://xmlns.com/foaf/0.1/OnlineGamingAccount
 */
export const OnlineGamingAccount: Class = 'http://xmlns.com/foaf/0.1/OnlineGamingAccount';

/**
 * Online E-commerce Account
 * 
 * An online e-commerce account.
 *
 * http://xmlns.com/foaf/0.1/OnlineEcommerceAccount
 */
export const OnlineEcommerceAccount: Class = 'http://xmlns.com/foaf/0.1/OnlineEcommerceAccount';

/**
 * Online Chat Account
 * 
 * An online chat account.
 *
 * http://xmlns.com/foaf/0.1/OnlineChatAccount
 */
export const OnlineChatAccount: Class = 'http://xmlns.com/foaf/0.1/OnlineChatAccount';
