// rdf-namespaces/dist/http.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * method
 * 
 * The method used in the request: 5.1.1
 *
 * http://www.w3.org/2007/ont/http#method
 */
export const method: Property = 'http://www.w3.org/2007/ont/http#method';

/**
 * request URI
 * 
 * The URI being requested: 5.1.2
 *
 * http://www.w3.org/2007/ont/http#requestURI
 */
export const requestURI: Property = 'http://www.w3.org/2007/ont/http#requestURI';

/**
 * version
 * 
 * The HTTP version of the request: 3.1,5.1,6.1
 *
 * http://www.w3.org/2007/ont/http#version
 */
export const version: Property = 'http://www.w3.org/2007/ont/http#version';

/**
 * status
 * 
 * The HTTP status of the response, 200 for OK, etc.
 *
 * http://www.w3.org/2007/ont/http#status
 */
export const status: Property = 'http://www.w3.org/2007/ont/http#status';

/**
 * HTTP Message
 * 
 * The class of all HTTP messages: 4.1
 *
 * http://www.w3.org/2007/ont/http#Message
 */
export const Message: Class = 'http://www.w3.org/2007/ont/http#Message';

/**
 * HTTP Request Message
 * 
 * This class of HTTP requests: 4.1
 *
 * http://www.w3.org/2007/ont/http#RequestMessage
 */
export const RequestMessage: Class = 'http://www.w3.org/2007/ont/http#RequestMessage';

/**
 * HTTP Response Message
 * 
 * This class of HTTP responses: 4.1
 *
 * http://www.w3.org/2007/ont/http#ResponseMessage
 */
export const ResponseMessage: Class = 'http://www.w3.org/2007/ont/http#ResponseMessage';
