// rdf-namespaces/dist/http-4a3ca0ad.js
/**
 * method
 *
 * The method used in the request: 5.1.1
 *
 * http://www.w3.org/2007/ont/http#method
 */
var method = 'http://www.w3.org/2007/ont/http#method';
/**
 * request URI
 *
 * The URI being requested: 5.1.2
 *
 * http://www.w3.org/2007/ont/http#requestURI
 */
var requestURI = 'http://www.w3.org/2007/ont/http#requestURI';
/**
 * version
 *
 * The HTTP version of the request: 3.1,5.1,6.1
 *
 * http://www.w3.org/2007/ont/http#version
 */
var version = 'http://www.w3.org/2007/ont/http#version';
/**
 * status
 *
 * The HTTP status of the response, 200 for OK, etc.
 *
 * http://www.w3.org/2007/ont/http#status
 */
var status = 'http://www.w3.org/2007/ont/http#status';
/**
 * HTTP Message
 *
 * The class of all HTTP messages: 4.1
 *
 * http://www.w3.org/2007/ont/http#Message
 */
var Message = 'http://www.w3.org/2007/ont/http#Message';
/**
 * HTTP Request Message
 *
 * This class of HTTP requests: 4.1
 *
 * http://www.w3.org/2007/ont/http#RequestMessage
 */
var RequestMessage = 'http://www.w3.org/2007/ont/http#RequestMessage';
/**
 * HTTP Response Message
 *
 * This class of HTTP responses: 4.1
 *
 * http://www.w3.org/2007/ont/http#ResponseMessage
 */
var ResponseMessage = 'http://www.w3.org/2007/ont/http#ResponseMessage';

var httpImport = /*#__PURE__*/Object.freeze({
	method: method,
	requestURI: requestURI,
	version: version,
	status: status,
	Message: Message,
	RequestMessage: RequestMessage,
	ResponseMessage: ResponseMessage
});

export { Message as M, RequestMessage as R, ResponseMessage as a, httpImport as h, method as m, requestURI as r, status as s, version as v };
