// rdf-namespaces/dist/hydra.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var hydra$1 = require('./hydra-633d3ceb.js');



exports.ApiDocumentation = hydra$1.ApiDocumentation;
exports.BasicRepresentation = hydra$1.BasicRepresentation;
exports.Class = hydra$1.Class;
exports.Collection = hydra$1.Collection;
exports.Error = hydra$1.Error;
exports.ExplicitRepresentation = hydra$1.ExplicitRepresentation;
exports.IriTemplate = hydra$1.IriTemplate;
exports.IriTemplateMapping = hydra$1.IriTemplateMapping;
exports.Link = hydra$1.Link;
exports.Operation = hydra$1.Operation;
exports.PartialCollectionView = hydra$1.PartialCollectionView;
exports.Resource = hydra$1.Resource;
exports.Rfc6570Template = hydra$1.Rfc6570Template;
exports.Status = hydra$1.Status;
exports.SupportedProperty = hydra$1.SupportedProperty;
exports.TemplatedLink = hydra$1.TemplatedLink;
exports.VariableRepresentation = hydra$1.VariableRepresentation;
exports.apiDocumentation = hydra$1.apiDocumentation;
exports.collection = hydra$1.collection;
exports.description = hydra$1.description;
exports.entrypoint = hydra$1.entrypoint;
exports.expects = hydra$1.expects;
exports.expectsHeader = hydra$1.expectsHeader;
exports.first = hydra$1.first;
exports.freetextQuery = hydra$1.freetextQuery;
exports.last = hydra$1.last;
exports.limit = hydra$1.limit;
exports.mapping = hydra$1.mapping;
exports.member = hydra$1.member;
exports.method = hydra$1.method;
exports.next = hydra$1.next;
exports.offset = hydra$1.offset;
exports.operation = hydra$1.operation;
exports.pageIndex = hydra$1.pageIndex;
exports.pageReference = hydra$1.pageReference;
exports.possibleStatus = hydra$1.possibleStatus;
exports.previous = hydra$1.previous;
exports.property = hydra$1.property;
exports.readable = hydra$1.readable;
exports.required = hydra$1.required;
exports.returns = hydra$1.returns;
exports.returnsHeader = hydra$1.returnsHeader;
exports.search = hydra$1.search;
exports.statusCode = hydra$1.statusCode;
exports.supportedClass = hydra$1.supportedClass;
exports.supportedOperation = hydra$1.supportedOperation;
exports.supportedProperty = hydra$1.supportedProperty;
exports.template = hydra$1.template;
exports.title = hydra$1.title;
exports.totalItems = hydra$1.totalItems;
exports.variable = hydra$1.variable;
exports.variableRepresentation = hydra$1.variableRepresentation;
exports.view = hydra$1.view;
exports.writeable = hydra$1.writeable;
