// rdf-namespaces/dist/hydra.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * property
 * 
 * A property
 *
 * http://www.w3.org/ns/hydra/core#property
 */
export const property: Property = 'http://www.w3.org/ns/hydra/core#property';

/**
 * required
 * 
 * True if the property is required, false otherwise.
 *
 * http://www.w3.org/ns/hydra/core#required
 */
export const required: Property = 'http://www.w3.org/ns/hydra/core#required';

/**
 * readable
 * 
 * True if the client can retrieve the property's value, false otherwise.
 *
 * http://www.w3.org/ns/hydra/core#readable
 */
export const readable: Property = 'http://www.w3.org/ns/hydra/core#readable';

/**
 * writeable
 * 
 * True if the client can change the property's value, false otherwise.
 *
 * http://www.w3.org/ns/hydra/core#writeable
 */
export const writeable: Property = 'http://www.w3.org/ns/hydra/core#writeable';

/**
 * method
 * 
 * The HTTP method.
 *
 * http://www.w3.org/ns/hydra/core#method
 */
export const method: Property = 'http://www.w3.org/ns/hydra/core#method';

/**
 * status code
 * 
 * The HTTP status code
 *
 * http://www.w3.org/ns/hydra/core#statusCode
 */
export const statusCode: Property = 'http://www.w3.org/ns/hydra/core#statusCode';

/**
 * title
 * 
 * A title, often used along with a description.
 *
 * http://www.w3.org/ns/hydra/core#title
 */
export const title: Property = 'http://www.w3.org/ns/hydra/core#title';

/**
 * description
 * 
 * A description.
 *
 * http://www.w3.org/ns/hydra/core#description
 */
export const description: Property = 'http://www.w3.org/ns/hydra/core#description';

/**
 * total items
 * 
 * The total number of items referenced by a collection.
 *
 * http://www.w3.org/ns/hydra/core#totalItems
 */
export const totalItems: Property = 'http://www.w3.org/ns/hydra/core#totalItems';

/**
 * freetext query
 * 
 * A property representing a freetext query.
 *
 * http://www.w3.org/ns/hydra/core#freetextQuery
 */
export const freetextQuery: Property = 'http://www.w3.org/ns/hydra/core#freetextQuery';

/**
 * template
 * 
 * A templated string with placeholders. The literal's datatype indicates the template syntax; if not specified, hydra:Rfc6570Template is assumed.
 *
 * http://www.w3.org/ns/hydra/core#template
 */
export const template: Property = 'http://www.w3.org/ns/hydra/core#template';

/**
 * variable representation
 * 
 * The representation format to use when expanding the IRI template.
 *
 * http://www.w3.org/ns/hydra/core#variableRepresentation
 */
export const variableRepresentation: Property = 'http://www.w3.org/ns/hydra/core#variableRepresentation';

/**
 * mapping
 * 
 * A variable-to-property mapping of the IRI template.
 *
 * http://www.w3.org/ns/hydra/core#mapping
 */
export const mapping: Property = 'http://www.w3.org/ns/hydra/core#mapping';

/**
 * variable
 * 
 * An IRI template variable
 *
 * http://www.w3.org/ns/hydra/core#variable
 */
export const variable: Property = 'http://www.w3.org/ns/hydra/core#variable';

/**
 * skip
 * 
 * Instructs to skip N elements of the set.
 *
 * http://www.w3.org/ns/hydra/core#offset
 */
export const offset: Property = 'http://www.w3.org/ns/hydra/core#offset';

/**
 * take
 * 
 * Instructs to limit set only to N elements.
 *
 * http://www.w3.org/ns/hydra/core#limit
 */
export const limit: Property = 'http://www.w3.org/ns/hydra/core#limit';

/**
 * page index
 * 
 * Instructs to provide a specific page of the collection at a given index.
 *
 * http://www.w3.org/ns/hydra/core#pageIndex
 */
export const pageIndex: Property = 'http://www.w3.org/ns/hydra/core#pageIndex';

/**
 * page reference
 * 
 * Instructs to provide a specific page reference of the collection.
 *
 * http://www.w3.org/ns/hydra/core#pageReference
 */
export const pageReference: Property = 'http://www.w3.org/ns/hydra/core#pageReference';

/**
 * returns header
 * 
 * Name of the header returned by the operation.
 *
 * http://www.w3.org/ns/hydra/core#returnsHeader
 */
export const returnsHeader: Property = 'http://www.w3.org/ns/hydra/core#returnsHeader';

/**
 * expects header
 * 
 * Specification of the header expected by the operation.
 *
 * http://www.w3.org/ns/hydra/core#expectsHeader
 */
export const expectsHeader: Property = 'http://www.w3.org/ns/hydra/core#expectsHeader';

/**
 * Hydra Class
 * 
 * The class of Hydra classes. Hydra classes and their instances are dereferenceable resources.
 *
 * http://www.w3.org/ns/hydra/core#Class
 */
export const Class: HydraResource = 'http://www.w3.org/ns/hydra/core#Class';

/**
 * RFC6570 IRI template
 * 
 * An IRI template as defined by RFC6570.
 *
 * http://www.w3.org/ns/hydra/core#Rfc6570Template
 */
export const Rfc6570Template: Datatype = 'http://www.w3.org/ns/hydra/core#Rfc6570Template';

/**
 * Hydra Resource
 * 
 * The class of dereferenceable resources.
 *
 * http://www.w3.org/ns/hydra/core#Resource
 */
export const Resource: HydraClass = 'http://www.w3.org/ns/hydra/core#Resource';

/**
 * Link
 * 
 * The class of properties representing links.
 *
 * http://www.w3.org/ns/hydra/core#Link
 */
export const Link: HydraClass = 'http://www.w3.org/ns/hydra/core#Link';

/**
 * ApiDocumentation
 * 
 * The Hydra API documentation class
 *
 * http://www.w3.org/ns/hydra/core#ApiDocumentation
 */
export const ApiDocumentation: HydraClass = 'http://www.w3.org/ns/hydra/core#ApiDocumentation';

/**
 * Operation
 * 
 * An operation.
 *
 * http://www.w3.org/ns/hydra/core#Operation
 */
export const Operation: HydraClass = 'http://www.w3.org/ns/hydra/core#Operation';

/**
 * Status code description
 * 
 * Additional information about a status code that might be returned.
 *
 * http://www.w3.org/ns/hydra/core#Status
 */
export const Status: HydraClass = 'http://www.w3.org/ns/hydra/core#Status';

/**
 * Supported Property
 * 
 * A property known to be supported by a Hydra class.
 *
 * http://www.w3.org/ns/hydra/core#SupportedProperty
 */
export const SupportedProperty: HydraClass = 'http://www.w3.org/ns/hydra/core#SupportedProperty';

/**
 * IriTemplateMapping
 * 
 * A mapping from an IRI template variable to a property.
 *
 * http://www.w3.org/ns/hydra/core#IriTemplateMapping
 */
export const IriTemplateMapping: HydraClass = 'http://www.w3.org/ns/hydra/core#IriTemplateMapping';

/**
 * Error
 * 
 * A runtime error, used to report information beyond the returned status code.
 *
 * http://www.w3.org/ns/hydra/core#Error
 */
export const Error: HydraClass = 'http://www.w3.org/ns/hydra/core#Error';

/**
 * Collection
 * 
 * A collection holding references to a number of related resources.
 *
 * http://www.w3.org/ns/hydra/core#Collection
 */
export const Collection: HydraClass = 'http://www.w3.org/ns/hydra/core#Collection';

/**
 * PartialCollectionView
 * 
 * A PartialCollectionView describes a partial view of a Collection. Multiple PartialCollectionViews can be connected with the the next/previous properties to allow a client to retrieve all members of the collection.
 *
 * http://www.w3.org/ns/hydra/core#PartialCollectionView
 */
export const PartialCollectionView: HydraClass = 'http://www.w3.org/ns/hydra/core#PartialCollectionView';

/**
 * IRI Template
 * 
 * The class of IRI templates.
 *
 * http://www.w3.org/ns/hydra/core#IriTemplate
 */
export const IriTemplate: HydraClass = 'http://www.w3.org/ns/hydra/core#IriTemplate';

/**
 * Templated Link
 * 
 * A templated link.
 *
 * http://www.w3.org/ns/hydra/core#TemplatedLink
 */
export const TemplatedLink: HydraClass = 'http://www.w3.org/ns/hydra/core#TemplatedLink';

/**
 * VariableRepresentation
 * 
 * A representation specifies how to serialize variable values into strings.
 *
 * http://www.w3.org/ns/hydra/core#VariableRepresentation
 */
export const VariableRepresentation: HydraClass = 'http://www.w3.org/ns/hydra/core#VariableRepresentation';

/**
 * apiDocumentation
 * 
 * A link to the API documentation
 *
 * http://www.w3.org/ns/hydra/core#apiDocumentation
 */
export const apiDocumentation: HydraLink = 'http://www.w3.org/ns/hydra/core#apiDocumentation';

/**
 * entrypoint
 * 
 * A link to main entry point of the Web API
 *
 * http://www.w3.org/ns/hydra/core#entrypoint
 */
export const entrypoint: HydraLink = 'http://www.w3.org/ns/hydra/core#entrypoint';

/**
 * supported classes
 * 
 * A class known to be supported by the Web API
 *
 * http://www.w3.org/ns/hydra/core#supportedClass
 */
export const supportedClass: HydraLink = 'http://www.w3.org/ns/hydra/core#supportedClass';

/**
 * possible status
 * 
 * A status that might be returned by the Web API (other statuses should be expected and properly handled as well)
 *
 * http://www.w3.org/ns/hydra/core#possibleStatus
 */
export const possibleStatus: HydraLink = 'http://www.w3.org/ns/hydra/core#possibleStatus';

/**
 * supported properties
 * 
 * The properties known to be supported by a Hydra class
 *
 * http://www.w3.org/ns/hydra/core#supportedProperty
 */
export const supportedProperty: HydraLink = 'http://www.w3.org/ns/hydra/core#supportedProperty';

/**
 * supported operation
 * 
 * An operation supported by instances of the specific Hydra class or the target of the Hydra link
 *
 * http://www.w3.org/ns/hydra/core#supportedOperation
 */
export const supportedOperation: HydraLink = 'http://www.w3.org/ns/hydra/core#supportedOperation';

/**
 * operation
 * 
 * An operation supported by the Hydra resource
 *
 * http://www.w3.org/ns/hydra/core#operation
 */
export const operation: HydraLink = 'http://www.w3.org/ns/hydra/core#operation';

/**
 * expects
 * 
 * The information expected by the Web API.
 *
 * http://www.w3.org/ns/hydra/core#expects
 */
export const expects: HydraLink = 'http://www.w3.org/ns/hydra/core#expects';

/**
 * returns
 * 
 * The information returned by the Web API on success
 *
 * http://www.w3.org/ns/hydra/core#returns
 */
export const returns: HydraLink = 'http://www.w3.org/ns/hydra/core#returns';

/**
 * collection
 * 
 * Collections somehow related to this resource.
 *
 * http://www.w3.org/ns/hydra/core#collection
 */
export const collection: HydraLink = 'http://www.w3.org/ns/hydra/core#collection';

/**
 * member
 * 
 * A member of the collection
 *
 * http://www.w3.org/ns/hydra/core#member
 */
export const member: HydraLink = 'http://www.w3.org/ns/hydra/core#member';

/**
 * view
 * 
 * A specific view of a resource.
 *
 * http://www.w3.org/ns/hydra/core#view
 */
export const view: HydraLink = 'http://www.w3.org/ns/hydra/core#view';

/**
 * first
 * 
 * The first resource of an interlinked set of resources.
 *
 * http://www.w3.org/ns/hydra/core#first
 */
export const first: HydraLink = 'http://www.w3.org/ns/hydra/core#first';

/**
 * last
 * 
 * The last resource of an interlinked set of resources.
 *
 * http://www.w3.org/ns/hydra/core#last
 */
export const last: HydraLink = 'http://www.w3.org/ns/hydra/core#last';

/**
 * next
 * 
 * The resource following the current instance in an interlinked set of resources.
 *
 * http://www.w3.org/ns/hydra/core#next
 */
export const next: HydraLink = 'http://www.w3.org/ns/hydra/core#next';

/**
 * previous
 * 
 * The resource preceding the current instance in an interlinked set of resources.
 *
 * http://www.w3.org/ns/hydra/core#previous
 */
export const previous: HydraLink = 'http://www.w3.org/ns/hydra/core#previous';

/**
 * search
 * 
 * A IRI template that can be used to query a collection.
 *
 * http://www.w3.org/ns/hydra/core#search
 */
export const search: HydraTemplatedLink = 'http://www.w3.org/ns/hydra/core#search';

/**
 * BasicRepresentation
 * 
 * A representation that serializes just the lexical form of a variable value, but omits language and type information.
 *
 * http://www.w3.org/ns/hydra/core#BasicRepresentation
 */
export const BasicRepresentation: HydraVariableRepresentation = 'http://www.w3.org/ns/hydra/core#BasicRepresentation';

/**
 * ExplicitRepresentation
 * 
 * A representation that serializes a variable value including its language and type information and thus differentiating between IRIs and literals.
 *
 * http://www.w3.org/ns/hydra/core#ExplicitRepresentation
 */
export const ExplicitRepresentation: HydraVariableRepresentation = 'http://www.w3.org/ns/hydra/core#ExplicitRepresentation';
