// rdf-namespaces/dist/hydra-e5a50a90.js
/**
 * property
 *
 * A property
 *
 * http://www.w3.org/ns/hydra/core#property
 */
var property = 'http://www.w3.org/ns/hydra/core#property';
/**
 * required
 *
 * True if the property is required, false otherwise.
 *
 * http://www.w3.org/ns/hydra/core#required
 */
var required = 'http://www.w3.org/ns/hydra/core#required';
/**
 * readable
 *
 * True if the client can retrieve the property's value, false otherwise.
 *
 * http://www.w3.org/ns/hydra/core#readable
 */
var readable = 'http://www.w3.org/ns/hydra/core#readable';
/**
 * writeable
 *
 * True if the client can change the property's value, false otherwise.
 *
 * http://www.w3.org/ns/hydra/core#writeable
 */
var writeable = 'http://www.w3.org/ns/hydra/core#writeable';
/**
 * method
 *
 * The HTTP method.
 *
 * http://www.w3.org/ns/hydra/core#method
 */
var method = 'http://www.w3.org/ns/hydra/core#method';
/**
 * status code
 *
 * The HTTP status code
 *
 * http://www.w3.org/ns/hydra/core#statusCode
 */
var statusCode = 'http://www.w3.org/ns/hydra/core#statusCode';
/**
 * title
 *
 * A title, often used along with a description.
 *
 * http://www.w3.org/ns/hydra/core#title
 */
var title = 'http://www.w3.org/ns/hydra/core#title';
/**
 * description
 *
 * A description.
 *
 * http://www.w3.org/ns/hydra/core#description
 */
var description = 'http://www.w3.org/ns/hydra/core#description';
/**
 * total items
 *
 * The total number of items referenced by a collection.
 *
 * http://www.w3.org/ns/hydra/core#totalItems
 */
var totalItems = 'http://www.w3.org/ns/hydra/core#totalItems';
/**
 * freetext query
 *
 * A property representing a freetext query.
 *
 * http://www.w3.org/ns/hydra/core#freetextQuery
 */
var freetextQuery = 'http://www.w3.org/ns/hydra/core#freetextQuery';
/**
 * template
 *
 * A templated string with placeholders. The literal's datatype indicates the template syntax; if not specified, hydra:Rfc6570Template is assumed.
 *
 * http://www.w3.org/ns/hydra/core#template
 */
var template = 'http://www.w3.org/ns/hydra/core#template';
/**
 * variable representation
 *
 * The representation format to use when expanding the IRI template.
 *
 * http://www.w3.org/ns/hydra/core#variableRepresentation
 */
var variableRepresentation = 'http://www.w3.org/ns/hydra/core#variableRepresentation';
/**
 * mapping
 *
 * A variable-to-property mapping of the IRI template.
 *
 * http://www.w3.org/ns/hydra/core#mapping
 */
var mapping = 'http://www.w3.org/ns/hydra/core#mapping';
/**
 * variable
 *
 * An IRI template variable
 *
 * http://www.w3.org/ns/hydra/core#variable
 */
var variable = 'http://www.w3.org/ns/hydra/core#variable';
/**
 * skip
 *
 * Instructs to skip N elements of the set.
 *
 * http://www.w3.org/ns/hydra/core#offset
 */
var offset = 'http://www.w3.org/ns/hydra/core#offset';
/**
 * take
 *
 * Instructs to limit set only to N elements.
 *
 * http://www.w3.org/ns/hydra/core#limit
 */
var limit = 'http://www.w3.org/ns/hydra/core#limit';
/**
 * page index
 *
 * Instructs to provide a specific page of the collection at a given index.
 *
 * http://www.w3.org/ns/hydra/core#pageIndex
 */
var pageIndex = 'http://www.w3.org/ns/hydra/core#pageIndex';
/**
 * page reference
 *
 * Instructs to provide a specific page reference of the collection.
 *
 * http://www.w3.org/ns/hydra/core#pageReference
 */
var pageReference = 'http://www.w3.org/ns/hydra/core#pageReference';
/**
 * returns header
 *
 * Name of the header returned by the operation.
 *
 * http://www.w3.org/ns/hydra/core#returnsHeader
 */
var returnsHeader = 'http://www.w3.org/ns/hydra/core#returnsHeader';
/**
 * expects header
 *
 * Specification of the header expected by the operation.
 *
 * http://www.w3.org/ns/hydra/core#expectsHeader
 */
var expectsHeader = 'http://www.w3.org/ns/hydra/core#expectsHeader';
/**
 * Hydra Class
 *
 * The class of Hydra classes. Hydra classes and their instances are dereferenceable resources.
 *
 * http://www.w3.org/ns/hydra/core#Class
 */
var Class = 'http://www.w3.org/ns/hydra/core#Class';
/**
 * RFC6570 IRI template
 *
 * An IRI template as defined by RFC6570.
 *
 * http://www.w3.org/ns/hydra/core#Rfc6570Template
 */
var Rfc6570Template = 'http://www.w3.org/ns/hydra/core#Rfc6570Template';
/**
 * Hydra Resource
 *
 * The class of dereferenceable resources.
 *
 * http://www.w3.org/ns/hydra/core#Resource
 */
var Resource = 'http://www.w3.org/ns/hydra/core#Resource';
/**
 * Link
 *
 * The class of properties representing links.
 *
 * http://www.w3.org/ns/hydra/core#Link
 */
var Link = 'http://www.w3.org/ns/hydra/core#Link';
/**
 * ApiDocumentation
 *
 * The Hydra API documentation class
 *
 * http://www.w3.org/ns/hydra/core#ApiDocumentation
 */
var ApiDocumentation = 'http://www.w3.org/ns/hydra/core#ApiDocumentation';
/**
 * Operation
 *
 * An operation.
 *
 * http://www.w3.org/ns/hydra/core#Operation
 */
var Operation = 'http://www.w3.org/ns/hydra/core#Operation';
/**
 * Status code description
 *
 * Additional information about a status code that might be returned.
 *
 * http://www.w3.org/ns/hydra/core#Status
 */
var Status = 'http://www.w3.org/ns/hydra/core#Status';
/**
 * Supported Property
 *
 * A property known to be supported by a Hydra class.
 *
 * http://www.w3.org/ns/hydra/core#SupportedProperty
 */
var SupportedProperty = 'http://www.w3.org/ns/hydra/core#SupportedProperty';
/**
 * IriTemplateMapping
 *
 * A mapping from an IRI template variable to a property.
 *
 * http://www.w3.org/ns/hydra/core#IriTemplateMapping
 */
var IriTemplateMapping = 'http://www.w3.org/ns/hydra/core#IriTemplateMapping';
/**
 * Error
 *
 * A runtime error, used to report information beyond the returned status code.
 *
 * http://www.w3.org/ns/hydra/core#Error
 */
var Error = 'http://www.w3.org/ns/hydra/core#Error';
/**
 * Collection
 *
 * A collection holding references to a number of related resources.
 *
 * http://www.w3.org/ns/hydra/core#Collection
 */
var Collection = 'http://www.w3.org/ns/hydra/core#Collection';
/**
 * PartialCollectionView
 *
 * A PartialCollectionView describes a partial view of a Collection. Multiple PartialCollectionViews can be connected with the the next/previous properties to allow a client to retrieve all members of the collection.
 *
 * http://www.w3.org/ns/hydra/core#PartialCollectionView
 */
var PartialCollectionView = 'http://www.w3.org/ns/hydra/core#PartialCollectionView';
/**
 * IRI Template
 *
 * The class of IRI templates.
 *
 * http://www.w3.org/ns/hydra/core#IriTemplate
 */
var IriTemplate = 'http://www.w3.org/ns/hydra/core#IriTemplate';
/**
 * Templated Link
 *
 * A templated link.
 *
 * http://www.w3.org/ns/hydra/core#TemplatedLink
 */
var TemplatedLink = 'http://www.w3.org/ns/hydra/core#TemplatedLink';
/**
 * VariableRepresentation
 *
 * A representation specifies how to serialize variable values into strings.
 *
 * http://www.w3.org/ns/hydra/core#VariableRepresentation
 */
var VariableRepresentation = 'http://www.w3.org/ns/hydra/core#VariableRepresentation';
/**
 * apiDocumentation
 *
 * A link to the API documentation
 *
 * http://www.w3.org/ns/hydra/core#apiDocumentation
 */
var apiDocumentation = 'http://www.w3.org/ns/hydra/core#apiDocumentation';
/**
 * entrypoint
 *
 * A link to main entry point of the Web API
 *
 * http://www.w3.org/ns/hydra/core#entrypoint
 */
var entrypoint = 'http://www.w3.org/ns/hydra/core#entrypoint';
/**
 * supported classes
 *
 * A class known to be supported by the Web API
 *
 * http://www.w3.org/ns/hydra/core#supportedClass
 */
var supportedClass = 'http://www.w3.org/ns/hydra/core#supportedClass';
/**
 * possible status
 *
 * A status that might be returned by the Web API (other statuses should be expected and properly handled as well)
 *
 * http://www.w3.org/ns/hydra/core#possibleStatus
 */
var possibleStatus = 'http://www.w3.org/ns/hydra/core#possibleStatus';
/**
 * supported properties
 *
 * The properties known to be supported by a Hydra class
 *
 * http://www.w3.org/ns/hydra/core#supportedProperty
 */
var supportedProperty = 'http://www.w3.org/ns/hydra/core#supportedProperty';
/**
 * supported operation
 *
 * An operation supported by instances of the specific Hydra class or the target of the Hydra link
 *
 * http://www.w3.org/ns/hydra/core#supportedOperation
 */
var supportedOperation = 'http://www.w3.org/ns/hydra/core#supportedOperation';
/**
 * operation
 *
 * An operation supported by the Hydra resource
 *
 * http://www.w3.org/ns/hydra/core#operation
 */
var operation = 'http://www.w3.org/ns/hydra/core#operation';
/**
 * expects
 *
 * The information expected by the Web API.
 *
 * http://www.w3.org/ns/hydra/core#expects
 */
var expects = 'http://www.w3.org/ns/hydra/core#expects';
/**
 * returns
 *
 * The information returned by the Web API on success
 *
 * http://www.w3.org/ns/hydra/core#returns
 */
var returns = 'http://www.w3.org/ns/hydra/core#returns';
/**
 * collection
 *
 * Collections somehow related to this resource.
 *
 * http://www.w3.org/ns/hydra/core#collection
 */
var collection = 'http://www.w3.org/ns/hydra/core#collection';
/**
 * member
 *
 * A member of the collection
 *
 * http://www.w3.org/ns/hydra/core#member
 */
var member = 'http://www.w3.org/ns/hydra/core#member';
/**
 * view
 *
 * A specific view of a resource.
 *
 * http://www.w3.org/ns/hydra/core#view
 */
var view = 'http://www.w3.org/ns/hydra/core#view';
/**
 * first
 *
 * The first resource of an interlinked set of resources.
 *
 * http://www.w3.org/ns/hydra/core#first
 */
var first = 'http://www.w3.org/ns/hydra/core#first';
/**
 * last
 *
 * The last resource of an interlinked set of resources.
 *
 * http://www.w3.org/ns/hydra/core#last
 */
var last = 'http://www.w3.org/ns/hydra/core#last';
/**
 * next
 *
 * The resource following the current instance in an interlinked set of resources.
 *
 * http://www.w3.org/ns/hydra/core#next
 */
var next = 'http://www.w3.org/ns/hydra/core#next';
/**
 * previous
 *
 * The resource preceding the current instance in an interlinked set of resources.
 *
 * http://www.w3.org/ns/hydra/core#previous
 */
var previous = 'http://www.w3.org/ns/hydra/core#previous';
/**
 * search
 *
 * A IRI template that can be used to query a collection.
 *
 * http://www.w3.org/ns/hydra/core#search
 */
var search = 'http://www.w3.org/ns/hydra/core#search';
/**
 * BasicRepresentation
 *
 * A representation that serializes just the lexical form of a variable value, but omits language and type information.
 *
 * http://www.w3.org/ns/hydra/core#BasicRepresentation
 */
var BasicRepresentation = 'http://www.w3.org/ns/hydra/core#BasicRepresentation';
/**
 * ExplicitRepresentation
 *
 * A representation that serializes a variable value including its language and type information and thus differentiating between IRIs and literals.
 *
 * http://www.w3.org/ns/hydra/core#ExplicitRepresentation
 */
var ExplicitRepresentation = 'http://www.w3.org/ns/hydra/core#ExplicitRepresentation';

var hydraImport = /*#__PURE__*/Object.freeze({
	property: property,
	required: required,
	readable: readable,
	writeable: writeable,
	method: method,
	statusCode: statusCode,
	title: title,
	description: description,
	totalItems: totalItems,
	freetextQuery: freetextQuery,
	template: template,
	variableRepresentation: variableRepresentation,
	mapping: mapping,
	variable: variable,
	offset: offset,
	limit: limit,
	pageIndex: pageIndex,
	pageReference: pageReference,
	returnsHeader: returnsHeader,
	expectsHeader: expectsHeader,
	Class: Class,
	Rfc6570Template: Rfc6570Template,
	Resource: Resource,
	Link: Link,
	ApiDocumentation: ApiDocumentation,
	Operation: Operation,
	Status: Status,
	SupportedProperty: SupportedProperty,
	IriTemplateMapping: IriTemplateMapping,
	Error: Error,
	Collection: Collection,
	PartialCollectionView: PartialCollectionView,
	IriTemplate: IriTemplate,
	TemplatedLink: TemplatedLink,
	VariableRepresentation: VariableRepresentation,
	apiDocumentation: apiDocumentation,
	entrypoint: entrypoint,
	supportedClass: supportedClass,
	possibleStatus: possibleStatus,
	supportedProperty: supportedProperty,
	supportedOperation: supportedOperation,
	operation: operation,
	expects: expects,
	returns: returns,
	collection: collection,
	member: member,
	view: view,
	first: first,
	last: last,
	next: next,
	previous: previous,
	search: search,
	BasicRepresentation: BasicRepresentation,
	ExplicitRepresentation: ExplicitRepresentation
});

export { ExplicitRepresentation as $, ApiDocumentation as A, supportedClass as B, Class as C, possibleStatus as D, Error as E, supportedProperty as F, supportedOperation as G, operation as H, IriTemplateMapping as I, expects as J, returns as K, Link as L, collection as M, member as N, Operation as O, PartialCollectionView as P, view as Q, Rfc6570Template as R, Status as S, TemplatedLink as T, first as U, VariableRepresentation as V, last as W, next as X, previous as Y, search as Z, BasicRepresentation as _, readable as a, hydraImport as a0, totalItems as b, template as c, description as d, mapping as e, freetextQuery as f, variable as g, pageIndex as h, pageReference as i, returnsHeader as j, expectsHeader as k, limit as l, method as m, Resource as n, offset as o, property as p, SupportedProperty as q, required as r, statusCode as s, title as t, Collection as u, variableRepresentation as v, writeable as w, IriTemplate as x, apiDocumentation as y, entrypoint as z };
