// rdf-namespaces/dist/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var acl$1 = require('./acl-561da199.js');
var arg$1 = require('./arg-0d9e7843.js');
var cal$1 = require('./cal-4a4b4cc5.js');
var contact$1 = require('./contact-c3e04d20.js');
var dc$1 = require('./dc-e8eba9f8.js');
var dct$1 = require('./dct-f8aebeba.js');
var foaf$1 = require('./foaf-acbc1a2a.js');
var http$1 = require('./http-318e3a8f.js');
var hydra$1 = require('./hydra-633d3ceb.js');
var ldp$1 = require('./ldp-2a468e08.js');
var link$1 = require('./link-acdabbb2.js');
var log$1 = require('./log-23e1654d.js');
var meeting$1 = require('./meeting-98b9dabc.js');
var owl$1 = require('./owl-ad7cf0f9.js');
var qu$1 = require('./qu-3b19bb57.js');
var trip$1 = require('./trip-f119c3cc.js');
var rdf$1 = require('./rdf-7d5902ad.js');
var rdfs$1 = require('./rdfs-4ceaa4df.js');
var sched$1 = require('./sched-bf21c384.js');
var schema$1 = require('./schema-a97ab12e.js');
var sioc$1 = require('./sioc-cc954753.js');
var skos$1 = require('./skos-0db70ed9.js');
var solid$1 = require('./solid-e10b9413.js');
var space$1 = require('./space-57bef06f.js');
var tab$1 = require('./tab-a6d0751c.js');
var tabont$1 = require('./tabont-ea3bced0.js');
var vcard$1 = require('./vcard-8871dca5.js');
var wf$1 = require('./wf-e7df4562.js');

var acl = acl$1.aclImport;
var arg = arg$1.argImport;
var cal = cal$1.calImport;
var contact = contact$1.contactImport;
var dc = dc$1.dcImport;
var dct = dct$1.dctImport;
var foaf = foaf$1.foafImport;
var http = http$1.httpImport;
var hydra = hydra$1.hydraImport;
var ldp = ldp$1.ldpImport;
var link = link$1.linkImport;
var log = log$1.logImport;
var meeting = meeting$1.meetingImport;
var owl = owl$1.owlImport;
var qu = qu$1.quImport;
var trip = trip$1.tripImport;
var rdf = rdf$1.rdfImport;
var rdfs = rdfs$1.rdfsImport;
var sched = sched$1.schedImport;
var schema = schema$1.schemaImport;
var sioc = sioc$1.siocImport;
var skos = skos$1.skosImport;
var solid = solid$1.solidImport;
var space = space$1.spaceImport;
var tab = tab$1.tabImport;
var tabont = tabont$1.tabontImport;
var vcard = vcard$1.vcardImport;
var wf = wf$1.wfImport;

exports.acl = acl;
exports.arg = arg;
exports.cal = cal;
exports.contact = contact;
exports.dc = dc;
exports.dct = dct;
exports.foaf = foaf;
exports.http = http;
exports.hydra = hydra;
exports.ldp = ldp;
exports.link = link;
exports.log = log;
exports.meeting = meeting;
exports.owl = owl;
exports.qu = qu;
exports.rdf = rdf;
exports.rdfs = rdfs;
exports.sched = sched;
exports.schema = schema;
exports.sioc = sioc;
exports.skos = skos;
exports.solid = solid;
exports.space = space;
exports.tab = tab;
exports.tabont = tabont;
exports.trip = trip;
exports.vcard = vcard;
exports.wf = wf;
