// rdf-namespaces/dist/index.ts
import * as aclImport from './acl'; export const acl = aclImport;
import * as argImport from './arg'; export const arg = argImport;
import * as calImport from './cal'; export const cal = calImport;
import * as contactImport from './contact'; export const contact = contactImport;
import * as dcImport from './dc'; export const dc = dcImport;
import * as dctImport from './dct'; export const dct = dctImport;
import * as foafImport from './foaf'; export const foaf = foafImport;
import * as httpImport from './http'; export const http = httpImport;
import * as hydraImport from './hydra'; export const hydra = hydraImport;
import * as ldpImport from './ldp'; export const ldp = ldpImport;
import * as linkImport from './link'; export const link = linkImport;
import * as logImport from './log'; export const log = logImport;
import * as meetingImport from './meeting'; export const meeting = meetingImport;
import * as owlImport from './owl'; export const owl = owlImport;
import * as quImport from './qu'; export const qu = quImport;
import * as tripImport from './trip'; export const trip = tripImport;
import * as rdfImport from './rdf'; export const rdf = rdfImport;
import * as rdfsImport from './rdfs'; export const rdfs = rdfsImport;
import * as schedImport from './sched'; export const sched = schedImport;
import * as schemaImport from './schema'; export const schema = schemaImport;
import * as siocImport from './sioc'; export const sioc = siocImport;
import * as skosImport from './skos'; export const skos = skosImport;
import * as solidImport from './solid'; export const solid = solidImport;
import * as spaceImport from './space'; export const space = spaceImport;
import * as tabImport from './tab'; export const tab = tabImport;
import * as tabontImport from './tabont'; export const tabont = tabontImport;
import * as vcardImport from './vcard'; export const vcard = vcardImport;
import * as wfImport from './wf'; export const wf = wfImport;
