// rdf-namespaces/dist/ldp-2a468e08.js
'use strict';

/**
 * hasMemberRelation
 *
 * Indicates which predicate is used in membership triples, and that the membership triple pattern is < membership-constant-URI , object-of-hasMemberRelation, member-URI >.
 *
 * http://www.w3.org/ns/ldp#hasMemberRelation
 */
var hasMemberRelation = 'http://www.w3.org/ns/ldp#hasMemberRelation';
/**
 * isMemmberOfRelation
 *
 * Indicates which predicate is used in membership triples, and that the membership triple pattern is < member-URI , object-of-isMemberOfRelation, membership-constant-URI >.
 *
 * http://www.w3.org/ns/ldp#isMemberOfRelation
 */
var isMemberOfRelation = 'http://www.w3.org/ns/ldp#isMemberOfRelation';
/**
 * membershipResource
 *
 * Indicates the membership-constant-URI in a membership triple.  Depending upon the membership triple pattern a container uses, as indicated by the presence of ldp:hasMemberRelation or ldp:isMemberOfRelation, the membership-constant-URI might occupy either the subject or object position in membership triples.
 *
 * http://www.w3.org/ns/ldp#membershipResource
 */
var membershipResource = 'http://www.w3.org/ns/ldp#membershipResource';
/**
 * insertedContentRelation
 *
 * Indicates which triple in a creation request should be used as the member-URI value in the membership triple added when the creation request is successful.
 *
 * http://www.w3.org/ns/ldp#insertedContentRelation
 */
var insertedContentRelation = 'http://www.w3.org/ns/ldp#insertedContentRelation';
/**
 * member
 *
 * LDP servers should use this predicate as the membership predicate if there is no obvious predicate from an application vocabulary to use.
 *
 * http://www.w3.org/ns/ldp#member
 */
var member = 'http://www.w3.org/ns/ldp#member';
/**
 * contains
 *
 * Links a container with resources created through the container.
 *
 * http://www.w3.org/ns/ldp#contains
 */
var contains = 'http://www.w3.org/ns/ldp#contains';
/**
 * constrainedBy
 *
 * Links a resource with constraints that the server requires requests like creation and update to conform to.
 *
 * http://www.w3.org/ns/ldp#constrainedBy
 */
var constrainedBy = 'http://www.w3.org/ns/ldp#constrainedBy';
/**
 * pageSortCriteria
 *
 * Link to the list of sorting criteria used by the server in a representation.  Typically used on Link response headers as an extension link relation URI in the rel= parameter.
 *
 * http://www.w3.org/ns/ldp#pageSortCriteria
 */
var pageSortCriteria = 'http://www.w3.org/ns/ldp#pageSortCriteria';
/**
 * pageSortPredicate
 *
 * Predicate used to specify the order of the members across a page sequence's in-sequence page resources; it asserts nothing about the order of members in the representation of a single page.
 *
 * http://www.w3.org/ns/ldp#pageSortPredicate
 */
var pageSortPredicate = 'http://www.w3.org/ns/ldp#pageSortPredicate';
/**
 * pageSortOrder
 *
 * The ascending/descending/etc order used to order the members across pages in a page sequence.
 *
 * http://www.w3.org/ns/ldp#pageSortOrder
 */
var pageSortOrder = 'http://www.w3.org/ns/ldp#pageSortOrder';
/**
 * pageSortCollation
 *
 * The collation used to order the members across pages in a page sequence when comparing strings.
 *
 * http://www.w3.org/ns/ldp#pageSortCollation
 */
var pageSortCollation = 'http://www.w3.org/ns/ldp#pageSortCollation';
/**
 * Page
 *
 * Link to a page sequence resource, as defined by LDP Paging.  Typically used to communicate the sorting criteria used to allocate LDPC members to pages.
 *
 * http://www.w3.org/ns/ldp#pageSequence
 */
var pageSequence = 'http://www.w3.org/ns/ldp#pageSequence';
/**
 * inbox
 *
 * Links a resource to a container where notifications for the resource can be created and discovered.
 *
 * http://www.w3.org/ns/ldp#inbox
 */
var inbox = 'http://www.w3.org/ns/ldp#inbox';
/**
 * Resource
 *
 * A HTTP-addressable resource whose lifecycle is managed by a LDP server.
 *
 * http://www.w3.org/ns/ldp#Resource
 */
var Resource = 'http://www.w3.org/ns/ldp#Resource';
/**
 * RDFSource
 *
 * A Linked Data Platform Resource (LDPR) whose state is represented as RDF.
 *
 * http://www.w3.org/ns/ldp#RDFSource
 */
var RDFSource = 'http://www.w3.org/ns/ldp#RDFSource';
/**
 * NonRDFSource
 *
 * A Linked Data Platform Resource (LDPR) whose state is NOT represented as RDF.
 *
 * http://www.w3.org/ns/ldp#NonRDFSource
 */
var NonRDFSource = 'http://www.w3.org/ns/ldp#NonRDFSource';
/**
 * Container
 *
 * A Linked Data Platform RDF Source (LDP-RS) that also conforms to additional patterns and conventions for managing membership. Readers should refer to the specification defining this ontology for the list of behaviors associated with it.
 *
 * http://www.w3.org/ns/ldp#Container
 */
var Container = 'http://www.w3.org/ns/ldp#Container';
/**
 * BasicContainer
 *
 * An LDPC that uses a predefined predicate to simply link to its contained resources.
 *
 * http://www.w3.org/ns/ldp#BasicContainer
 */
var BasicContainer = 'http://www.w3.org/ns/ldp#BasicContainer';
/**
 * DirectContainer
 *
 * An LDPC that is similar to a LDP-DC but it allows an indirection with the ability to list as member a resource, such as a URI representing a real-world object, that is different from the resource that is created.
 *
 * http://www.w3.org/ns/ldp#DirectContainer
 */
var DirectContainer = 'http://www.w3.org/ns/ldp#DirectContainer';
/**
 * IndirectContainer
 *
 * An LDPC that has the flexibility of choosing what form the membership triples take.
 *
 * http://www.w3.org/ns/ldp#IndirectContainer
 */
var IndirectContainer = 'http://www.w3.org/ns/ldp#IndirectContainer';
/**
 * PageSortCriterion
 *
 * Element in the list of sorting criteria used by the server to assign container members to pages.
 *
 * http://www.w3.org/ns/ldp#PageSortCriterion
 */
var PageSortCriterion = 'http://www.w3.org/ns/ldp#PageSortCriterion';
/**
 * Page
 *
 * URI signifying that the resource is an in-sequence page resource, as defined by LDP Paging.  Typically used on Link rel='type' response headers.
 *
 * http://www.w3.org/ns/ldp#Page
 */
var Page = 'http://www.w3.org/ns/ldp#Page';

var ldpImport = /*#__PURE__*/Object.freeze({
	hasMemberRelation: hasMemberRelation,
	isMemberOfRelation: isMemberOfRelation,
	membershipResource: membershipResource,
	insertedContentRelation: insertedContentRelation,
	member: member,
	contains: contains,
	constrainedBy: constrainedBy,
	pageSortCriteria: pageSortCriteria,
	pageSortPredicate: pageSortPredicate,
	pageSortOrder: pageSortOrder,
	pageSortCollation: pageSortCollation,
	pageSequence: pageSequence,
	inbox: inbox,
	Resource: Resource,
	RDFSource: RDFSource,
	NonRDFSource: NonRDFSource,
	Container: Container,
	BasicContainer: BasicContainer,
	DirectContainer: DirectContainer,
	IndirectContainer: IndirectContainer,
	PageSortCriterion: PageSortCriterion,
	Page: Page
});

exports.BasicContainer = BasicContainer;
exports.Container = Container;
exports.DirectContainer = DirectContainer;
exports.IndirectContainer = IndirectContainer;
exports.NonRDFSource = NonRDFSource;
exports.Page = Page;
exports.PageSortCriterion = PageSortCriterion;
exports.RDFSource = RDFSource;
exports.Resource = Resource;
exports.constrainedBy = constrainedBy;
exports.contains = contains;
exports.hasMemberRelation = hasMemberRelation;
exports.inbox = inbox;
exports.insertedContentRelation = insertedContentRelation;
exports.isMemberOfRelation = isMemberOfRelation;
exports.ldpImport = ldpImport;
exports.member = member;
exports.membershipResource = membershipResource;
exports.pageSequence = pageSequence;
exports.pageSortCollation = pageSortCollation;
exports.pageSortCriteria = pageSortCriteria;
exports.pageSortOrder = pageSortOrder;
exports.pageSortPredicate = pageSortPredicate;
