// rdf-namespaces/dist/log.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var log$1 = require('./log-23e1654d.js');



exports.Chaff = log$1.Chaff;
exports.Formula = log$1.Formula;
exports.List = log$1.List;
exports.N3Document = log$1.N3Document;
exports.Other = log$1.Other;
exports.String = log$1.String;
exports.Truth = log$1.Truth;
exports.conclusion = log$1.conclusion;
exports.conjunction = log$1.conjunction;
exports.content = log$1.content;
exports.definitiveDocument = log$1.definitiveDocument;
exports.definitiveService = log$1.definitiveService;
exports.dtlit = log$1.dtlit;
exports.equalTo = log$1.equalTo;
exports.implies = log$1.implies;
exports.includes = log$1.includes;
exports.n3String = log$1.n3String;
exports.notEqualTo = log$1.notEqualTo;
exports.notIncludes = log$1.notIncludes;
exports.outputString = log$1.outputString;
exports.parsedAsN3 = log$1.parsedAsN3;
exports.racine = log$1.racine;
exports.rawType = log$1.rawType;
exports.rawUri = log$1.rawUri;
exports.semantics = log$1.semantics;
exports.semanticsOrError = log$1.semanticsOrError;
exports.uri = log$1.uri;
