// rdf-namespaces/dist/meeting.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * action items
 * 
 * 
 *
 * http://www.w3.org/ns/pim/meeting#actionItems
 */
export const actionItems: Property = 'http://www.w3.org/ns/pim/meeting#actionItems';

/**
 * agenda
 * 
 * 
 *
 * http://www.w3.org/ns/pim/meeting#agenda
 */
export const agenda: Property = 'http://www.w3.org/ns/pim/meeting#agenda';

/**
 * attendees
 * 
 * 
 *
 * http://www.w3.org/ns/pim/meeting#attendeeGroup
 */
export const attendeeGroup: Property = 'http://www.w3.org/ns/pim/meeting#attendeeGroup';

/**
 * chat
 * 
 * A live text-based chat object or chat document.
 *
 * http://www.w3.org/ns/pim/meeting#chat
 */
export const chat: Property = 'http://www.w3.org/ns/pim/meeting#chat';

/**
 * next meeting
 * 
 * 
 *
 * http://www.w3.org/ns/pim/meeting#nextMeeting
 */
export const nextMeeting: Property = 'http://www.w3.org/ns/pim/meeting#nextMeeting';

/**
 * previous meeting
 * 
 * 
 *
 * http://www.w3.org/ns/pim/meeting#previousMeeting
 */
export const previousMeeting: Property = 'http://www.w3.org/ns/pim/meeting#previousMeeting';

/**
 * notes
 * 
 * A shared notepad to capture the course and result of the meeting
 *
 * http://www.w3.org/ns/pim/meeting#sharedNotes
 */
export const sharedNotes: Property = 'http://www.w3.org/ns/pim/meeting#sharedNotes';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/pim/meeting#toolList
 */
export const toolList: Property = 'http://www.w3.org/ns/pim/meeting#toolList';

/**
 * trip
 * 
 * 
 *
 * http://www.w3.org/ns/pim/meeting#trip
 */
export const trip: Property = 'http://www.w3.org/ns/pim/meeting#trip';

/**
 * video call
 * 
 * For a meeting, a web page (eg on appear.in, or wherever) for  a live video chat for the meeting.
 *   The URI should be as accurate as possible ideally without requiring meeting numbers etc to be entered.
 *
 * http://www.w3.org/ns/pim/meeting#videoCallPage
 */
export const videoCallPage: Property = 'http://www.w3.org/ns/pim/meeting#videoCallPage';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/pim/meeting#Meeting
 */
export const Meeting: Class = 'http://www.w3.org/ns/pim/meeting#Meeting';

/**
 * video call page
 * 
 * A web page (eg on appear.in, or wherever) for  a live video chat.
 *     The URI should be as accurate as possible ideally without requiring meeting numbers, room numbers, etc to be entered.
 *
 * http://www.w3.org/ns/pim/meeting#VideoCallPage
 */
export const VideoCallPage: Class = 'http://www.w3.org/ns/pim/meeting#VideoCallPage';
