// rdf-namespaces/dist/meeting-1395c9ba.js
/**
 * action items
 *
 *
 *
 * http://www.w3.org/ns/pim/meeting#actionItems
 */
var actionItems = 'http://www.w3.org/ns/pim/meeting#actionItems';
/**
 * agenda
 *
 *
 *
 * http://www.w3.org/ns/pim/meeting#agenda
 */
var agenda = 'http://www.w3.org/ns/pim/meeting#agenda';
/**
 * attendees
 *
 *
 *
 * http://www.w3.org/ns/pim/meeting#attendeeGroup
 */
var attendeeGroup = 'http://www.w3.org/ns/pim/meeting#attendeeGroup';
/**
 * chat
 *
 * A live text-based chat object or chat document.
 *
 * http://www.w3.org/ns/pim/meeting#chat
 */
var chat = 'http://www.w3.org/ns/pim/meeting#chat';
/**
 * next meeting
 *
 *
 *
 * http://www.w3.org/ns/pim/meeting#nextMeeting
 */
var nextMeeting = 'http://www.w3.org/ns/pim/meeting#nextMeeting';
/**
 * previous meeting
 *
 *
 *
 * http://www.w3.org/ns/pim/meeting#previousMeeting
 */
var previousMeeting = 'http://www.w3.org/ns/pim/meeting#previousMeeting';
/**
 * notes
 *
 * A shared notepad to capture the course and result of the meeting
 *
 * http://www.w3.org/ns/pim/meeting#sharedNotes
 */
var sharedNotes = 'http://www.w3.org/ns/pim/meeting#sharedNotes';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/pim/meeting#toolList
 */
var toolList = 'http://www.w3.org/ns/pim/meeting#toolList';
/**
 * trip
 *
 *
 *
 * http://www.w3.org/ns/pim/meeting#trip
 */
var trip = 'http://www.w3.org/ns/pim/meeting#trip';
/**
 * video call
 *
 * For a meeting, a web page (eg on appear.in, or wherever) for  a live video chat for the meeting.
 *   The URI should be as accurate as possible ideally without requiring meeting numbers etc to be entered.
 *
 * http://www.w3.org/ns/pim/meeting#videoCallPage
 */
var videoCallPage = 'http://www.w3.org/ns/pim/meeting#videoCallPage';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/pim/meeting#Meeting
 */
var Meeting = 'http://www.w3.org/ns/pim/meeting#Meeting';
/**
 * video call page
 *
 * A web page (eg on appear.in, or wherever) for  a live video chat.
 *     The URI should be as accurate as possible ideally without requiring meeting numbers, room numbers, etc to be entered.
 *
 * http://www.w3.org/ns/pim/meeting#VideoCallPage
 */
var VideoCallPage = 'http://www.w3.org/ns/pim/meeting#VideoCallPage';

var meetingImport = /*#__PURE__*/Object.freeze({
	actionItems: actionItems,
	agenda: agenda,
	attendeeGroup: attendeeGroup,
	chat: chat,
	nextMeeting: nextMeeting,
	previousMeeting: previousMeeting,
	sharedNotes: sharedNotes,
	toolList: toolList,
	trip: trip,
	videoCallPage: videoCallPage,
	Meeting: Meeting,
	VideoCallPage: VideoCallPage
});

export { Meeting as M, VideoCallPage as V, actionItems as a, agenda as b, attendeeGroup as c, chat as d, trip as e, meetingImport as m, nextMeeting as n, previousMeeting as p, sharedNotes as s, toolList as t, videoCallPage as v };
