// rdf-namespaces/dist/owl.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var owl$1 = require('./owl-ad7cf0f9.js');



exports.AllDifferent = owl$1.AllDifferent;
exports.AllDisjointClasses = owl$1.AllDisjointClasses;
exports.AllDisjointProperties = owl$1.AllDisjointProperties;
exports.Annotation = owl$1.Annotation;
exports.AnnotationProperty = owl$1.AnnotationProperty;
exports.AsymmetricProperty = owl$1.AsymmetricProperty;
exports.Axiom = owl$1.Axiom;
exports.Class = owl$1.Class;
exports.DataRange = owl$1.DataRange;
exports.DatatypeProperty = owl$1.DatatypeProperty;
exports.DeprecatedClass = owl$1.DeprecatedClass;
exports.DeprecatedProperty = owl$1.DeprecatedProperty;
exports.FunctionalProperty = owl$1.FunctionalProperty;
exports.InverseFunctionalProperty = owl$1.InverseFunctionalProperty;
exports.IrreflexiveProperty = owl$1.IrreflexiveProperty;
exports.NamedIndividual = owl$1.NamedIndividual;
exports.NegativePropertyAssertion = owl$1.NegativePropertyAssertion;
exports.Nothing = owl$1.Nothing;
exports.ObjectProperty = owl$1.ObjectProperty;
exports.Ontology = owl$1.Ontology;
exports.OntologyProperty = owl$1.OntologyProperty;
exports.ReflexiveProperty = owl$1.ReflexiveProperty;
exports.Restriction = owl$1.Restriction;
exports.SymmetricProperty = owl$1.SymmetricProperty;
exports.Thing = owl$1.Thing;
exports.TransitiveProperty = owl$1.TransitiveProperty;
exports.allValuesFrom = owl$1.allValuesFrom;
exports.annotatedProperty = owl$1.annotatedProperty;
exports.annotatedSource = owl$1.annotatedSource;
exports.annotatedTarget = owl$1.annotatedTarget;
exports.assertionProperty = owl$1.assertionProperty;
exports.bottomDataProperty = owl$1.bottomDataProperty;
exports.bottomObjectProperty = owl$1.bottomObjectProperty;
exports.cardinality = owl$1.cardinality;
exports.complementOf = owl$1.complementOf;
exports.datatypeComplementOf = owl$1.datatypeComplementOf;
exports.differentFrom = owl$1.differentFrom;
exports.disjointUnionOf = owl$1.disjointUnionOf;
exports.disjointWith = owl$1.disjointWith;
exports.distinctMembers = owl$1.distinctMembers;
exports.equivalentClass = owl$1.equivalentClass;
exports.equivalentProperty = owl$1.equivalentProperty;
exports.hasKey = owl$1.hasKey;
exports.hasSelf = owl$1.hasSelf;
exports.hasValue = owl$1.hasValue;
exports.intersectionOf = owl$1.intersectionOf;
exports.inverseOf = owl$1.inverseOf;
exports.maxCardinality = owl$1.maxCardinality;
exports.maxQualifiedCardinality = owl$1.maxQualifiedCardinality;
exports.members = owl$1.members;
exports.minCardinality = owl$1.minCardinality;
exports.minQualifiedCardinality = owl$1.minQualifiedCardinality;
exports.onClass = owl$1.onClass;
exports.onDataRange = owl$1.onDataRange;
exports.onDatatype = owl$1.onDatatype;
exports.onProperties = owl$1.onProperties;
exports.onProperty = owl$1.onProperty;
exports.oneOf = owl$1.oneOf;
exports.propertyChainAxiom = owl$1.propertyChainAxiom;
exports.propertyDisjointWith = owl$1.propertyDisjointWith;
exports.qualifiedCardinality = owl$1.qualifiedCardinality;
exports.sameAs = owl$1.sameAs;
exports.someValuesFrom = owl$1.someValuesFrom;
exports.sourceIndividual = owl$1.sourceIndividual;
exports.targetIndividual = owl$1.targetIndividual;
exports.targetValue = owl$1.targetValue;
exports.topDataProperty = owl$1.topDataProperty;
exports.topObjectProperty = owl$1.topObjectProperty;
exports.unionOf = owl$1.unionOf;
exports.withRestrictions = owl$1.withRestrictions;
