// rdf-namespaces/dist/qu.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * category
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/qif#category
 */
export const category: Property = 'http://www.w3.org/2000/10/swap/pim/qif#category';

/**
 * amount
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/qif#amount
 */
export const amount: Property = 'http://www.w3.org/2000/10/swap/pim/qif#amount';

/**
 * date
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/qif#date
 */
export const date: Property = 'http://www.w3.org/2000/10/swap/pim/qif#date';

/**
 * payee
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/qif#payee
 */
export const payee: Property = 'http://www.w3.org/2000/10/swap/pim/qif#payee';

/**
 * account
 * 
 * The account to which money flows when the amount is positive.
 *
 * http://www.w3.org/2000/10/swap/pim/qif#toAccount
 */
export const toAccount: Property = 'http://www.w3.org/2000/10/swap/pim/qif#toAccount';

/**
 * statement
 * 
 * The document which is the source of data for this transaction.
 *
 * http://www.w3.org/2000/10/swap/pim/qif#accordingTo
 */
export const accordingTo: Property = 'http://www.w3.org/2000/10/swap/pim/qif#accordingTo';

/**
 * part
 * 
 * A transaction representing a part of this transaction.
 *     The sum of the amounts of the parts must equal the amount of this transaction.
 *
 * http://www.w3.org/2000/10/swap/pim/qif#splitPart
 */
export const splitPart: Property = 'http://www.w3.org/2000/10/swap/pim/qif#splitPart';

/**
 * part
 * 
 * A reverse transaction, putting this transaction back in the Split account, for redistribution as parts.
 *     The amount of the reverse transaction must be the negation of this transaction.
 *
 * http://www.w3.org/2000/10/swap/pim/qif#splitReverse
 */
export const splitReverse: Property = 'http://www.w3.org/2000/10/swap/pim/qif#splitReverse';

/**
 * transaction
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/qif#Transaction
 */
export const Transaction: Class = 'http://www.w3.org/2000/10/swap/pim/qif#Transaction';

/**
 * classified
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/qif#Classified
 */
export const Classified: Class = 'http://www.w3.org/2000/10/swap/pim/qif#Classified';

/**
 * unclassified
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/qif#Unclassified
 */
export const Unclassified: Class = 'http://www.w3.org/2000/10/swap/pim/qif#Unclassified';

/**
 * unclassified income
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/qif#UnclassifiedIncome
 */
export const UnclassifiedIncome: Class = 'http://www.w3.org/2000/10/swap/pim/qif#UnclassifiedIncome';

/**
 * unclassified outgoing
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/qif#UnclassifiedOutgoing
 */
export const UnclassifiedOutgoing: Class = 'http://www.w3.org/2000/10/swap/pim/qif#UnclassifiedOutgoing';

/**
 * account
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/qif#Account
 */
export const Account: Class = 'http://www.w3.org/2000/10/swap/pim/qif#Account';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/qif#Bank
 */
export const Bank: Class = 'http://www.w3.org/2000/10/swap/pim/qif#Bank';

/**
 * category
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/qif#Cat
 */
export const Cat: Class = 'http://www.w3.org/2000/10/swap/pim/qif#Cat';

/**
 * supporting document
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/qif#SupportingDocument
 */
export const SupportingDocument: Class = 'http://www.w3.org/2000/10/swap/pim/qif#SupportingDocument';

/**
 * receipt
 * 
 * 
 *
 * http://www.w3.org/2000/10/swap/pim/qif#Receipt
 */
export const Receipt: Class = 'http://www.w3.org/2000/10/swap/pim/qif#Receipt';

/**
 * split transaction
 * 
 * A transaction which has to be split into parts for classification.
 *
 * http://www.w3.org/2000/10/swap/pim/qif#SplitTransaction
 */
export const SplitTransaction: Class = 'http://www.w3.org/2000/10/swap/pim/qif#SplitTransaction';
