// rdf-namespaces/dist/rdf.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * type
 * 
 * The subject is an instance of a class.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#type
 */
export const type: Property = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#type';

/**
 * subject
 * 
 * The subject of the subject RDF statement.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#subject
 */
export const subject: Property = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#subject';

/**
 * predicate
 * 
 * The predicate of the subject RDF statement.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#predicate
 */
export const predicate: Property = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#predicate';

/**
 * object
 * 
 * The object of the subject RDF statement.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#object
 */
export const object: Property = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#object';

/**
 * value
 * 
 * Idiomatic property used for structured values.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#value
 */
export const value: Property = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#value';

/**
 * first
 * 
 * The first item in the subject RDF list.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#first
 */
export const first: Property = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#first';

/**
 * rest
 * 
 * The rest of the subject RDF list after the first item.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#rest
 */
export const rest: Property = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#rest';

/**
 * language
 * 
 * The language component of a CompoundLiteral.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#language
 */
export const language: Property = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#language';

/**
 * direction
 * 
 * The base direction component of a CompoundLiteral.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#direction
 */
export const direction: Property = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#direction';

/**
 * Property
 * 
 * The class of RDF properties.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#Property
 */
export const Property: Class = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#Property';

/**
 * Statement
 * 
 * The class of RDF statements.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#Statement
 */
export const Statement: Class = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#Statement';

/**
 * Bag
 * 
 * The class of unordered containers.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#Bag
 */
export const Bag: Class = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#Bag';

/**
 * Seq
 * 
 * The class of ordered containers.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#Seq
 */
export const Seq: Class = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#Seq';

/**
 * Alt
 * 
 * The class of containers of alternatives.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#Alt
 */
export const Alt: Class = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#Alt';

/**
 * List
 * 
 * The class of RDF Lists.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#List
 */
export const List: Class = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#List';

/**
 * CompoundLiteral
 * 
 * A class representing a compound literal.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#CompoundLiteral
 */
export const CompoundLiteral: Class = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#CompoundLiteral';

/**
 * HTML
 * 
 * The datatype of RDF literals storing fragments of HTML content
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#HTML
 */
export const HTML: Datatype = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#HTML';

/**
 * langString
 * 
 * The datatype of language-tagged string values
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#langString
 */
export const langString: Datatype = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#langString';

/**
 * PlainLiteral
 * 
 * The class of plain (i.e. untyped) literal values, as used in RIF and OWL 2
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#PlainLiteral
 */
export const PlainLiteral: Datatype = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#PlainLiteral';

/**
 * XMLLiteral
 * 
 * The datatype of XML literal values.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#XMLLiteral
 */
export const XMLLiteral: Datatype = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#XMLLiteral';

/**
 * JSON
 * 
 * The datatype of RDF literals storing JSON content.
 *
 * http://www.w3.org/1999/02/22-rdf-syntax-ns#JSON
 */
export const JSON: Datatype = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#JSON';
