// rdf-namespaces/dist/rdfs.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * subClassOf
 * 
 * The subject is a subclass of a class.
 *
 * http://www.w3.org/2000/01/rdf-schema#subClassOf
 */
export const subClassOf: Property = 'http://www.w3.org/2000/01/rdf-schema#subClassOf';

/**
 * subPropertyOf
 * 
 * The subject is a subproperty of a property.
 *
 * http://www.w3.org/2000/01/rdf-schema#subPropertyOf
 */
export const subPropertyOf: Property = 'http://www.w3.org/2000/01/rdf-schema#subPropertyOf';

/**
 * comment
 * 
 * A description of the subject resource.
 *
 * http://www.w3.org/2000/01/rdf-schema#comment
 */
export const comment: Property = 'http://www.w3.org/2000/01/rdf-schema#comment';

/**
 * label
 * 
 * A human-readable name for the subject.
 *
 * http://www.w3.org/2000/01/rdf-schema#label
 */
export const label: Property = 'http://www.w3.org/2000/01/rdf-schema#label';

/**
 * domain
 * 
 * A domain of the subject property.
 *
 * http://www.w3.org/2000/01/rdf-schema#domain
 */
export const domain: Property = 'http://www.w3.org/2000/01/rdf-schema#domain';

/**
 * range
 * 
 * A range of the subject property.
 *
 * http://www.w3.org/2000/01/rdf-schema#range
 */
export const range: Property = 'http://www.w3.org/2000/01/rdf-schema#range';

/**
 * seeAlso
 * 
 * Further information about the subject resource.
 *
 * http://www.w3.org/2000/01/rdf-schema#seeAlso
 */
export const seeAlso: Property = 'http://www.w3.org/2000/01/rdf-schema#seeAlso';

/**
 * isDefinedBy
 * 
 * The defininition of the subject resource.
 *
 * http://www.w3.org/2000/01/rdf-schema#isDefinedBy
 */
export const isDefinedBy: Property = 'http://www.w3.org/2000/01/rdf-schema#isDefinedBy';

/**
 * member
 * 
 * A member of the subject resource.
 *
 * http://www.w3.org/2000/01/rdf-schema#member
 */
export const member: Property = 'http://www.w3.org/2000/01/rdf-schema#member';

/**
 * Resource
 * 
 * The class resource, everything.
 *
 * http://www.w3.org/2000/01/rdf-schema#Resource
 */
export const Resource: Class = 'http://www.w3.org/2000/01/rdf-schema#Resource';

/**
 * Class
 * 
 * The class of classes.
 *
 * http://www.w3.org/2000/01/rdf-schema#Class
 */
export const Class: Class = 'http://www.w3.org/2000/01/rdf-schema#Class';

/**
 * Literal
 * 
 * The class of literal values, eg. textual strings and integers.
 *
 * http://www.w3.org/2000/01/rdf-schema#Literal
 */
export const Literal: Class = 'http://www.w3.org/2000/01/rdf-schema#Literal';

/**
 * Container
 * 
 * The class of RDF containers.
 *
 * http://www.w3.org/2000/01/rdf-schema#Container
 */
export const Container: Class = 'http://www.w3.org/2000/01/rdf-schema#Container';

/**
 * ContainerMembershipProperty
 * 
 * The class of container membership properties, rdf:_1, rdf:_2, ...,
 *                     all of which are sub-properties of 'member'.
 *
 * http://www.w3.org/2000/01/rdf-schema#ContainerMembershipProperty
 */
export const ContainerMembershipProperty: Class = 'http://www.w3.org/2000/01/rdf-schema#ContainerMembershipProperty';

/**
 * Datatype
 * 
 * The class of RDF datatypes.
 *
 * http://www.w3.org/2000/01/rdf-schema#Datatype
 */
export const Datatype: Class = 'http://www.w3.org/2000/01/rdf-schema#Datatype';
