// rdf-namespaces/dist/sched.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/pim/schedule#availabilityOptions
 */
export const availabilityOptions: Property = 'http://www.w3.org/ns/pim/schedule#availabilityOptions';

/**
 * availability
 * 
 * 
 *
 * http://www.w3.org/ns/pim/schedule#availibility
 */
export const availibility: Property = 'http://www.w3.org/ns/pim/schedule#availibility';

/**
 * possible time for event
 * 
 * 
 *
 * http://www.w3.org/ns/pim/schedule#option
 */
export const option: Property = 'http://www.w3.org/ns/pim/schedule#option';

/**
 * response
 * 
 * 
 *
 * http://www.w3.org/ns/pim/schedule#response
 */
export const response: Property = 'http://www.w3.org/ns/pim/schedule#response';

/**
 * Degrees of difficulty
 * 
 * Provides users of a poll with a way to reflect the difficulty, pain
 *     or disruption which would be involved in a given choice.
 *
 * http://www.w3.org/ns/pim/schedule#DegreesOfDifficulty
 */
export const DegreesOfDifficulty: Class = 'http://www.w3.org/ns/pim/schedule#DegreesOfDifficulty';

/**
 * difficult
 * 
 * I would rather a different choice.
 *
 * http://www.w3.org/ns/pim/schedule#Difficult
 */
export const Difficult: Class = 'http://www.w3.org/ns/pim/schedule#Difficult';

/**
 * ideal
 * 
 * This would be perfect for me.
 *
 * http://www.w3.org/ns/pim/schedule#Ideal
 */
export const Ideal: Class = 'http://www.w3.org/ns/pim/schedule#Ideal';

/**
 * impossible
 * 
 * I really cannot do this.
 *
 * http://www.w3.org/ns/pim/schedule#Impossible
 */
export const Impossible: Class = 'http://www.w3.org/ns/pim/schedule#Impossible';

/**
 * painfull
 * 
 * This would cause acute disruption
 *
 * http://www.w3.org/ns/pim/schedule#Painful
 */
export const Painful: Class = 'http://www.w3.org/ns/pim/schedule#Painful';

/**
 * tolerable
 * 
 * I can do this, not ideal.
 *
 * http://www.w3.org/ns/pim/schedule#Tolerable
 */
export const Tolerable: Class = 'http://www.w3.org/ns/pim/schedule#Tolerable';

/**
 * Yes or No
 * 
 * Provides users of a poll with the simplest choice.
 *
 * http://www.w3.org/ns/pim/schedule#YesNo
 */
export const YesNo: Class = 'http://www.w3.org/ns/pim/schedule#YesNo';

/**
 * Yes, No or Maybe
 * 
 * Provides users of a poll with a little uncertainty.
 *
 * http://www.w3.org/ns/pim/schedule#YesNoMaybe
 */
export const YesNoMaybe: Class = 'http://www.w3.org/ns/pim/schedule#YesNoMaybe';
