// rdf-namespaces/dist/schema.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var schema$1 = require('./schema-a97ab12e.js');



exports.AMRadioChannel = schema$1.AMRadioChannel;
exports.APIReference = schema$1.APIReference;
exports.AboutPage = schema$1.AboutPage;
exports.AcceptAction = schema$1.AcceptAction;
exports.Accommodation = schema$1.Accommodation;
exports.AccountingService = schema$1.AccountingService;
exports.AchieveAction = schema$1.AchieveAction;
exports.Action = schema$1.Action;
exports.ActionAccessSpecification = schema$1.ActionAccessSpecification;
exports.ActionStatusType = schema$1.ActionStatusType;
exports.ActivateAction = schema$1.ActivateAction;
exports.AddAction = schema$1.AddAction;
exports.AdministrativeArea = schema$1.AdministrativeArea;
exports.AdultEntertainment = schema$1.AdultEntertainment;
exports.AggregateOffer = schema$1.AggregateOffer;
exports.AggregateRating = schema$1.AggregateRating;
exports.AgreeAction = schema$1.AgreeAction;
exports.Airline = schema$1.Airline;
exports.Airport = schema$1.Airport;
exports.AlignmentObject = schema$1.AlignmentObject;
exports.AllocateAction = schema$1.AllocateAction;
exports.AmusementPark = schema$1.AmusementPark;
exports.AnimalShelter = schema$1.AnimalShelter;
exports.Answer = schema$1.Answer;
exports.Apartment = schema$1.Apartment;
exports.ApartmentComplex = schema$1.ApartmentComplex;
exports.AppendAction = schema$1.AppendAction;
exports.ApplyAction = schema$1.ApplyAction;
exports.Aquarium = schema$1.Aquarium;
exports.ArriveAction = schema$1.ArriveAction;
exports.ArtGallery = schema$1.ArtGallery;
exports.Article = schema$1.Article;
exports.AskAction = schema$1.AskAction;
exports.AssessAction = schema$1.AssessAction;
exports.AssignAction = schema$1.AssignAction;
exports.Attorney = schema$1.Attorney;
exports.Audience = schema$1.Audience;
exports.AudioObject = schema$1.AudioObject;
exports.AuthorizeAction = schema$1.AuthorizeAction;
exports.AutoBodyShop = schema$1.AutoBodyShop;
exports.AutoDealer = schema$1.AutoDealer;
exports.AutoPartsStore = schema$1.AutoPartsStore;
exports.AutoRental = schema$1.AutoRental;
exports.AutoRepair = schema$1.AutoRepair;
exports.AutoWash = schema$1.AutoWash;
exports.AutomatedTeller = schema$1.AutomatedTeller;
exports.AutomotiveBusiness = schema$1.AutomotiveBusiness;
exports.Bakery = schema$1.Bakery;
exports.BankAccount = schema$1.BankAccount;
exports.BankOrCreditUnion = schema$1.BankOrCreditUnion;
exports.BarOrPub = schema$1.BarOrPub;
exports.Barcode = schema$1.Barcode;
exports.Beach = schema$1.Beach;
exports.BeautySalon = schema$1.BeautySalon;
exports.BedAndBreakfast = schema$1.BedAndBreakfast;
exports.BedDetails = schema$1.BedDetails;
exports.BedType = schema$1.BedType;
exports.BefriendAction = schema$1.BefriendAction;
exports.BikeStore = schema$1.BikeStore;
exports.Blog = schema$1.Blog;
exports.BlogPosting = schema$1.BlogPosting;
exports.BoardingPolicyType = schema$1.BoardingPolicyType;
exports.BodyOfWater = schema$1.BodyOfWater;
exports.Book = schema$1.Book;
exports.BookFormatType = schema$1.BookFormatType;
exports.BookSeries = schema$1.BookSeries;
exports.BookStore = schema$1.BookStore;
exports.BookmarkAction = schema$1.BookmarkAction;
exports.Boolean = schema$1.Boolean;
exports.BorrowAction = schema$1.BorrowAction;
exports.BowlingAlley = schema$1.BowlingAlley;
exports.Brand = schema$1.Brand;
exports.BreadcrumbList = schema$1.BreadcrumbList;
exports.Brewery = schema$1.Brewery;
exports.Bridge = schema$1.Bridge;
exports.BroadcastChannel = schema$1.BroadcastChannel;
exports.BroadcastEvent = schema$1.BroadcastEvent;
exports.BroadcastFrequencySpecification = schema$1.BroadcastFrequencySpecification;
exports.BroadcastService = schema$1.BroadcastService;
exports.BuddhistTemple = schema$1.BuddhistTemple;
exports.BusReservation = schema$1.BusReservation;
exports.BusStation = schema$1.BusStation;
exports.BusStop = schema$1.BusStop;
exports.BusTrip = schema$1.BusTrip;
exports.BusinessAudience = schema$1.BusinessAudience;
exports.BusinessEntityType = schema$1.BusinessEntityType;
exports.BusinessEvent = schema$1.BusinessEvent;
exports.BusinessFunction = schema$1.BusinessFunction;
exports.BuyAction = schema$1.BuyAction;
exports.CableOrSatelliteService = schema$1.CableOrSatelliteService;
exports.CafeOrCoffeeShop = schema$1.CafeOrCoffeeShop;
exports.Campground = schema$1.Campground;
exports.CampingPitch = schema$1.CampingPitch;
exports.Canal = schema$1.Canal;
exports.CancelAction = schema$1.CancelAction;
exports.Car = schema$1.Car;
exports.Casino = schema$1.Casino;
exports.CatholicChurch = schema$1.CatholicChurch;
exports.Cemetery = schema$1.Cemetery;
exports.CheckAction = schema$1.CheckAction;
exports.CheckInAction = schema$1.CheckInAction;
exports.CheckOutAction = schema$1.CheckOutAction;
exports.CheckoutPage = schema$1.CheckoutPage;
exports.ChildCare = schema$1.ChildCare;
exports.ChildrensEvent = schema$1.ChildrensEvent;
exports.ChooseAction = schema$1.ChooseAction;
exports.Church = schema$1.Church;
exports.City = schema$1.City;
exports.CityHall = schema$1.CityHall;
exports.CivicStructure = schema$1.CivicStructure;
exports.ClaimReview = schema$1.ClaimReview;
exports.Clip = schema$1.Clip;
exports.ClothingStore = schema$1.ClothingStore;
exports.Code = schema$1.Code;
exports.CollectionPage = schema$1.CollectionPage;
exports.CollegeOrUniversity = schema$1.CollegeOrUniversity;
exports.ComedyClub = schema$1.ComedyClub;
exports.ComedyEvent = schema$1.ComedyEvent;
exports.Comment = schema$1.Comment;
exports.CommentAction = schema$1.CommentAction;
exports.CommunicateAction = schema$1.CommunicateAction;
exports.CompoundPriceSpecification = schema$1.CompoundPriceSpecification;
exports.ComputerLanguage = schema$1.ComputerLanguage;
exports.ComputerStore = schema$1.ComputerStore;
exports.ConfirmAction = schema$1.ConfirmAction;
exports.ConsumeAction = schema$1.ConsumeAction;
exports.ContactPage = schema$1.ContactPage;
exports.ContactPoint = schema$1.ContactPoint;
exports.ContactPointOption = schema$1.ContactPointOption;
exports.Continent = schema$1.Continent;
exports.ControlAction = schema$1.ControlAction;
exports.ConvenienceStore = schema$1.ConvenienceStore;
exports.Conversation = schema$1.Conversation;
exports.CookAction = schema$1.CookAction;
exports.Corporation = schema$1.Corporation;
exports.Country = schema$1.Country;
exports.Course = schema$1.Course;
exports.CourseInstance = schema$1.CourseInstance;
exports.Courthouse = schema$1.Courthouse;
exports.CreateAction = schema$1.CreateAction;
exports.CreativeWork = schema$1.CreativeWork;
exports.CreativeWorkSeason = schema$1.CreativeWorkSeason;
exports.CreativeWorkSeries = schema$1.CreativeWorkSeries;
exports.CreditCard = schema$1.CreditCard;
exports.Crematorium = schema$1.Crematorium;
exports.CurrencyConversionService = schema$1.CurrencyConversionService;
exports.DanceEvent = schema$1.DanceEvent;
exports.DanceGroup = schema$1.DanceGroup;
exports.DataCatalog = schema$1.DataCatalog;
exports.DataDownload = schema$1.DataDownload;
exports.DataFeed = schema$1.DataFeed;
exports.DataFeedItem = schema$1.DataFeedItem;
exports.DataType = schema$1.DataType;
exports.Dataset = schema$1.Dataset;
exports.Date = schema$1.Date;
exports.DateTime = schema$1.DateTime;
exports.DatedMoneySpecification = schema$1.DatedMoneySpecification;
exports.DayOfWeek = schema$1.DayOfWeek;
exports.DaySpa = schema$1.DaySpa;
exports.DeactivateAction = schema$1.DeactivateAction;
exports.DefenceEstablishment = schema$1.DefenceEstablishment;
exports.DeleteAction = schema$1.DeleteAction;
exports.DeliveryChargeSpecification = schema$1.DeliveryChargeSpecification;
exports.DeliveryEvent = schema$1.DeliveryEvent;
exports.DeliveryMethod = schema$1.DeliveryMethod;
exports.Demand = schema$1.Demand;
exports.Dentist = schema$1.Dentist;
exports.DepartAction = schema$1.DepartAction;
exports.DepartmentStore = schema$1.DepartmentStore;
exports.DepositAccount = schema$1.DepositAccount;
exports.DigitalDocument = schema$1.DigitalDocument;
exports.DigitalDocumentPermission = schema$1.DigitalDocumentPermission;
exports.DigitalDocumentPermissionType = schema$1.DigitalDocumentPermissionType;
exports.DisagreeAction = schema$1.DisagreeAction;
exports.DiscoverAction = schema$1.DiscoverAction;
exports.DiscussionForumPosting = schema$1.DiscussionForumPosting;
exports.DislikeAction = schema$1.DislikeAction;
exports.Distance = schema$1.Distance;
exports.Distillery = schema$1.Distillery;
exports.DonateAction = schema$1.DonateAction;
exports.DownloadAction = schema$1.DownloadAction;
exports.DrawAction = schema$1.DrawAction;
exports.DrinkAction = schema$1.DrinkAction;
exports.DriveWheelConfigurationValue = schema$1.DriveWheelConfigurationValue;
exports.DryCleaningOrLaundry = schema$1.DryCleaningOrLaundry;
exports.Duration = schema$1.Duration;
exports.EatAction = schema$1.EatAction;
exports.EducationEvent = schema$1.EducationEvent;
exports.EducationalAudience = schema$1.EducationalAudience;
exports.EducationalOrganization = schema$1.EducationalOrganization;
exports.Electrician = schema$1.Electrician;
exports.ElectronicsStore = schema$1.ElectronicsStore;
exports.ElementarySchool = schema$1.ElementarySchool;
exports.EmailMessage = schema$1.EmailMessage;
exports.Embassy = schema$1.Embassy;
exports.EmergencyService = schema$1.EmergencyService;
exports.EmployeeRole = schema$1.EmployeeRole;
exports.EmployerAggregateRating = schema$1.EmployerAggregateRating;
exports.EmploymentAgency = schema$1.EmploymentAgency;
exports.EndorseAction = schema$1.EndorseAction;
exports.EndorsementRating = schema$1.EndorsementRating;
exports.Energy = schema$1.Energy;
exports.EngineSpecification = schema$1.EngineSpecification;
exports.EntertainmentBusiness = schema$1.EntertainmentBusiness;
exports.EntryPoint = schema$1.EntryPoint;
exports.Enumeration = schema$1.Enumeration;
exports.Episode = schema$1.Episode;
exports.Event = schema$1.Event;
exports.EventReservation = schema$1.EventReservation;
exports.EventStatusType = schema$1.EventStatusType;
exports.EventVenue = schema$1.EventVenue;
exports.ExerciseAction = schema$1.ExerciseAction;
exports.ExerciseGym = schema$1.ExerciseGym;
exports.ExhibitionEvent = schema$1.ExhibitionEvent;
exports.FAQPage = schema$1.FAQPage;
exports.FMRadioChannel = schema$1.FMRadioChannel;
exports.FastFoodRestaurant = schema$1.FastFoodRestaurant;
exports.Festival = schema$1.Festival;
exports.FilmAction = schema$1.FilmAction;
exports.FinancialProduct = schema$1.FinancialProduct;
exports.FinancialService = schema$1.FinancialService;
exports.FindAction = schema$1.FindAction;
exports.FireStation = schema$1.FireStation;
exports.Flight = schema$1.Flight;
exports.FlightReservation = schema$1.FlightReservation;
exports.Float = schema$1.Float;
exports.Florist = schema$1.Florist;
exports.FollowAction = schema$1.FollowAction;
exports.FoodEstablishment = schema$1.FoodEstablishment;
exports.FoodEstablishmentReservation = schema$1.FoodEstablishmentReservation;
exports.FoodEvent = schema$1.FoodEvent;
exports.FoodService = schema$1.FoodService;
exports.FurnitureStore = schema$1.FurnitureStore;
exports.Game = schema$1.Game;
exports.GamePlayMode = schema$1.GamePlayMode;
exports.GameServer = schema$1.GameServer;
exports.GameServerStatus = schema$1.GameServerStatus;
exports.GardenStore = schema$1.GardenStore;
exports.GasStation = schema$1.GasStation;
exports.GatedResidenceCommunity = schema$1.GatedResidenceCommunity;
exports.GenderType = schema$1.GenderType;
exports.GeneralContractor = schema$1.GeneralContractor;
exports.GeoCircle = schema$1.GeoCircle;
exports.GeoCoordinates = schema$1.GeoCoordinates;
exports.GeoShape = schema$1.GeoShape;
exports.GiveAction = schema$1.GiveAction;
exports.GolfCourse = schema$1.GolfCourse;
exports.GovernmentBuilding = schema$1.GovernmentBuilding;
exports.GovernmentOffice = schema$1.GovernmentOffice;
exports.GovernmentOrganization = schema$1.GovernmentOrganization;
exports.GovernmentPermit = schema$1.GovernmentPermit;
exports.GovernmentService = schema$1.GovernmentService;
exports.GroceryStore = schema$1.GroceryStore;
exports.HVACBusiness = schema$1.HVACBusiness;
exports.HairSalon = schema$1.HairSalon;
exports.HardwareStore = schema$1.HardwareStore;
exports.HealthAndBeautyBusiness = schema$1.HealthAndBeautyBusiness;
exports.HealthClub = schema$1.HealthClub;
exports.HighSchool = schema$1.HighSchool;
exports.HinduTemple = schema$1.HinduTemple;
exports.HobbyShop = schema$1.HobbyShop;
exports.HomeAndConstructionBusiness = schema$1.HomeAndConstructionBusiness;
exports.HomeGoodsStore = schema$1.HomeGoodsStore;
exports.Hospital = schema$1.Hospital;
exports.Hostel = schema$1.Hostel;
exports.Hotel = schema$1.Hotel;
exports.HotelRoom = schema$1.HotelRoom;
exports.House = schema$1.House;
exports.HousePainter = schema$1.HousePainter;
exports.HowTo = schema$1.HowTo;
exports.HowToDirection = schema$1.HowToDirection;
exports.HowToItem = schema$1.HowToItem;
exports.HowToSection = schema$1.HowToSection;
exports.HowToStep = schema$1.HowToStep;
exports.HowToSupply = schema$1.HowToSupply;
exports.HowToTip = schema$1.HowToTip;
exports.HowToTool = schema$1.HowToTool;
exports.IceCreamShop = schema$1.IceCreamShop;
exports.IgnoreAction = schema$1.IgnoreAction;
exports.ImageGallery = schema$1.ImageGallery;
exports.ImageObject = schema$1.ImageObject;
exports.IndividualProduct = schema$1.IndividualProduct;
exports.InformAction = schema$1.InformAction;
exports.InsertAction = schema$1.InsertAction;
exports.InstallAction = schema$1.InstallAction;
exports.InsuranceAgency = schema$1.InsuranceAgency;
exports.Intangible = schema$1.Intangible;
exports.Integer = schema$1.Integer;
exports.InteractAction = schema$1.InteractAction;
exports.InteractionCounter = schema$1.InteractionCounter;
exports.InternetCafe = schema$1.InternetCafe;
exports.InvestmentOrDeposit = schema$1.InvestmentOrDeposit;
exports.InviteAction = schema$1.InviteAction;
exports.Invoice = schema$1.Invoice;
exports.ItemAvailability = schema$1.ItemAvailability;
exports.ItemList = schema$1.ItemList;
exports.ItemListOrderType = schema$1.ItemListOrderType;
exports.ItemPage = schema$1.ItemPage;
exports.JewelryStore = schema$1.JewelryStore;
exports.JobPosting = schema$1.JobPosting;
exports.JoinAction = schema$1.JoinAction;
exports.LakeBodyOfWater = schema$1.LakeBodyOfWater;
exports.Landform = schema$1.Landform;
exports.LandmarksOrHistoricalBuildings = schema$1.LandmarksOrHistoricalBuildings;
exports.Language = schema$1.Language;
exports.LeaveAction = schema$1.LeaveAction;
exports.LegalService = schema$1.LegalService;
exports.LegislativeBuilding = schema$1.LegislativeBuilding;
exports.LendAction = schema$1.LendAction;
exports.Library = schema$1.Library;
exports.LikeAction = schema$1.LikeAction;
exports.LiquorStore = schema$1.LiquorStore;
exports.ListItem = schema$1.ListItem;
exports.ListenAction = schema$1.ListenAction;
exports.LiteraryEvent = schema$1.LiteraryEvent;
exports.LiveBlogPosting = schema$1.LiveBlogPosting;
exports.LoanOrCredit = schema$1.LoanOrCredit;
exports.LocalBusiness = schema$1.LocalBusiness;
exports.LocationFeatureSpecification = schema$1.LocationFeatureSpecification;
exports.LockerDelivery = schema$1.LockerDelivery;
exports.Locksmith = schema$1.Locksmith;
exports.LodgingBusiness = schema$1.LodgingBusiness;
exports.LodgingReservation = schema$1.LodgingReservation;
exports.LoseAction = schema$1.LoseAction;
exports.Map = schema$1.Map;
exports.MapCategoryType = schema$1.MapCategoryType;
exports.MarryAction = schema$1.MarryAction;
exports.Mass = schema$1.Mass;
exports.MediaObject = schema$1.MediaObject;
exports.MediaSubscription = schema$1.MediaSubscription;
exports.MedicalOrganization = schema$1.MedicalOrganization;
exports.MeetingRoom = schema$1.MeetingRoom;
exports.MensClothingStore = schema$1.MensClothingStore;
exports.Menu = schema$1.Menu;
exports.MenuItem = schema$1.MenuItem;
exports.MenuSection = schema$1.MenuSection;
exports.Message = schema$1.Message;
exports.MiddleSchool = schema$1.MiddleSchool;
exports.MobileApplication = schema$1.MobileApplication;
exports.MobilePhoneStore = schema$1.MobilePhoneStore;
exports.MonetaryAmount = schema$1.MonetaryAmount;
exports.MonetaryAmountDistribution = schema$1.MonetaryAmountDistribution;
exports.Mosque = schema$1.Mosque;
exports.Motel = schema$1.Motel;
exports.MotorcycleDealer = schema$1.MotorcycleDealer;
exports.MotorcycleRepair = schema$1.MotorcycleRepair;
exports.Mountain = schema$1.Mountain;
exports.MoveAction = schema$1.MoveAction;
exports.Movie = schema$1.Movie;
exports.MovieClip = schema$1.MovieClip;
exports.MovieRentalStore = schema$1.MovieRentalStore;
exports.MovieSeries = schema$1.MovieSeries;
exports.MovieTheater = schema$1.MovieTheater;
exports.MovingCompany = schema$1.MovingCompany;
exports.Museum = schema$1.Museum;
exports.MusicAlbum = schema$1.MusicAlbum;
exports.MusicAlbumProductionType = schema$1.MusicAlbumProductionType;
exports.MusicAlbumReleaseType = schema$1.MusicAlbumReleaseType;
exports.MusicComposition = schema$1.MusicComposition;
exports.MusicEvent = schema$1.MusicEvent;
exports.MusicGroup = schema$1.MusicGroup;
exports.MusicPlaylist = schema$1.MusicPlaylist;
exports.MusicRecording = schema$1.MusicRecording;
exports.MusicRelease = schema$1.MusicRelease;
exports.MusicReleaseFormatType = schema$1.MusicReleaseFormatType;
exports.MusicStore = schema$1.MusicStore;
exports.MusicVenue = schema$1.MusicVenue;
exports.MusicVideoObject = schema$1.MusicVideoObject;
exports.NGO = schema$1.NGO;
exports.NailSalon = schema$1.NailSalon;
exports.NewsArticle = schema$1.NewsArticle;
exports.NightClub = schema$1.NightClub;
exports.Notary = schema$1.Notary;
exports.NoteDigitalDocument = schema$1.NoteDigitalDocument;
exports.Number = schema$1.Number;
exports.NutritionInformation = schema$1.NutritionInformation;
exports.Occupation = schema$1.Occupation;
exports.OceanBodyOfWater = schema$1.OceanBodyOfWater;
exports.Offer = schema$1.Offer;
exports.OfferCatalog = schema$1.OfferCatalog;
exports.OfferItemCondition = schema$1.OfferItemCondition;
exports.OfficeEquipmentStore = schema$1.OfficeEquipmentStore;
exports.OnDemandEvent = schema$1.OnDemandEvent;
exports.OpeningHoursSpecification = schema$1.OpeningHoursSpecification;
exports.Order = schema$1.Order;
exports.OrderAction = schema$1.OrderAction;
exports.OrderItem = schema$1.OrderItem;
exports.OrderStatus = schema$1.OrderStatus;
exports.Organization = schema$1.Organization;
exports.OrganizationRole = schema$1.OrganizationRole;
exports.OrganizeAction = schema$1.OrganizeAction;
exports.OutletStore = schema$1.OutletStore;
exports.OwnershipInfo = schema$1.OwnershipInfo;
exports.PaintAction = schema$1.PaintAction;
exports.Painting = schema$1.Painting;
exports.ParcelDelivery = schema$1.ParcelDelivery;
exports.ParcelService = schema$1.ParcelService;
exports.ParentAudience = schema$1.ParentAudience;
exports.Park = schema$1.Park;
exports.ParkingFacility = schema$1.ParkingFacility;
exports.PawnShop = schema$1.PawnShop;
exports.PayAction = schema$1.PayAction;
exports.PaymentCard = schema$1.PaymentCard;
exports.PaymentChargeSpecification = schema$1.PaymentChargeSpecification;
exports.PaymentMethod = schema$1.PaymentMethod;
exports.PaymentService = schema$1.PaymentService;
exports.PaymentStatusType = schema$1.PaymentStatusType;
exports.PeopleAudience = schema$1.PeopleAudience;
exports.PerformAction = schema$1.PerformAction;
exports.PerformanceRole = schema$1.PerformanceRole;
exports.PerformingArtsTheater = schema$1.PerformingArtsTheater;
exports.PerformingGroup = schema$1.PerformingGroup;
exports.Periodical = schema$1.Periodical;
exports.Permit = schema$1.Permit;
exports.Person = schema$1.Person;
exports.PetStore = schema$1.PetStore;
exports.Pharmacy = schema$1.Pharmacy;
exports.Photograph = schema$1.Photograph;
exports.PhotographAction = schema$1.PhotographAction;
exports.Physician = schema$1.Physician;
exports.Place = schema$1.Place;
exports.PlaceOfWorship = schema$1.PlaceOfWorship;
exports.PlanAction = schema$1.PlanAction;
exports.PlayAction = schema$1.PlayAction;
exports.Playground = schema$1.Playground;
exports.Plumber = schema$1.Plumber;
exports.PoliceStation = schema$1.PoliceStation;
exports.Pond = schema$1.Pond;
exports.PostOffice = schema$1.PostOffice;
exports.PostalAddress = schema$1.PostalAddress;
exports.PreOrderAction = schema$1.PreOrderAction;
exports.PrependAction = schema$1.PrependAction;
exports.Preschool = schema$1.Preschool;
exports.PresentationDigitalDocument = schema$1.PresentationDigitalDocument;
exports.PriceSpecification = schema$1.PriceSpecification;
exports.Product = schema$1.Product;
exports.ProductModel = schema$1.ProductModel;
exports.ProfessionalService = schema$1.ProfessionalService;
exports.ProfilePage = schema$1.ProfilePage;
exports.ProgramMembership = schema$1.ProgramMembership;
exports.PropertyValue = schema$1.PropertyValue;
exports.PropertyValueSpecification = schema$1.PropertyValueSpecification;
exports.PublicSwimmingPool = schema$1.PublicSwimmingPool;
exports.PublicationEvent = schema$1.PublicationEvent;
exports.PublicationIssue = schema$1.PublicationIssue;
exports.PublicationVolume = schema$1.PublicationVolume;
exports.QAPage = schema$1.QAPage;
exports.QualitativeValue = schema$1.QualitativeValue;
exports.QuantitativeValue = schema$1.QuantitativeValue;
exports.QuantitativeValueDistribution = schema$1.QuantitativeValueDistribution;
exports.Quantity = schema$1.Quantity;
exports.Question = schema$1.Question;
exports.QuoteAction = schema$1.QuoteAction;
exports.RVPark = schema$1.RVPark;
exports.RadioChannel = schema$1.RadioChannel;
exports.RadioClip = schema$1.RadioClip;
exports.RadioEpisode = schema$1.RadioEpisode;
exports.RadioSeason = schema$1.RadioSeason;
exports.RadioSeries = schema$1.RadioSeries;
exports.RadioStation = schema$1.RadioStation;
exports.Rating = schema$1.Rating;
exports.ReactAction = schema$1.ReactAction;
exports.ReadAction = schema$1.ReadAction;
exports.RealEstateAgent = schema$1.RealEstateAgent;
exports.ReceiveAction = schema$1.ReceiveAction;
exports.Recipe = schema$1.Recipe;
exports.RecyclingCenter = schema$1.RecyclingCenter;
exports.RegisterAction = schema$1.RegisterAction;
exports.RejectAction = schema$1.RejectAction;
exports.RentAction = schema$1.RentAction;
exports.RentalCarReservation = schema$1.RentalCarReservation;
exports.ReplaceAction = schema$1.ReplaceAction;
exports.ReplyAction = schema$1.ReplyAction;
exports.Report = schema$1.Report;
exports.Reservation = schema$1.Reservation;
exports.ReservationPackage = schema$1.ReservationPackage;
exports.ReservationStatusType = schema$1.ReservationStatusType;
exports.ReserveAction = schema$1.ReserveAction;
exports.Reservoir = schema$1.Reservoir;
exports.Residence = schema$1.Residence;
exports.Resort = schema$1.Resort;
exports.Restaurant = schema$1.Restaurant;
exports.RestrictedDiet = schema$1.RestrictedDiet;
exports.ResumeAction = schema$1.ResumeAction;
exports.ReturnAction = schema$1.ReturnAction;
exports.Review = schema$1.Review;
exports.ReviewAction = schema$1.ReviewAction;
exports.RiverBodyOfWater = schema$1.RiverBodyOfWater;
exports.Role = schema$1.Role;
exports.RoofingContractor = schema$1.RoofingContractor;
exports.Room = schema$1.Room;
exports.RsvpAction = schema$1.RsvpAction;
exports.RsvpResponseType = schema$1.RsvpResponseType;
exports.SaleEvent = schema$1.SaleEvent;
exports.ScheduleAction = schema$1.ScheduleAction;
exports.ScholarlyArticle = schema$1.ScholarlyArticle;
exports.School = schema$1.School;
exports.ScreeningEvent = schema$1.ScreeningEvent;
exports.Sculpture = schema$1.Sculpture;
exports.SeaBodyOfWater = schema$1.SeaBodyOfWater;
exports.SearchAction = schema$1.SearchAction;
exports.SearchResultsPage = schema$1.SearchResultsPage;
exports.Season = schema$1.Season;
exports.Seat = schema$1.Seat;
exports.SelfStorage = schema$1.SelfStorage;
exports.SellAction = schema$1.SellAction;
exports.SendAction = schema$1.SendAction;
exports.Series = schema$1.Series;
exports.Service = schema$1.Service;
exports.ServiceChannel = schema$1.ServiceChannel;
exports.ShareAction = schema$1.ShareAction;
exports.ShoeStore = schema$1.ShoeStore;
exports.ShoppingCenter = schema$1.ShoppingCenter;
exports.SingleFamilyResidence = schema$1.SingleFamilyResidence;
exports.SiteNavigationElement = schema$1.SiteNavigationElement;
exports.SkiResort = schema$1.SkiResort;
exports.SocialEvent = schema$1.SocialEvent;
exports.SocialMediaPosting = schema$1.SocialMediaPosting;
exports.SoftwareApplication = schema$1.SoftwareApplication;
exports.SoftwareSourceCode = schema$1.SoftwareSourceCode;
exports.SomeProducts = schema$1.SomeProducts;
exports.SpeakableSpecification = schema$1.SpeakableSpecification;
exports.Specialty = schema$1.Specialty;
exports.SportingGoodsStore = schema$1.SportingGoodsStore;
exports.SportsActivityLocation = schema$1.SportsActivityLocation;
exports.SportsClub = schema$1.SportsClub;
exports.SportsEvent = schema$1.SportsEvent;
exports.SportsOrganization = schema$1.SportsOrganization;
exports.SportsTeam = schema$1.SportsTeam;
exports.SpreadsheetDigitalDocument = schema$1.SpreadsheetDigitalDocument;
exports.StadiumOrArena = schema$1.StadiumOrArena;
exports.State = schema$1.State;
exports.SteeringPositionValue = schema$1.SteeringPositionValue;
exports.Store = schema$1.Store;
exports.StructuredValue = schema$1.StructuredValue;
exports.SubscribeAction = schema$1.SubscribeAction;
exports.SubwayStation = schema$1.SubwayStation;
exports.Suite = schema$1.Suite;
exports.SuspendAction = schema$1.SuspendAction;
exports.Synagogue = schema$1.Synagogue;
exports.TVClip = schema$1.TVClip;
exports.TVEpisode = schema$1.TVEpisode;
exports.TVSeason = schema$1.TVSeason;
exports.TVSeries = schema$1.TVSeries;
exports.Table = schema$1.Table;
exports.TakeAction = schema$1.TakeAction;
exports.TattooParlor = schema$1.TattooParlor;
exports.Taxi = schema$1.Taxi;
exports.TaxiReservation = schema$1.TaxiReservation;
exports.TaxiService = schema$1.TaxiService;
exports.TaxiStand = schema$1.TaxiStand;
exports.TechArticle = schema$1.TechArticle;
exports.TelevisionChannel = schema$1.TelevisionChannel;
exports.TelevisionStation = schema$1.TelevisionStation;
exports.TennisComplex = schema$1.TennisComplex;
exports.Text = schema$1.Text;
exports.TextDigitalDocument = schema$1.TextDigitalDocument;
exports.TheaterEvent = schema$1.TheaterEvent;
exports.TheaterGroup = schema$1.TheaterGroup;
exports.Thing = schema$1.Thing;
exports.Ticket = schema$1.Ticket;
exports.TieAction = schema$1.TieAction;
exports.Time = schema$1.Time;
exports.TipAction = schema$1.TipAction;
exports.TireShop = schema$1.TireShop;
exports.TouristAttraction = schema$1.TouristAttraction;
exports.TouristInformationCenter = schema$1.TouristInformationCenter;
exports.ToyStore = schema$1.ToyStore;
exports.TrackAction = schema$1.TrackAction;
exports.TradeAction = schema$1.TradeAction;
exports.TrainReservation = schema$1.TrainReservation;
exports.TrainStation = schema$1.TrainStation;
exports.TrainTrip = schema$1.TrainTrip;
exports.TransferAction = schema$1.TransferAction;
exports.TravelAction = schema$1.TravelAction;
exports.TravelAgency = schema$1.TravelAgency;
exports.Trip = schema$1.Trip;
exports.TypeAndQuantityNode = schema$1.TypeAndQuantityNode;
exports.URL = schema$1.URL;
exports.UnRegisterAction = schema$1.UnRegisterAction;
exports.UnitPriceSpecification = schema$1.UnitPriceSpecification;
exports.UpdateAction = schema$1.UpdateAction;
exports.UseAction = schema$1.UseAction;
exports.UserBlocks = schema$1.UserBlocks;
exports.UserCheckins = schema$1.UserCheckins;
exports.UserComments = schema$1.UserComments;
exports.UserDownloads = schema$1.UserDownloads;
exports.UserInteraction = schema$1.UserInteraction;
exports.UserLikes = schema$1.UserLikes;
exports.UserPageVisits = schema$1.UserPageVisits;
exports.UserPlays = schema$1.UserPlays;
exports.UserPlusOnes = schema$1.UserPlusOnes;
exports.UserTweets = schema$1.UserTweets;
exports.Vehicle = schema$1.Vehicle;
exports.VideoGallery = schema$1.VideoGallery;
exports.VideoGame = schema$1.VideoGame;
exports.VideoGameClip = schema$1.VideoGameClip;
exports.VideoGameSeries = schema$1.VideoGameSeries;
exports.VideoObject = schema$1.VideoObject;
exports.ViewAction = schema$1.ViewAction;
exports.VisualArtsEvent = schema$1.VisualArtsEvent;
exports.VisualArtwork = schema$1.VisualArtwork;
exports.Volcano = schema$1.Volcano;
exports.VoteAction = schema$1.VoteAction;
exports.WPAdBlock = schema$1.WPAdBlock;
exports.WPFooter = schema$1.WPFooter;
exports.WPHeader = schema$1.WPHeader;
exports.WPSideBar = schema$1.WPSideBar;
exports.WantAction = schema$1.WantAction;
exports.WarrantyPromise = schema$1.WarrantyPromise;
exports.WarrantyScope = schema$1.WarrantyScope;
exports.WatchAction = schema$1.WatchAction;
exports.Waterfall = schema$1.Waterfall;
exports.WearAction = schema$1.WearAction;
exports.WebApplication = schema$1.WebApplication;
exports.WebPage = schema$1.WebPage;
exports.WebPageElement = schema$1.WebPageElement;
exports.WebSite = schema$1.WebSite;
exports.WholesaleStore = schema$1.WholesaleStore;
exports.WinAction = schema$1.WinAction;
exports.Winery = schema$1.Winery;
exports.WorkersUnion = schema$1.WorkersUnion;
exports.WriteAction = schema$1.WriteAction;
exports.Zoo = schema$1.Zoo;
exports.about = schema$1.about;
exports.acceptedAnswer = schema$1.acceptedAnswer;
exports.acceptedOffer = schema$1.acceptedOffer;
exports.acceptedPaymentMethod = schema$1.acceptedPaymentMethod;
exports.acceptsReservations = schema$1.acceptsReservations;
exports.accessCode = schema$1.accessCode;
exports.accessMode = schema$1.accessMode;
exports.accessModeSufficient = schema$1.accessModeSufficient;
exports.accessibilityAPI = schema$1.accessibilityAPI;
exports.accessibilityControl = schema$1.accessibilityControl;
exports.accessibilityFeature = schema$1.accessibilityFeature;
exports.accessibilityHazard = schema$1.accessibilityHazard;
exports.accessibilitySummary = schema$1.accessibilitySummary;
exports.accountId = schema$1.accountId;
exports.accountablePerson = schema$1.accountablePerson;
exports.acquiredFrom = schema$1.acquiredFrom;
exports.actionAccessibilityRequirement = schema$1.actionAccessibilityRequirement;
exports.actionApplication = schema$1.actionApplication;
exports.actionOption = schema$1.actionOption;
exports.actionPlatform = schema$1.actionPlatform;
exports.actionStatus = schema$1.actionStatus;
exports.actor = schema$1.actor;
exports.actors = schema$1.actors;
exports.addOn = schema$1.addOn;
exports.additionalName = schema$1.additionalName;
exports.additionalNumberOfGuests = schema$1.additionalNumberOfGuests;
exports.additionalProperty = schema$1.additionalProperty;
exports.additionalType = schema$1.additionalType;
exports.address = schema$1.address;
exports.addressCountry = schema$1.addressCountry;
exports.addressLocality = schema$1.addressLocality;
exports.addressRegion = schema$1.addressRegion;
exports.advanceBookingRequirement = schema$1.advanceBookingRequirement;
exports.affiliation = schema$1.affiliation;
exports.afterMedia = schema$1.afterMedia;
exports.agent = schema$1.agent;
exports.aggregateRating = schema$1.aggregateRating;
exports.aircraft = schema$1.aircraft;
exports.album = schema$1.album;
exports.albumProductionType = schema$1.albumProductionType;
exports.albumRelease = schema$1.albumRelease;
exports.albumReleaseType = schema$1.albumReleaseType;
exports.albums = schema$1.albums;
exports.alignmentType = schema$1.alignmentType;
exports.alternateName = schema$1.alternateName;
exports.alternativeHeadline = schema$1.alternativeHeadline;
exports.alumni = schema$1.alumni;
exports.alumniOf = schema$1.alumniOf;
exports.amenityFeature = schema$1.amenityFeature;
exports.amount = schema$1.amount;
exports.amountOfThisGood = schema$1.amountOfThisGood;
exports.annualPercentageRate = schema$1.annualPercentageRate;
exports.answerCount = schema$1.answerCount;
exports.application = schema$1.application;
exports.applicationCategory = schema$1.applicationCategory;
exports.applicationSubCategory = schema$1.applicationSubCategory;
exports.applicationSuite = schema$1.applicationSuite;
exports.appliesToDeliveryMethod = schema$1.appliesToDeliveryMethod;
exports.appliesToPaymentMethod = schema$1.appliesToPaymentMethod;
exports.area = schema$1.area;
exports.areaServed = schema$1.areaServed;
exports.arrivalAirport = schema$1.arrivalAirport;
exports.arrivalBusStop = schema$1.arrivalBusStop;
exports.arrivalGate = schema$1.arrivalGate;
exports.arrivalPlatform = schema$1.arrivalPlatform;
exports.arrivalStation = schema$1.arrivalStation;
exports.arrivalTerminal = schema$1.arrivalTerminal;
exports.arrivalTime = schema$1.arrivalTime;
exports.artEdition = schema$1.artEdition;
exports.artMedium = schema$1.artMedium;
exports.artform = schema$1.artform;
exports.articleBody = schema$1.articleBody;
exports.articleSection = schema$1.articleSection;
exports.artworkSurface = schema$1.artworkSurface;
exports.assembly = schema$1.assembly;
exports.assemblyVersion = schema$1.assemblyVersion;
exports.associatedArticle = schema$1.associatedArticle;
exports.associatedMedia = schema$1.associatedMedia;
exports.athlete = schema$1.athlete;
exports.attendee = schema$1.attendee;
exports.attendees = schema$1.attendees;
exports.audience = schema$1.audience;
exports.audienceType = schema$1.audienceType;
exports.audio = schema$1.audio;
exports.authenticator = schema$1.authenticator;
exports.author = schema$1.author;
exports.availability = schema$1.availability;
exports.availabilityEnds = schema$1.availabilityEnds;
exports.availabilityStarts = schema$1.availabilityStarts;
exports.availableAtOrFrom = schema$1.availableAtOrFrom;
exports.availableChannel = schema$1.availableChannel;
exports.availableDeliveryMethod = schema$1.availableDeliveryMethod;
exports.availableFrom = schema$1.availableFrom;
exports.availableLanguage = schema$1.availableLanguage;
exports.availableOnDevice = schema$1.availableOnDevice;
exports.availableThrough = schema$1.availableThrough;
exports.award = schema$1.award;
exports.awards = schema$1.awards;
exports.awayTeam = schema$1.awayTeam;
exports.baseSalary = schema$1.baseSalary;
exports.bccRecipient = schema$1.bccRecipient;
exports.bed = schema$1.bed;
exports.beforeMedia = schema$1.beforeMedia;
exports.benefits = schema$1.benefits;
exports.bestRating = schema$1.bestRating;
exports.billingAddress = schema$1.billingAddress;
exports.billingIncrement = schema$1.billingIncrement;
exports.billingPeriod = schema$1.billingPeriod;
exports.birthDate = schema$1.birthDate;
exports.birthPlace = schema$1.birthPlace;
exports.bitrate = schema$1.bitrate;
exports.blogPost = schema$1.blogPost;
exports.blogPosts = schema$1.blogPosts;
exports.boardingGroup = schema$1.boardingGroup;
exports.boardingPolicy = schema$1.boardingPolicy;
exports.bookEdition = schema$1.bookEdition;
exports.bookFormat = schema$1.bookFormat;
exports.bookingAgent = schema$1.bookingAgent;
exports.bookingTime = schema$1.bookingTime;
exports.borrower = schema$1.borrower;
exports.box = schema$1.box;
exports.branchCode = schema$1.branchCode;
exports.branchOf = schema$1.branchOf;
exports.brand = schema$1.brand;
exports.breadcrumb = schema$1.breadcrumb;
exports.broadcastAffiliateOf = schema$1.broadcastAffiliateOf;
exports.broadcastChannelId = schema$1.broadcastChannelId;
exports.broadcastDisplayName = schema$1.broadcastDisplayName;
exports.broadcastFrequency = schema$1.broadcastFrequency;
exports.broadcastFrequencyValue = schema$1.broadcastFrequencyValue;
exports.broadcastOfEvent = schema$1.broadcastOfEvent;
exports.broadcastServiceTier = schema$1.broadcastServiceTier;
exports.broadcastTimezone = schema$1.broadcastTimezone;
exports.broadcaster = schema$1.broadcaster;
exports.broker = schema$1.broker;
exports.browserRequirements = schema$1.browserRequirements;
exports.busName = schema$1.busName;
exports.busNumber = schema$1.busNumber;
exports.businessFunction = schema$1.businessFunction;
exports.buyer = schema$1.buyer;
exports.byArtist = schema$1.byArtist;
exports.calories = schema$1.calories;
exports.candidate = schema$1.candidate;
exports.caption = schema$1.caption;
exports.carbohydrateContent = schema$1.carbohydrateContent;
exports.cargoVolume = schema$1.cargoVolume;
exports.carrier = schema$1.carrier;
exports.carrierRequirements = schema$1.carrierRequirements;
exports.catalog = schema$1.catalog;
exports.catalogNumber = schema$1.catalogNumber;
exports.category = schema$1.category;
exports.ccRecipient = schema$1.ccRecipient;
exports.character = schema$1.character;
exports.characterAttribute = schema$1.characterAttribute;
exports.characterName = schema$1.characterName;
exports.cheatCode = schema$1.cheatCode;
exports.checkinTime = schema$1.checkinTime;
exports.checkoutTime = schema$1.checkoutTime;
exports.childMaxAge = schema$1.childMaxAge;
exports.childMinAge = schema$1.childMinAge;
exports.children = schema$1.children;
exports.cholesterolContent = schema$1.cholesterolContent;
exports.circle = schema$1.circle;
exports.citation = schema$1.citation;
exports.claimReviewed = schema$1.claimReviewed;
exports.clipNumber = schema$1.clipNumber;
exports.closes = schema$1.closes;
exports.coach = schema$1.coach;
exports.codeRepository = schema$1.codeRepository;
exports.codeSampleType = schema$1.codeSampleType;
exports.colleague = schema$1.colleague;
exports.colleagues = schema$1.colleagues;
exports.collection = schema$1.collection;
exports.color = schema$1.color;
exports.comment = schema$1.comment;
exports.commentCount = schema$1.commentCount;
exports.commentText = schema$1.commentText;
exports.commentTime = schema$1.commentTime;
exports.competitor = schema$1.competitor;
exports.composer = schema$1.composer;
exports.confirmationNumber = schema$1.confirmationNumber;
exports.contactOption = schema$1.contactOption;
exports.contactPoint = schema$1.contactPoint;
exports.contactPoints = schema$1.contactPoints;
exports.contactType = schema$1.contactType;
exports.containedIn = schema$1.containedIn;
exports.containedInPlace = schema$1.containedInPlace;
exports.containsPlace = schema$1.containsPlace;
exports.containsSeason = schema$1.containsSeason;
exports.contentLocation = schema$1.contentLocation;
exports.contentRating = schema$1.contentRating;
exports.contentSize = schema$1.contentSize;
exports.contentType = schema$1.contentType;
exports.contentUrl = schema$1.contentUrl;
exports.contributor = schema$1.contributor;
exports.cookTime = schema$1.cookTime;
exports.cookingMethod = schema$1.cookingMethod;
exports.copyrightHolder = schema$1.copyrightHolder;
exports.copyrightYear = schema$1.copyrightYear;
exports.countriesNotSupported = schema$1.countriesNotSupported;
exports.countriesSupported = schema$1.countriesSupported;
exports.countryOfOrigin = schema$1.countryOfOrigin;
exports.course = schema$1.course;
exports.courseCode = schema$1.courseCode;
exports.courseMode = schema$1.courseMode;
exports.coursePrerequisites = schema$1.coursePrerequisites;
exports.coverageEndTime = schema$1.coverageEndTime;
exports.coverageStartTime = schema$1.coverageStartTime;
exports.creator = schema$1.creator;
exports.creditedTo = schema$1.creditedTo;
exports.cssSelector = schema$1.cssSelector;
exports.currenciesAccepted = schema$1.currenciesAccepted;
exports.currency = schema$1.currency;
exports.customer = schema$1.customer;
exports.dataFeedElement = schema$1.dataFeedElement;
exports.dataset = schema$1.dataset;
exports.datasetTimeInterval = schema$1.datasetTimeInterval;
exports.dateCreated = schema$1.dateCreated;
exports.dateDeleted = schema$1.dateDeleted;
exports.dateIssued = schema$1.dateIssued;
exports.dateModified = schema$1.dateModified;
exports.datePosted = schema$1.datePosted;
exports.datePublished = schema$1.datePublished;
exports.dateRead = schema$1.dateRead;
exports.dateReceived = schema$1.dateReceived;
exports.dateSent = schema$1.dateSent;
exports.dateVehicleFirstRegistered = schema$1.dateVehicleFirstRegistered;
exports.dateline = schema$1.dateline;
exports.dayOfWeek = schema$1.dayOfWeek;
exports.deathDate = schema$1.deathDate;
exports.deathPlace = schema$1.deathPlace;
exports.defaultValue = schema$1.defaultValue;
exports.deliveryAddress = schema$1.deliveryAddress;
exports.deliveryLeadTime = schema$1.deliveryLeadTime;
exports.deliveryMethod = schema$1.deliveryMethod;
exports.deliveryStatus = schema$1.deliveryStatus;
exports.department = schema$1.department;
exports.departureAirport = schema$1.departureAirport;
exports.departureBusStop = schema$1.departureBusStop;
exports.departureGate = schema$1.departureGate;
exports.departurePlatform = schema$1.departurePlatform;
exports.departureStation = schema$1.departureStation;
exports.departureTerminal = schema$1.departureTerminal;
exports.departureTime = schema$1.departureTime;
exports.dependencies = schema$1.dependencies;
exports.depth = schema$1.depth;
exports.description = schema$1.description;
exports.device = schema$1.device;
exports.director = schema$1.director;
exports.directors = schema$1.directors;
exports.disambiguatingDescription = schema$1.disambiguatingDescription;
exports.discount = schema$1.discount;
exports.discountCode = schema$1.discountCode;
exports.discountCurrency = schema$1.discountCurrency;
exports.discusses = schema$1.discusses;
exports.discussionUrl = schema$1.discussionUrl;
exports.dissolutionDate = schema$1.dissolutionDate;
exports.distance = schema$1.distance;
exports.distribution = schema$1.distribution;
exports.doorTime = schema$1.doorTime;
exports.downloadUrl = schema$1.downloadUrl;
exports.downvoteCount = schema$1.downvoteCount;
exports.driveWheelConfiguration = schema$1.driveWheelConfiguration;
exports.dropoffLocation = schema$1.dropoffLocation;
exports.dropoffTime = schema$1.dropoffTime;
exports.duns = schema$1.duns;
exports.duration = schema$1.duration;
exports.durationOfWarranty = schema$1.durationOfWarranty;
exports.duringMedia = schema$1.duringMedia;
exports.editor = schema$1.editor;
exports.educationalAlignment = schema$1.educationalAlignment;
exports.educationalCredentialAwarded = schema$1.educationalCredentialAwarded;
exports.educationalFramework = schema$1.educationalFramework;
exports.educationalRole = schema$1.educationalRole;
exports.educationalUse = schema$1.educationalUse;
exports.elevation = schema$1.elevation;
exports.eligibleCustomerType = schema$1.eligibleCustomerType;
exports.eligibleDuration = schema$1.eligibleDuration;
exports.eligibleQuantity = schema$1.eligibleQuantity;
exports.eligibleRegion = schema$1.eligibleRegion;
exports.eligibleTransactionVolume = schema$1.eligibleTransactionVolume;
exports.email = schema$1.email;
exports.embedUrl = schema$1.embedUrl;
exports.employee = schema$1.employee;
exports.employees = schema$1.employees;
exports.employmentType = schema$1.employmentType;
exports.encodesCreativeWork = schema$1.encodesCreativeWork;
exports.encoding = schema$1.encoding;
exports.encodingFormat = schema$1.encodingFormat;
exports.encodingType = schema$1.encodingType;
exports.encodings = schema$1.encodings;
exports.endDate = schema$1.endDate;
exports.endTime = schema$1.endTime;
exports.endorsee = schema$1.endorsee;
exports.entertainmentBusiness = schema$1.entertainmentBusiness;
exports.episode = schema$1.episode;
exports.episodeNumber = schema$1.episodeNumber;
exports.episodes = schema$1.episodes;
exports.equal = schema$1.equal;
exports.error = schema$1.error;
exports.estimatedCost = schema$1.estimatedCost;
exports.estimatedFlightDuration = schema$1.estimatedFlightDuration;
exports.estimatedSalary = schema$1.estimatedSalary;
exports.event = schema$1.event;
exports.eventStatus = schema$1.eventStatus;
exports.events = schema$1.events;
exports.exampleOfWork = schema$1.exampleOfWork;
exports.executableLibraryName = schema$1.executableLibraryName;
exports.exerciseCourse = schema$1.exerciseCourse;
exports.exifData = schema$1.exifData;
exports.expectedArrivalFrom = schema$1.expectedArrivalFrom;
exports.expectedArrivalUntil = schema$1.expectedArrivalUntil;
exports.expectsAcceptanceOf = schema$1.expectsAcceptanceOf;
exports.experienceRequirements = schema$1.experienceRequirements;
exports.expires = schema$1.expires;
exports.familyName = schema$1.familyName;
exports.fatContent = schema$1.fatContent;
exports.faxNumber = schema$1.faxNumber;
exports.featureList = schema$1.featureList;
exports.feesAndCommissionsSpecification = schema$1.feesAndCommissionsSpecification;
exports.fiberContent = schema$1.fiberContent;
exports.fileFormat = schema$1.fileFormat;
exports.fileSize = schema$1.fileSize;
exports.firstPerformance = schema$1.firstPerformance;
exports.flightDistance = schema$1.flightDistance;
exports.flightNumber = schema$1.flightNumber;
exports.floorSize = schema$1.floorSize;
exports.followee = schema$1.followee;
exports.follows = schema$1.follows;
exports.foodEstablishment = schema$1.foodEstablishment;
exports.foodEvent = schema$1.foodEvent;
exports.founder = schema$1.founder;
exports.founders = schema$1.founders;
exports.foundingDate = schema$1.foundingDate;
exports.foundingLocation = schema$1.foundingLocation;
exports.free = schema$1.free;
exports.fromLocation = schema$1.fromLocation;
exports.fuelConsumption = schema$1.fuelConsumption;
exports.fuelEfficiency = schema$1.fuelEfficiency;
exports.fuelType = schema$1.fuelType;
exports.funder = schema$1.funder;
exports.game = schema$1.game;
exports.gameItem = schema$1.gameItem;
exports.gameLocation = schema$1.gameLocation;
exports.gamePlatform = schema$1.gamePlatform;
exports.gameServer = schema$1.gameServer;
exports.gameTip = schema$1.gameTip;
exports.gender = schema$1.gender;
exports.genre = schema$1.genre;
exports.geo = schema$1.geo;
exports.geoContains = schema$1.geoContains;
exports.geoCoveredBy = schema$1.geoCoveredBy;
exports.geoCovers = schema$1.geoCovers;
exports.geoCrosses = schema$1.geoCrosses;
exports.geoDisjoint = schema$1.geoDisjoint;
exports.geoEquals = schema$1.geoEquals;
exports.geoIntersects = schema$1.geoIntersects;
exports.geoMidpoint = schema$1.geoMidpoint;
exports.geoOverlaps = schema$1.geoOverlaps;
exports.geoRadius = schema$1.geoRadius;
exports.geoTouches = schema$1.geoTouches;
exports.geoWithin = schema$1.geoWithin;
exports.geographicArea = schema$1.geographicArea;
exports.givenName = schema$1.givenName;
exports.globalLocationNumber = schema$1.globalLocationNumber;
exports.grantee = schema$1.grantee;
exports.greater = schema$1.greater;
exports.greaterOrEqual = schema$1.greaterOrEqual;
exports.gtin12 = schema$1.gtin12;
exports.gtin13 = schema$1.gtin13;
exports.gtin14 = schema$1.gtin14;
exports.gtin8 = schema$1.gtin8;
exports.hasBroadcastChannel = schema$1.hasBroadcastChannel;
exports.hasCourseInstance = schema$1.hasCourseInstance;
exports.hasDeliveryMethod = schema$1.hasDeliveryMethod;
exports.hasDigitalDocumentPermission = schema$1.hasDigitalDocumentPermission;
exports.hasMap = schema$1.hasMap;
exports.hasMenu = schema$1.hasMenu;
exports.hasMenuItem = schema$1.hasMenuItem;
exports.hasMenuSection = schema$1.hasMenuSection;
exports.hasOccupation = schema$1.hasOccupation;
exports.hasOfferCatalog = schema$1.hasOfferCatalog;
exports.hasPOS = schema$1.hasPOS;
exports.hasPart = schema$1.hasPart;
exports.headline = schema$1.headline;
exports.height = schema$1.height;
exports.highPrice = schema$1.highPrice;
exports.hiringOrganization = schema$1.hiringOrganization;
exports.homeLocation = schema$1.homeLocation;
exports.homeTeam = schema$1.homeTeam;
exports.honorificPrefix = schema$1.honorificPrefix;
exports.honorificSuffix = schema$1.honorificSuffix;
exports.hostingOrganization = schema$1.hostingOrganization;
exports.hoursAvailable = schema$1.hoursAvailable;
exports.httpMethod = schema$1.httpMethod;
exports.iataCode = schema$1.iataCode;
exports.icaoCode = schema$1.icaoCode;
exports.identifier = schema$1.identifier;
exports.illustrator = schema$1.illustrator;
exports.image = schema$1.image;
exports.inAlbum = schema$1.inAlbum;
exports.inBroadcastLineup = schema$1.inBroadcastLineup;
exports.inLanguage = schema$1.inLanguage;
exports.inPlaylist = schema$1.inPlaylist;
exports.incentiveCompensation = schema$1.incentiveCompensation;
exports.incentives = schema$1.incentives;
exports.includedComposition = schema$1.includedComposition;
exports.includedDataCatalog = schema$1.includedDataCatalog;
exports.includedInDataCatalog = schema$1.includedInDataCatalog;
exports.includesObject = schema$1.includesObject;
exports.industry = schema$1.industry;
exports.ineligibleRegion = schema$1.ineligibleRegion;
exports.ingredients = schema$1.ingredients;
exports.installUrl = schema$1.installUrl;
exports.instructor = schema$1.instructor;
exports.instrument = schema$1.instrument;
exports.interactionCount = schema$1.interactionCount;
exports.interactionService = schema$1.interactionService;
exports.interactionStatistic = schema$1.interactionStatistic;
exports.interactionType = schema$1.interactionType;
exports.interactivityType = schema$1.interactivityType;
exports.interestRate = schema$1.interestRate;
exports.inventoryLevel = schema$1.inventoryLevel;
exports.isAccessibleForFree = schema$1.isAccessibleForFree;
exports.isAccessoryOrSparePartFor = schema$1.isAccessoryOrSparePartFor;
exports.isBasedOn = schema$1.isBasedOn;
exports.isBasedOnUrl = schema$1.isBasedOnUrl;
exports.isConsumableFor = schema$1.isConsumableFor;
exports.isFamilyFriendly = schema$1.isFamilyFriendly;
exports.isGift = schema$1.isGift;
exports.isLiveBroadcast = schema$1.isLiveBroadcast;
exports.isPartOf = schema$1.isPartOf;
exports.isRelatedTo = schema$1.isRelatedTo;
exports.isSimilarTo = schema$1.isSimilarTo;
exports.isVariantOf = schema$1.isVariantOf;
exports.isbn = schema$1.isbn;
exports.isicV4 = schema$1.isicV4;
exports.isrcCode = schema$1.isrcCode;
exports.issn = schema$1.issn;
exports.issueNumber = schema$1.issueNumber;
exports.issuedBy = schema$1.issuedBy;
exports.issuedThrough = schema$1.issuedThrough;
exports.iswcCode = schema$1.iswcCode;
exports.item = schema$1.item;
exports.itemCondition = schema$1.itemCondition;
exports.itemListElement = schema$1.itemListElement;
exports.itemListOrder = schema$1.itemListOrder;
exports.itemOffered = schema$1.itemOffered;
exports.itemReviewed = schema$1.itemReviewed;
exports.itemShipped = schema$1.itemShipped;
exports.jobBenefits = schema$1.jobBenefits;
exports.jobLocation = schema$1.jobLocation;
exports.keywords = schema$1.keywords;
exports.knownVehicleDamages = schema$1.knownVehicleDamages;
exports.knows = schema$1.knows;
exports.landlord = schema$1.landlord;
exports.language = schema$1.language;
exports.lastReviewed = schema$1.lastReviewed;
exports.latitude = schema$1.latitude;
exports.learningResourceType = schema$1.learningResourceType;
exports.legalName = schema$1.legalName;
exports.leiCode = schema$1.leiCode;
exports.lender = schema$1.lender;
exports.lesser = schema$1.lesser;
exports.lesserOrEqual = schema$1.lesserOrEqual;
exports.license = schema$1.license;
exports.line = schema$1.line;
exports.liveBlogUpdate = schema$1.liveBlogUpdate;
exports.loanTerm = schema$1.loanTerm;
exports.location = schema$1.location;
exports.locationCreated = schema$1.locationCreated;
exports.lodgingUnitDescription = schema$1.lodgingUnitDescription;
exports.lodgingUnitType = schema$1.lodgingUnitType;
exports.logo = schema$1.logo;
exports.longitude = schema$1.longitude;
exports.loser = schema$1.loser;
exports.lowPrice = schema$1.lowPrice;
exports.lyricist = schema$1.lyricist;
exports.lyrics = schema$1.lyrics;
exports.mainContentOfPage = schema$1.mainContentOfPage;
exports.mainEntity = schema$1.mainEntity;
exports.mainEntityOfPage = schema$1.mainEntityOfPage;
exports.makesOffer = schema$1.makesOffer;
exports.manufacturer = schema$1.manufacturer;
exports.map = schema$1.map;
exports.mapType = schema$1.mapType;
exports.maps = schema$1.maps;
exports.material = schema$1.material;
exports.maxPrice = schema$1.maxPrice;
exports.maxValue = schema$1.maxValue;
exports.maximumAttendeeCapacity = schema$1.maximumAttendeeCapacity;
exports.mealService = schema$1.mealService;
exports.median = schema$1.median;
exports.member = schema$1.member;
exports.memberOf = schema$1.memberOf;
exports.members = schema$1.members;
exports.membershipNumber = schema$1.membershipNumber;
exports.memoryRequirements = schema$1.memoryRequirements;
exports.mentions = schema$1.mentions;
exports.menu = schema$1.menu;
exports.menuAddOn = schema$1.menuAddOn;
exports.merchant = schema$1.merchant;
exports.messageAttachment = schema$1.messageAttachment;
exports.mileageFromOdometer = schema$1.mileageFromOdometer;
exports.minPrice = schema$1.minPrice;
exports.minValue = schema$1.minValue;
exports.minimumPaymentDue = schema$1.minimumPaymentDue;
exports.model = schema$1.model;
exports.modifiedTime = schema$1.modifiedTime;
exports.mpn = schema$1.mpn;
exports.multipleValues = schema$1.multipleValues;
exports.musicArrangement = schema$1.musicArrangement;
exports.musicBy = schema$1.musicBy;
exports.musicCompositionForm = schema$1.musicCompositionForm;
exports.musicGroupMember = schema$1.musicGroupMember;
exports.musicReleaseFormat = schema$1.musicReleaseFormat;
exports.musicalKey = schema$1.musicalKey;
exports.naics = schema$1.naics;
exports.name = schema$1.name;
exports.namedPosition = schema$1.namedPosition;
exports.nationality = schema$1.nationality;
exports.netWorth = schema$1.netWorth;
exports.nextItem = schema$1.nextItem;
exports.nonEqual = schema$1.nonEqual;
exports.numAdults = schema$1.numAdults;
exports.numChildren = schema$1.numChildren;
exports.numTracks = schema$1.numTracks;
exports.numberOfAirbags = schema$1.numberOfAirbags;
exports.numberOfAxles = schema$1.numberOfAxles;
exports.numberOfBeds = schema$1.numberOfBeds;
exports.numberOfDoors = schema$1.numberOfDoors;
exports.numberOfEmployees = schema$1.numberOfEmployees;
exports.numberOfEpisodes = schema$1.numberOfEpisodes;
exports.numberOfForwardGears = schema$1.numberOfForwardGears;
exports.numberOfItems = schema$1.numberOfItems;
exports.numberOfPages = schema$1.numberOfPages;
exports.numberOfPlayers = schema$1.numberOfPlayers;
exports.numberOfPreviousOwners = schema$1.numberOfPreviousOwners;
exports.numberOfRooms = schema$1.numberOfRooms;
exports.numberOfSeasons = schema$1.numberOfSeasons;
exports.numberedPosition = schema$1.numberedPosition;
exports.nutrition = schema$1.nutrition;
exports.object = schema$1.object;
exports.occupancy = schema$1.occupancy;
exports.occupationLocation = schema$1.occupationLocation;
exports.occupationalCategory = schema$1.occupationalCategory;
exports.offerCount = schema$1.offerCount;
exports.offeredBy = schema$1.offeredBy;
exports.offers = schema$1.offers;
exports.openingHours = schema$1.openingHours;
exports.openingHoursSpecification = schema$1.openingHoursSpecification;
exports.opens = schema$1.opens;
exports.operatingSystem = schema$1.operatingSystem;
exports.opponent = schema$1.opponent;
exports.option = schema$1.option;
exports.orderDate = schema$1.orderDate;
exports.orderDelivery = schema$1.orderDelivery;
exports.orderItemNumber = schema$1.orderItemNumber;
exports.orderItemStatus = schema$1.orderItemStatus;
exports.orderNumber = schema$1.orderNumber;
exports.orderQuantity = schema$1.orderQuantity;
exports.orderStatus = schema$1.orderStatus;
exports.orderedItem = schema$1.orderedItem;
exports.organizer = schema$1.organizer;
exports.originAddress = schema$1.originAddress;
exports.ownedFrom = schema$1.ownedFrom;
exports.ownedThrough = schema$1.ownedThrough;
exports.owns = schema$1.owns;
exports.pageEnd = schema$1.pageEnd;
exports.pageStart = schema$1.pageStart;
exports.pagination = schema$1.pagination;
exports.parent = schema$1.parent;
exports.parentItem = schema$1.parentItem;
exports.parentOrganization = schema$1.parentOrganization;
exports.parentService = schema$1.parentService;
exports.parents = schema$1.parents;
exports.partOfEpisode = schema$1.partOfEpisode;
exports.partOfInvoice = schema$1.partOfInvoice;
exports.partOfOrder = schema$1.partOfOrder;
exports.partOfSeason = schema$1.partOfSeason;
exports.partOfSeries = schema$1.partOfSeries;
exports.partOfTVSeries = schema$1.partOfTVSeries;
exports.participant = schema$1.participant;
exports.partySize = schema$1.partySize;
exports.passengerPriorityStatus = schema$1.passengerPriorityStatus;
exports.passengerSequenceNumber = schema$1.passengerSequenceNumber;
exports.paymentAccepted = schema$1.paymentAccepted;
exports.paymentDue = schema$1.paymentDue;
exports.paymentDueDate = schema$1.paymentDueDate;
exports.paymentMethod = schema$1.paymentMethod;
exports.paymentMethodId = schema$1.paymentMethodId;
exports.paymentStatus = schema$1.paymentStatus;
exports.paymentUrl = schema$1.paymentUrl;
exports.percentile10 = schema$1.percentile10;
exports.percentile25 = schema$1.percentile25;
exports.percentile75 = schema$1.percentile75;
exports.percentile90 = schema$1.percentile90;
exports.performTime = schema$1.performTime;
exports.performer = schema$1.performer;
exports.performerIn = schema$1.performerIn;
exports.performers = schema$1.performers;
exports.permissionType = schema$1.permissionType;
exports.permissions = schema$1.permissions;
exports.permitAudience = schema$1.permitAudience;
exports.permittedUsage = schema$1.permittedUsage;
exports.petsAllowed = schema$1.petsAllowed;
exports.photo = schema$1.photo;
exports.photos = schema$1.photos;
exports.pickupLocation = schema$1.pickupLocation;
exports.pickupTime = schema$1.pickupTime;
exports.playMode = schema$1.playMode;
exports.playerType = schema$1.playerType;
exports.playersOnline = schema$1.playersOnline;
exports.polygon = schema$1.polygon;
exports.position = schema$1.position;
exports.postOfficeBoxNumber = schema$1.postOfficeBoxNumber;
exports.postalCode = schema$1.postalCode;
exports.potentialAction = schema$1.potentialAction;
exports.predecessorOf = schema$1.predecessorOf;
exports.prepTime = schema$1.prepTime;
exports.previousItem = schema$1.previousItem;
exports.previousStartDate = schema$1.previousStartDate;
exports.price = schema$1.price;
exports.priceComponent = schema$1.priceComponent;
exports.priceCurrency = schema$1.priceCurrency;
exports.priceRange = schema$1.priceRange;
exports.priceSpecification = schema$1.priceSpecification;
exports.priceType = schema$1.priceType;
exports.priceValidUntil = schema$1.priceValidUntil;
exports.primaryImageOfPage = schema$1.primaryImageOfPage;
exports.printColumn = schema$1.printColumn;
exports.printEdition = schema$1.printEdition;
exports.printPage = schema$1.printPage;
exports.printSection = schema$1.printSection;
exports.processingTime = schema$1.processingTime;
exports.processorRequirements = schema$1.processorRequirements;
exports.producer = schema$1.producer;
exports.produces = schema$1.produces;
exports.productID = schema$1.productID;
exports.productSupported = schema$1.productSupported;
exports.productionCompany = schema$1.productionCompany;
exports.productionDate = schema$1.productionDate;
exports.proficiencyLevel = schema$1.proficiencyLevel;
exports.programMembershipUsed = schema$1.programMembershipUsed;
exports.programName = schema$1.programName;
exports.programmingLanguage = schema$1.programmingLanguage;
exports.programmingModel = schema$1.programmingModel;
exports.propertyID = schema$1.propertyID;
exports.proteinContent = schema$1.proteinContent;
exports.provider = schema$1.provider;
exports.providerMobility = schema$1.providerMobility;
exports.providesBroadcastService = schema$1.providesBroadcastService;
exports.providesService = schema$1.providesService;
exports.publicAccess = schema$1.publicAccess;
exports.publication = schema$1.publication;
exports.publishedOn = schema$1.publishedOn;
exports.publisher = schema$1.publisher;
exports.publishingPrinciples = schema$1.publishingPrinciples;
exports.purchaseDate = schema$1.purchaseDate;
exports.query = schema$1.query;
exports.quest = schema$1.quest;
exports.question = schema$1.question;
exports.ratingCount = schema$1.ratingCount;
exports.ratingValue = schema$1.ratingValue;
exports.readonlyValue = schema$1.readonlyValue;
exports.realEstateAgent = schema$1.realEstateAgent;
exports.recipe = schema$1.recipe;
exports.recipeCategory = schema$1.recipeCategory;
exports.recipeCuisine = schema$1.recipeCuisine;
exports.recipeIngredient = schema$1.recipeIngredient;
exports.recipeInstructions = schema$1.recipeInstructions;
exports.recipeYield = schema$1.recipeYield;
exports.recipient = schema$1.recipient;
exports.recordLabel = schema$1.recordLabel;
exports.recordedAs = schema$1.recordedAs;
exports.recordedAt = schema$1.recordedAt;
exports.recordedIn = schema$1.recordedIn;
exports.recordingOf = schema$1.recordingOf;
exports.referenceQuantity = schema$1.referenceQuantity;
exports.referencesOrder = schema$1.referencesOrder;
exports.regionsAllowed = schema$1.regionsAllowed;
exports.relatedLink = schema$1.relatedLink;
exports.relatedTo = schema$1.relatedTo;
exports.releaseDate = schema$1.releaseDate;
exports.releaseNotes = schema$1.releaseNotes;
exports.releaseOf = schema$1.releaseOf;
exports.releasedEvent = schema$1.releasedEvent;
exports.relevantOccupation = schema$1.relevantOccupation;
exports.remainingAttendeeCapacity = schema$1.remainingAttendeeCapacity;
exports.replacee = schema$1.replacee;
exports.replacer = schema$1.replacer;
exports.replyToUrl = schema$1.replyToUrl;
exports.reportNumber = schema$1.reportNumber;
exports.representativeOfPage = schema$1.representativeOfPage;
exports.requiredCollateral = schema$1.requiredCollateral;
exports.requiredGender = schema$1.requiredGender;
exports.requiredMaxAge = schema$1.requiredMaxAge;
exports.requiredMinAge = schema$1.requiredMinAge;
exports.requiredQuantity = schema$1.requiredQuantity;
exports.requirements = schema$1.requirements;
exports.requiresSubscription = schema$1.requiresSubscription;
exports.reservationFor = schema$1.reservationFor;
exports.reservationId = schema$1.reservationId;
exports.reservationStatus = schema$1.reservationStatus;
exports.reservedTicket = schema$1.reservedTicket;
exports.responsibilities = schema$1.responsibilities;
exports.result = schema$1.result;
exports.resultComment = schema$1.resultComment;
exports.resultReview = schema$1.resultReview;
exports.review = schema$1.review;
exports.reviewAspect = schema$1.reviewAspect;
exports.reviewBody = schema$1.reviewBody;
exports.reviewCount = schema$1.reviewCount;
exports.reviewRating = schema$1.reviewRating;
exports.reviewedBy = schema$1.reviewedBy;
exports.reviews = schema$1.reviews;
exports.roleName = schema$1.roleName;
exports.rsvpResponse = schema$1.rsvpResponse;
exports.runtime = schema$1.runtime;
exports.runtimePlatform = schema$1.runtimePlatform;
exports.salaryCurrency = schema$1.salaryCurrency;
exports.sameAs = schema$1.sameAs;
exports.sampleType = schema$1.sampleType;
exports.saturatedFatContent = schema$1.saturatedFatContent;
exports.scheduledPaymentDate = schema$1.scheduledPaymentDate;
exports.scheduledTime = schema$1.scheduledTime;
exports.schemaVersion = schema$1.schemaVersion;
exports.screenCount = schema$1.screenCount;
exports.screenshot = schema$1.screenshot;
exports.season = schema$1.season;
exports.seasonNumber = schema$1.seasonNumber;
exports.seasons = schema$1.seasons;
exports.seatNumber = schema$1.seatNumber;
exports.seatRow = schema$1.seatRow;
exports.seatSection = schema$1.seatSection;
exports.seatingType = schema$1.seatingType;
exports.securityScreening = schema$1.securityScreening;
exports.seeks = schema$1.seeks;
exports.seller = schema$1.seller;
exports.sender = schema$1.sender;
exports.serialNumber = schema$1.serialNumber;
exports.serverStatus = schema$1.serverStatus;
exports.servesCuisine = schema$1.servesCuisine;
exports.serviceArea = schema$1.serviceArea;
exports.serviceAudience = schema$1.serviceAudience;
exports.serviceLocation = schema$1.serviceLocation;
exports.serviceOperator = schema$1.serviceOperator;
exports.serviceOutput = schema$1.serviceOutput;
exports.servicePhone = schema$1.servicePhone;
exports.servicePostalAddress = schema$1.servicePostalAddress;
exports.serviceSmsNumber = schema$1.serviceSmsNumber;
exports.serviceType = schema$1.serviceType;
exports.serviceUrl = schema$1.serviceUrl;
exports.servingSize = schema$1.servingSize;
exports.sharedContent = schema$1.sharedContent;
exports.sibling = schema$1.sibling;
exports.siblings = schema$1.siblings;
exports.significantLink = schema$1.significantLink;
exports.significantLinks = schema$1.significantLinks;
exports.skills = schema$1.skills;
exports.sku = schema$1.sku;
exports.slogan = schema$1.slogan;
exports.smokingAllowed = schema$1.smokingAllowed;
exports.sodiumContent = schema$1.sodiumContent;
exports.softwareAddOn = schema$1.softwareAddOn;
exports.softwareHelp = schema$1.softwareHelp;
exports.softwareRequirements = schema$1.softwareRequirements;
exports.softwareVersion = schema$1.softwareVersion;
exports.sourceOrganization = schema$1.sourceOrganization;
exports.spatial = schema$1.spatial;
exports.spatialCoverage = schema$1.spatialCoverage;
exports.speakable = schema$1.speakable;
exports.specialCommitments = schema$1.specialCommitments;
exports.specialOpeningHoursSpecification = schema$1.specialOpeningHoursSpecification;
exports.specialty = schema$1.specialty;
exports.sponsor = schema$1.sponsor;
exports.sport = schema$1.sport;
exports.sportsActivityLocation = schema$1.sportsActivityLocation;
exports.sportsEvent = schema$1.sportsEvent;
exports.sportsTeam = schema$1.sportsTeam;
exports.spouse = schema$1.spouse;
exports.starRating = schema$1.starRating;
exports.startDate = schema$1.startDate;
exports.startTime = schema$1.startTime;
exports.steeringPosition = schema$1.steeringPosition;
exports.step = schema$1.step;
exports.stepValue = schema$1.stepValue;
exports.steps = schema$1.steps;
exports.storageRequirements = schema$1.storageRequirements;
exports.streetAddress = schema$1.streetAddress;
exports.subEvent = schema$1.subEvent;
exports.subEvents = schema$1.subEvents;
exports.subOrganization = schema$1.subOrganization;
exports.subReservation = schema$1.subReservation;
exports.subjectOf = schema$1.subjectOf;
exports.subtitleLanguage = schema$1.subtitleLanguage;
exports.successorOf = schema$1.successorOf;
exports.sugarContent = schema$1.sugarContent;
exports.suggestedAnswer = schema$1.suggestedAnswer;
exports.suggestedGender = schema$1.suggestedGender;
exports.suggestedMaxAge = schema$1.suggestedMaxAge;
exports.suggestedMinAge = schema$1.suggestedMinAge;
exports.suitableForDiet = schema$1.suitableForDiet;
exports.superEvent = schema$1.superEvent;
exports.supply = schema$1.supply;
exports.supportingData = schema$1.supportingData;
exports.surface = schema$1.surface;
exports.target = schema$1.target;
exports.targetCollection = schema$1.targetCollection;
exports.targetDescription = schema$1.targetDescription;
exports.targetName = schema$1.targetName;
exports.targetPlatform = schema$1.targetPlatform;
exports.targetProduct = schema$1.targetProduct;
exports.targetUrl = schema$1.targetUrl;
exports.taxID = schema$1.taxID;
exports.telephone = schema$1.telephone;
exports.temporal = schema$1.temporal;
exports.temporalCoverage = schema$1.temporalCoverage;
exports.text = schema$1.text;
exports.thumbnail = schema$1.thumbnail;
exports.thumbnailUrl = schema$1.thumbnailUrl;
exports.tickerSymbol = schema$1.tickerSymbol;
exports.ticketNumber = schema$1.ticketNumber;
exports.ticketToken = schema$1.ticketToken;
exports.ticketedSeat = schema$1.ticketedSeat;
exports.timeRequired = schema$1.timeRequired;
exports.title = schema$1.title;
exports.toLocation = schema$1.toLocation;
exports.toRecipient = schema$1.toRecipient;
exports.tool = schema$1.tool;
exports.totalPaymentDue = schema$1.totalPaymentDue;
exports.totalPrice = schema$1.totalPrice;
exports.totalTime = schema$1.totalTime;
exports.touristType = schema$1.touristType;
exports.track = schema$1.track;
exports.trackingNumber = schema$1.trackingNumber;
exports.trackingUrl = schema$1.trackingUrl;
exports.tracks = schema$1.tracks;
exports.trailer = schema$1.trailer;
exports.trainName = schema$1.trainName;
exports.trainNumber = schema$1.trainNumber;
exports.transFatContent = schema$1.transFatContent;
exports.transcript = schema$1.transcript;
exports.translator = schema$1.translator;
exports.typeOfBed = schema$1.typeOfBed;
exports.typeOfGood = schema$1.typeOfGood;
exports.typicalAgeRange = schema$1.typicalAgeRange;
exports.underName = schema$1.underName;
exports.unitCode = schema$1.unitCode;
exports.unitText = schema$1.unitText;
exports.unsaturatedFatContent = schema$1.unsaturatedFatContent;
exports.uploadDate = schema$1.uploadDate;
exports.upvoteCount = schema$1.upvoteCount;
exports.url = schema$1.url;
exports.urlTemplate = schema$1.urlTemplate;
exports.userInteractionCount = schema$1.userInteractionCount;
exports.validFor = schema$1.validFor;
exports.validFrom = schema$1.validFrom;
exports.validIn = schema$1.validIn;
exports.validThrough = schema$1.validThrough;
exports.validUntil = schema$1.validUntil;
exports.value = schema$1.value;
exports.valueAddedTaxIncluded = schema$1.valueAddedTaxIncluded;
exports.valueMaxLength = schema$1.valueMaxLength;
exports.valueMinLength = schema$1.valueMinLength;
exports.valueName = schema$1.valueName;
exports.valuePattern = schema$1.valuePattern;
exports.valueReference = schema$1.valueReference;
exports.valueRequired = schema$1.valueRequired;
exports.vatID = schema$1.vatID;
exports.vehicleConfiguration = schema$1.vehicleConfiguration;
exports.vehicleEngine = schema$1.vehicleEngine;
exports.vehicleIdentificationNumber = schema$1.vehicleIdentificationNumber;
exports.vehicleInteriorColor = schema$1.vehicleInteriorColor;
exports.vehicleInteriorType = schema$1.vehicleInteriorType;
exports.vehicleModelDate = schema$1.vehicleModelDate;
exports.vehicleSeatingCapacity = schema$1.vehicleSeatingCapacity;
exports.vehicleTransmission = schema$1.vehicleTransmission;
exports.vendor = schema$1.vendor;
exports.version = schema$1.version;
exports.video = schema$1.video;
exports.videoFormat = schema$1.videoFormat;
exports.videoFrameSize = schema$1.videoFrameSize;
exports.videoQuality = schema$1.videoQuality;
exports.volumeNumber = schema$1.volumeNumber;
exports.warranty = schema$1.warranty;
exports.warrantyPromise = schema$1.warrantyPromise;
exports.warrantyScope = schema$1.warrantyScope;
exports.webCheckinTime = schema$1.webCheckinTime;
exports.weight = schema$1.weight;
exports.width = schema$1.width;
exports.winner = schema$1.winner;
exports.wordCount = schema$1.wordCount;
exports.workExample = schema$1.workExample;
exports.workFeatured = schema$1.workFeatured;
exports.workHours = schema$1.workHours;
exports.workLocation = schema$1.workLocation;
exports.workPerformed = schema$1.workPerformed;
exports.workPresented = schema$1.workPresented;
exports.worksFor = schema$1.worksFor;
exports.worstRating = schema$1.worstRating;
exports.xpath = schema$1.xpath;
exports.yearlyRevenue = schema$1.yearlyRevenue;
exports.yearsInOperation = schema$1.yearsInOperation;
exports.yield__workaround = schema$1.yield__workaround;
