// rdf-namespaces/dist/sched-bf21c384.js
'use strict';

/**
 *
 *
 *
 *
 * http://www.w3.org/ns/pim/schedule#availabilityOptions
 */
var availabilityOptions = 'http://www.w3.org/ns/pim/schedule#availabilityOptions';
/**
 * availability
 *
 *
 *
 * http://www.w3.org/ns/pim/schedule#availibility
 */
var availibility = 'http://www.w3.org/ns/pim/schedule#availibility';
/**
 * possible time for event
 *
 *
 *
 * http://www.w3.org/ns/pim/schedule#option
 */
var option = 'http://www.w3.org/ns/pim/schedule#option';
/**
 * response
 *
 *
 *
 * http://www.w3.org/ns/pim/schedule#response
 */
var response = 'http://www.w3.org/ns/pim/schedule#response';
/**
 * Degrees of difficulty
 *
 * Provides users of a poll with a way to reflect the difficulty, pain
 *     or disruption which would be involved in a given choice.
 *
 * http://www.w3.org/ns/pim/schedule#DegreesOfDifficulty
 */
var DegreesOfDifficulty = 'http://www.w3.org/ns/pim/schedule#DegreesOfDifficulty';
/**
 * difficult
 *
 * I would rather a different choice.
 *
 * http://www.w3.org/ns/pim/schedule#Difficult
 */
var Difficult = 'http://www.w3.org/ns/pim/schedule#Difficult';
/**
 * ideal
 *
 * This would be perfect for me.
 *
 * http://www.w3.org/ns/pim/schedule#Ideal
 */
var Ideal = 'http://www.w3.org/ns/pim/schedule#Ideal';
/**
 * impossible
 *
 * I really cannot do this.
 *
 * http://www.w3.org/ns/pim/schedule#Impossible
 */
var Impossible = 'http://www.w3.org/ns/pim/schedule#Impossible';
/**
 * painfull
 *
 * This would cause acute disruption
 *
 * http://www.w3.org/ns/pim/schedule#Painful
 */
var Painful = 'http://www.w3.org/ns/pim/schedule#Painful';
/**
 * tolerable
 *
 * I can do this, not ideal.
 *
 * http://www.w3.org/ns/pim/schedule#Tolerable
 */
var Tolerable = 'http://www.w3.org/ns/pim/schedule#Tolerable';
/**
 * Yes or No
 *
 * Provides users of a poll with the simplest choice.
 *
 * http://www.w3.org/ns/pim/schedule#YesNo
 */
var YesNo = 'http://www.w3.org/ns/pim/schedule#YesNo';
/**
 * Yes, No or Maybe
 *
 * Provides users of a poll with a little uncertainty.
 *
 * http://www.w3.org/ns/pim/schedule#YesNoMaybe
 */
var YesNoMaybe = 'http://www.w3.org/ns/pim/schedule#YesNoMaybe';

var schedImport = /*#__PURE__*/Object.freeze({
	availabilityOptions: availabilityOptions,
	availibility: availibility,
	option: option,
	response: response,
	DegreesOfDifficulty: DegreesOfDifficulty,
	Difficult: Difficult,
	Ideal: Ideal,
	Impossible: Impossible,
	Painful: Painful,
	Tolerable: Tolerable,
	YesNo: YesNo,
	YesNoMaybe: YesNoMaybe
});

exports.DegreesOfDifficulty = DegreesOfDifficulty;
exports.Difficult = Difficult;
exports.Ideal = Ideal;
exports.Impossible = Impossible;
exports.Painful = Painful;
exports.Tolerable = Tolerable;
exports.YesNo = YesNo;
exports.YesNoMaybe = YesNoMaybe;
exports.availabilityOptions = availabilityOptions;
exports.availibility = availibility;
exports.option = option;
exports.response = response;
exports.schedImport = schedImport;
