// rdf-namespaces/dist/sioc.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var sioc$1 = require('./sioc-cc954753.js');



exports.Community = sioc$1.Community;
exports.Container = sioc$1.Container;
exports.Forum = sioc$1.Forum;
exports.Item = sioc$1.Item;
exports.Post = sioc$1.Post;
exports.Role = sioc$1.Role;
exports.Site = sioc$1.Site;
exports.Space = sioc$1.Space;
exports.Thread = sioc$1.Thread;
exports.UserAccount = sioc$1.UserAccount;
exports.Usergroup = sioc$1.Usergroup;
exports.about = sioc$1.about;
exports.account_of = sioc$1.account_of;
exports.addressed_to = sioc$1.addressed_to;
exports.administrator_of = sioc$1.administrator_of;
exports.attachment = sioc$1.attachment;
exports.avatar = sioc$1.avatar;
exports.container_of = sioc$1.container_of;
exports.content = sioc$1.content;
exports.content_encoded = sioc$1.content_encoded;
exports.created_at = sioc$1.created_at;
exports.creator_of = sioc$1.creator_of;
exports.delivered_at = sioc$1.delivered_at;
exports.description = sioc$1.description;
exports.discussion_of = sioc$1.discussion_of;
exports.email = sioc$1.email;
exports.email_sha1 = sioc$1.email_sha1;
exports.embeds_knowledge = sioc$1.embeds_knowledge;
exports.feed = sioc$1.feed;
exports.first_name = sioc$1.first_name;
exports.follows = sioc$1.follows;
exports.function_of = sioc$1.function_of;
exports.generator = sioc$1.generator;
exports.group_of = sioc$1.group_of;
exports.has_administrator = sioc$1.has_administrator;
exports.has_container = sioc$1.has_container;
exports.has_creator = sioc$1.has_creator;
exports.has_discussion = sioc$1.has_discussion;
exports.has_function = sioc$1.has_function;
exports.has_group = sioc$1.has_group;
exports.has_host = sioc$1.has_host;
exports.has_member = sioc$1.has_member;
exports.has_moderator = sioc$1.has_moderator;
exports.has_modifier = sioc$1.has_modifier;
exports.has_owner = sioc$1.has_owner;
exports.has_parent = sioc$1.has_parent;
exports.has_part = sioc$1.has_part;
exports.has_reply = sioc$1.has_reply;
exports.has_scope = sioc$1.has_scope;
exports.has_space = sioc$1.has_space;
exports.has_subscriber = sioc$1.has_subscriber;
exports.has_usergroup = sioc$1.has_usergroup;
exports.host_of = sioc$1.host_of;
exports.id = sioc$1.id;
exports.ip_address = sioc$1.ip_address;
exports.last_activity_date = sioc$1.last_activity_date;
exports.last_item_date = sioc$1.last_item_date;
exports.last_name = sioc$1.last_name;
exports.last_reply_date = sioc$1.last_reply_date;
exports.latest_version = sioc$1.latest_version;
exports.likes = sioc$1.likes;
exports.link = sioc$1.link;
exports.links_to = sioc$1.links_to;
exports.member_of = sioc$1.member_of;
exports.mentions = sioc$1.mentions;
exports.moderator_of = sioc$1.moderator_of;
exports.modified_at = sioc$1.modified_at;
exports.modifier_of = sioc$1.modifier_of;
exports.name = sioc$1.name;
exports.next_by_date = sioc$1.next_by_date;
exports.next_version = sioc$1.next_version;
exports.note = sioc$1.note;
exports.num_authors = sioc$1.num_authors;
exports.num_items = sioc$1.num_items;
exports.num_replies = sioc$1.num_replies;
exports.num_threads = sioc$1.num_threads;
exports.num_views = sioc$1.num_views;
exports.owner_of = sioc$1.owner_of;
exports.parent_of = sioc$1.parent_of;
exports.part_of = sioc$1.part_of;
exports.previous_by_date = sioc$1.previous_by_date;
exports.previous_version = sioc$1.previous_version;
exports.read_at = sioc$1.read_at;
exports.reference = sioc$1.reference;
exports.related_to = sioc$1.related_to;
exports.reply_of = sioc$1.reply_of;
exports.respond_to = sioc$1.respond_to;
exports.scope_of = sioc$1.scope_of;
exports.shared_by = sioc$1.shared_by;
exports.space_of = sioc$1.space_of;
exports.subject = sioc$1.subject;
exports.subscriber_of = sioc$1.subscriber_of;
exports.title = sioc$1.title;
exports.topic = sioc$1.topic;
exports.usergroup_of = sioc$1.usergroup_of;
