// rdf-namespaces/dist/sioc.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * Community
 * 
 * Community is a high-level concept that defines an online community and what it consists of.
 *
 * http://rdfs.org/sioc/ns#Community
 */
export const Community: OwlClass = 'http://rdfs.org/sioc/ns#Community';

/**
 * Container
 * 
 * An area in which content Items are contained.
 *
 * http://rdfs.org/sioc/ns#Container
 */
export const Container: OwlClass = 'http://rdfs.org/sioc/ns#Container';

/**
 * Forum
 * 
 * A discussion area on which Posts or entries are made.
 *
 * http://rdfs.org/sioc/ns#Forum
 */
export const Forum: OwlClass = 'http://rdfs.org/sioc/ns#Forum';

/**
 * Item
 * 
 * An Item is something which can be in a Container.
 *
 * http://rdfs.org/sioc/ns#Item
 */
export const Item: OwlClass = 'http://rdfs.org/sioc/ns#Item';

/**
 * Post
 * 
 * An article or message that can be posted to a Forum.
 *
 * http://rdfs.org/sioc/ns#Post
 */
export const Post: OwlClass = 'http://rdfs.org/sioc/ns#Post';

/**
 * Role
 * 
 * A Role is a function of a UserAccount within a scope of a particular Forum, Site, etc.
 *
 * http://rdfs.org/sioc/ns#Role
 */
export const Role: OwlClass = 'http://rdfs.org/sioc/ns#Role';

/**
 * Space
 * 
 * A Space is a place where data resides, e.g. on a website, desktop, fileshare, etc.
 *
 * http://rdfs.org/sioc/ns#Space
 */
export const Space: OwlClass = 'http://rdfs.org/sioc/ns#Space';

/**
 * Site
 * 
 * A Site can be the location of an online community or set of communities, with UserAccounts and Usergroups creating Items in a set of Containers. It can be thought of as a web-accessible data Space.
 *
 * http://rdfs.org/sioc/ns#Site
 */
export const Site: OwlClass = 'http://rdfs.org/sioc/ns#Site';

/**
 * Thread
 * 
 * A container for a series of threaded discussion Posts or Items.
 *
 * http://rdfs.org/sioc/ns#Thread
 */
export const Thread: OwlClass = 'http://rdfs.org/sioc/ns#Thread';

/**
 * User Account
 * 
 * A user account in an online community site.
 *
 * http://rdfs.org/sioc/ns#UserAccount
 */
export const UserAccount: OwlClass = 'http://rdfs.org/sioc/ns#UserAccount';

/**
 * Usergroup
 * 
 * A set of UserAccounts whose owners have a common purpose or interest. Can be used for access control purposes.
 *
 * http://rdfs.org/sioc/ns#Usergroup
 */
export const Usergroup: OwlClass = 'http://rdfs.org/sioc/ns#Usergroup';

/**
 * about
 * 
 * Specifies that this Item is about a particular resource, e.g. a Post describing a book, hotel, etc.
 *
 * http://rdfs.org/sioc/ns#about
 */
export const about: OwlObjectProperty = 'http://rdfs.org/sioc/ns#about';

/**
 * account of
 * 
 * Refers to the foaf:Agent or foaf:Person who owns this sioc:UserAccount.
 *
 * http://rdfs.org/sioc/ns#account_of
 */
export const account_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#account_of';

/**
 * addressed to
 * 
 * Refers to who (e.g. a UserAccount, e-mail address, etc.) a particular Item is addressed to.
 *
 * http://rdfs.org/sioc/ns#addressed_to
 */
export const addressed_to: OwlObjectProperty = 'http://rdfs.org/sioc/ns#addressed_to';

/**
 * administrator of
 * 
 * A Site that the UserAccount is an administrator of.
 *
 * http://rdfs.org/sioc/ns#administrator_of
 */
export const administrator_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#administrator_of';

/**
 * attachment
 * 
 * The URI of a file attached to an Item.
 *
 * http://rdfs.org/sioc/ns#attachment
 */
export const attachment: OwlObjectProperty = 'http://rdfs.org/sioc/ns#attachment';

/**
 * avatar
 * 
 * An image or depiction used to represent this UserAccount.
 *
 * http://rdfs.org/sioc/ns#avatar
 */
export const avatar: OwlObjectProperty = 'http://rdfs.org/sioc/ns#avatar';

/**
 * container of
 * 
 * An Item that this Container contains.
 *
 * http://rdfs.org/sioc/ns#container_of
 */
export const container_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#container_of';

/**
 * creator of
 * 
 * A resource that the UserAccount is a creator of.
 *
 * http://rdfs.org/sioc/ns#creator_of
 */
export const creator_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#creator_of';

/**
 * delivered at
 * 
 * When this was delivered, in ISO 8601 format.
 *
 * http://rdfs.org/sioc/ns#delivered_at
 */
export const delivered_at: OwlObjectProperty = 'http://rdfs.org/sioc/ns#delivered_at';

/**
 * discussion of
 * 
 * The Item that this discussion is about.
 *
 * http://rdfs.org/sioc/ns#discussion_of
 */
export const discussion_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#discussion_of';

/**
 * email
 * 
 * An electronic mail address of the UserAccount.
 *
 * http://rdfs.org/sioc/ns#email
 */
export const email: OwlObjectProperty = 'http://rdfs.org/sioc/ns#email';

/**
 * embeds knowledge
 * 
 * This links Items to embedded statements, facts and structured content.
 *
 * http://rdfs.org/sioc/ns#embeds_knowledge
 */
export const embeds_knowledge: OwlObjectProperty = 'http://rdfs.org/sioc/ns#embeds_knowledge';

/**
 * feed
 * 
 * A feed (e.g. RSS, Atom, etc.) pertaining to this resource (e.g. for a Forum, Site, UserAccount, etc.).
 *
 * http://rdfs.org/sioc/ns#feed
 */
export const feed: OwlObjectProperty = 'http://rdfs.org/sioc/ns#feed';

/**
 * follows
 * 
 * Indicates that one UserAccount follows another UserAccount (e.g. for microblog posts or other content item updates).
 *
 * http://rdfs.org/sioc/ns#follows
 */
export const follows: OwlObjectProperty = 'http://rdfs.org/sioc/ns#follows';

/**
 * function of
 * 
 * A UserAccount that has this Role.
 *
 * http://rdfs.org/sioc/ns#function_of
 */
export const function_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#function_of';

/**
 * generator
 * 
 * A URI for the application used to generate this Item.
 *
 * http://rdfs.org/sioc/ns#generator
 */
export const generator: OwlObjectProperty = 'http://rdfs.org/sioc/ns#generator';

/**
 * has administrator
 * 
 * A UserAccount that is an administrator of this Site.
 *
 * http://rdfs.org/sioc/ns#has_administrator
 */
export const has_administrator: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_administrator';

/**
 * has container
 * 
 * The Container to which this Item belongs.
 *
 * http://rdfs.org/sioc/ns#has_container
 */
export const has_container: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_container';

/**
 * has creator
 * 
 * This is the UserAccount that made this resource.
 *
 * http://rdfs.org/sioc/ns#has_creator
 */
export const has_creator: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_creator';

/**
 * has discussion
 * 
 * A discussion that is related to this Item. The discussion can be anything, for example, a sioc:Forum or sioc:Thread, a sioct:WikiArticle or simply a foaf:Document.
 *
 * http://rdfs.org/sioc/ns#has_discussion
 */
export const has_discussion: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_discussion';

/**
 * has function
 * 
 * A Role that this UserAccount has.
 *
 * http://rdfs.org/sioc/ns#has_function
 */
export const has_function: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_function';

/**
 * has host
 * 
 * The Site that hosts this Container.
 *
 * http://rdfs.org/sioc/ns#has_host
 */
export const has_host: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_host';

/**
 * has member
 * 
 * A UserAccount that is a member of this Usergroup.
 *
 * http://rdfs.org/sioc/ns#has_member
 */
export const has_member: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_member';

/**
 * has moderator
 * 
 * A UserAccount that is a moderator of this Forum.
 *
 * http://rdfs.org/sioc/ns#has_moderator
 */
export const has_moderator: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_moderator';

/**
 * has modifier
 * 
 * A UserAccount that modified this resource (e.g. Item, Container, Space).
 *
 * http://rdfs.org/sioc/ns#has_modifier
 */
export const has_modifier: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_modifier';

/**
 * has owner
 * 
 * A UserAccount that this resource is owned by.
 *
 * http://rdfs.org/sioc/ns#has_owner
 */
export const has_owner: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_owner';

/**
 * has parent
 * 
 * A Container or Forum that this Container or Forum is a child of.
 *
 * http://rdfs.org/sioc/ns#has_parent
 */
export const has_parent: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_parent';

/**
 * has reply
 * 
 * Points to an Item or Post that is a reply or response to this Item or Post.
 *
 * http://rdfs.org/sioc/ns#has_reply
 */
export const has_reply: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_reply';

/**
 * has scope
 * 
 * A resource that this Role applies to.
 *
 * http://rdfs.org/sioc/ns#has_scope
 */
export const has_scope: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_scope';

/**
 * has space
 * 
 * A data Space which this resource is a part of.
 *
 * http://rdfs.org/sioc/ns#has_space
 */
export const has_space: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_space';

/**
 * has subscriber
 * 
 * A UserAccount that is subscribed to this Container.
 *
 * http://rdfs.org/sioc/ns#has_subscriber
 */
export const has_subscriber: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_subscriber';

/**
 * has usergroup
 * 
 * Points to a Usergroup that has certain access to this Space.
 *
 * http://rdfs.org/sioc/ns#has_usergroup
 */
export const has_usergroup: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_usergroup';

/**
 * host of
 * 
 * A Container that is hosted on this Site.
 *
 * http://rdfs.org/sioc/ns#host_of
 */
export const host_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#host_of';

/**
 * latest version
 * 
 * Links to the latest revision of this Item or Post.
 *
 * http://rdfs.org/sioc/ns#latest_version
 */
export const latest_version: OwlObjectProperty = 'http://rdfs.org/sioc/ns#latest_version';

/**
 * likes
 * 
 * Used to indicate some form of endorsement by a UserAccount or Agent of an Item, Container, Space, UserAccount, etc.
 *
 * http://rdfs.org/sioc/ns#likes
 */
export const likes: OwlObjectProperty = 'http://rdfs.org/sioc/ns#likes';

/**
 * link
 * 
 * A URI of a document which contains this SIOC object.
 *
 * http://rdfs.org/sioc/ns#link
 */
export const link: OwlObjectProperty = 'http://rdfs.org/sioc/ns#link';

/**
 * links to
 * 
 * Links extracted from hyperlinks within a SIOC concept, e.g. Post or Site.
 *
 * http://rdfs.org/sioc/ns#links_to
 */
export const links_to: OwlObjectProperty = 'http://rdfs.org/sioc/ns#links_to';

/**
 * member of
 * 
 * A Usergroup that this UserAccount is a member of.
 *
 * http://rdfs.org/sioc/ns#member_of
 */
export const member_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#member_of';

/**
 * mentions
 * 
 * Refers to a UserAccount that a particular Item mentions.
 *
 * http://rdfs.org/sioc/ns#mentions
 */
export const mentions: OwlObjectProperty = 'http://rdfs.org/sioc/ns#mentions';

/**
 * moderator of
 * 
 * A Forum that a UserAccount is a moderator of.
 *
 * http://rdfs.org/sioc/ns#moderator_of
 */
export const moderator_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#moderator_of';

/**
 * modifier of
 * 
 * A resource that this UserAccount has modified.
 *
 * http://rdfs.org/sioc/ns#modifier_of
 */
export const modifier_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#modifier_of';

/**
 * next by date
 * 
 * Next Item or Post in a given Container sorted by date.
 *
 * http://rdfs.org/sioc/ns#next_by_date
 */
export const next_by_date: OwlObjectProperty = 'http://rdfs.org/sioc/ns#next_by_date';

/**
 * next version
 * 
 * Links to the next revision of this Item or Post.
 *
 * http://rdfs.org/sioc/ns#next_version
 */
export const next_version: OwlObjectProperty = 'http://rdfs.org/sioc/ns#next_version';

/**
 * owner of
 * 
 * A resource owned by a particular UserAccount, for example, a weblog or image gallery.
 *
 * http://rdfs.org/sioc/ns#owner_of
 */
export const owner_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#owner_of';

/**
 * parent of
 * 
 * A child Container or Forum that this Container or Forum is a parent of.
 *
 * http://rdfs.org/sioc/ns#parent_of
 */
export const parent_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#parent_of';

/**
 * previous by date
 * 
 * Previous Item or Post in a given Container sorted by date.
 *
 * http://rdfs.org/sioc/ns#previous_by_date
 */
export const previous_by_date: OwlObjectProperty = 'http://rdfs.org/sioc/ns#previous_by_date';

/**
 * previous version
 * 
 * Links to the previous revision of this Item or Post.
 *
 * http://rdfs.org/sioc/ns#previous_version
 */
export const previous_version: OwlObjectProperty = 'http://rdfs.org/sioc/ns#previous_version';

/**
 * read at
 * 
 * When this was read, in ISO 8601 format.
 *
 * http://rdfs.org/sioc/ns#read_at
 */
export const read_at: OwlObjectProperty = 'http://rdfs.org/sioc/ns#read_at';

/**
 * related to
 * 
 * Related resources for this resource, e.g. for Posts, perhaps determined implicitly from topics or references.
 *
 * http://rdfs.org/sioc/ns#related_to
 */
export const related_to: OwlObjectProperty = 'http://rdfs.org/sioc/ns#related_to';

/**
 * reply of
 * 
 * Links to an Item or Post which this Item or Post is a reply to.
 *
 * http://rdfs.org/sioc/ns#reply_of
 */
export const reply_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#reply_of';

/**
 * respond to
 * 
 * For the reply-to address set in email messages, IMs, etc. The property name was chosen to avoid confusion with has_reply/reply_of (the reply graph).
 *
 * http://rdfs.org/sioc/ns#respond_to
 */
export const respond_to: OwlObjectProperty = 'http://rdfs.org/sioc/ns#respond_to';

/**
 * scope of
 * 
 * A Role that has a scope of this resource.
 *
 * http://rdfs.org/sioc/ns#scope_of
 */
export const scope_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#scope_of';

/**
 * shared by
 * 
 * For shared Items where there is a certain creator_of and an intermediary who shares or forwards it (e.g. as a sibling Item).
 *
 * http://rdfs.org/sioc/ns#shared_by
 */
export const shared_by: OwlObjectProperty = 'http://rdfs.org/sioc/ns#shared_by';

/**
 * space of
 * 
 * A resource which belongs to this data Space.
 *
 * http://rdfs.org/sioc/ns#space_of
 */
export const space_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#space_of';

/**
 * subscriber of
 * 
 * A Container that a UserAccount is subscribed to.
 *
 * http://rdfs.org/sioc/ns#subscriber_of
 */
export const subscriber_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#subscriber_of';

/**
 * topic
 * 
 * A topic of interest, linking to the appropriate URI, e.g. in the Open Directory Project or of a SKOS category.
 *
 * http://rdfs.org/sioc/ns#topic
 */
export const topic: OwlObjectProperty = 'http://rdfs.org/sioc/ns#topic';

/**
 * usergroup of
 * 
 * A Space that the Usergroup has access to.
 *
 * http://rdfs.org/sioc/ns#usergroup_of
 */
export const usergroup_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#usergroup_of';

/**
 * group of
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#group_of
 */
export const group_of: string = 'http://rdfs.org/sioc/ns#group_of';

/**
 * has group
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_group
 */
export const has_group: string = 'http://rdfs.org/sioc/ns#has_group';

/**
 * has part
 * 
 * An resource that is a part of this subject.
 *
 * http://rdfs.org/sioc/ns#has_part
 */
export const has_part: string = 'http://rdfs.org/sioc/ns#has_part';

/**
 * part of
 * 
 * A resource that the subject is a part of.
 *
 * http://rdfs.org/sioc/ns#part_of
 */
export const part_of: string = 'http://rdfs.org/sioc/ns#part_of';

/**
 * reference
 * 
 * Links either created explicitly or extracted implicitly on the HTML level from the Post.
 *
 * http://rdfs.org/sioc/ns#reference
 */
export const reference: string = 'http://rdfs.org/sioc/ns#reference';

/**
 * content
 * 
 * The content of the Item in plain text format.
 *
 * http://rdfs.org/sioc/ns#content
 */
export const content: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#content';

/**
 * email sha1
 * 
 * An electronic mail address of the UserAccount, encoded using SHA1.
 *
 * http://rdfs.org/sioc/ns#email_sha1
 */
export const email_sha1: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#email_sha1';

/**
 * id
 * 
 * An identifier of a SIOC concept instance. For example, a user ID. Must be unique for instances of each type of SIOC concept within the same site.
 *
 * http://rdfs.org/sioc/ns#id
 */
export const id: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#id';

/**
 * ip address
 * 
 * The IP address used when creating this Item, UserAccount, etc. This can be associated with a creator. Some wiki articles list the IP addresses for the creator or modifiers when the usernames are absent.
 *
 * http://rdfs.org/sioc/ns#ip_address
 */
export const ip_address: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#ip_address';

/**
 * last activity date
 * 
 * The date and time of the last activity associated with a SIOC concept instance, and expressed in ISO 8601 format. This could be due to a reply Post or Comment, a modification to an Item, etc.
 *
 * http://rdfs.org/sioc/ns#last_activity_date
 */
export const last_activity_date: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#last_activity_date';

/**
 * last item date
 * 
 * The date and time of the last Post (or Item) in a Forum (or a Container), in ISO 8601 format.
 *
 * http://rdfs.org/sioc/ns#last_item_date
 */
export const last_item_date: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#last_item_date';

/**
 * last reply date
 * 
 * The date and time of the last reply Post or Comment, which could be associated with a starter Item or Post or with a Thread, and expressed in ISO 8601 format.
 *
 * http://rdfs.org/sioc/ns#last_reply_date
 */
export const last_reply_date: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#last_reply_date';

/**
 * name
 * 
 * The name of a SIOC concept instance, e.g. a username for a UserAccount, group name for a Usergroup, etc.
 *
 * http://rdfs.org/sioc/ns#name
 */
export const name: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#name';

/**
 * note
 * 
 * A note associated with this resource, for example, if it has been edited by a UserAccount.
 *
 * http://rdfs.org/sioc/ns#note
 */
export const note: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#note';

/**
 * num authors
 * 
 * The number of unique authors (UserAccounts and unregistered posters) who have contributed to this Item, Thread, Post, etc.
 *
 * http://rdfs.org/sioc/ns#num_authors
 */
export const num_authors: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#num_authors';

/**
 * num items
 * 
 * The number of Posts (or Items) in a Forum (or a Container).
 *
 * http://rdfs.org/sioc/ns#num_items
 */
export const num_items: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#num_items';

/**
 * num replies
 * 
 * The number of replies that this Item, Thread, Post, etc. has. Useful for when the reply structure is absent.
 *
 * http://rdfs.org/sioc/ns#num_replies
 */
export const num_replies: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#num_replies';

/**
 * num threads
 * 
 * The number of Threads (AKA discussion topics) in a Forum.
 *
 * http://rdfs.org/sioc/ns#num_threads
 */
export const num_threads: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#num_threads';

/**
 * num views
 * 
 * The number of times this Item, Thread, UserAccount profile, etc. has been viewed.
 *
 * http://rdfs.org/sioc/ns#num_views
 */
export const num_views: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#num_views';

/**
 * title
 * 
 * This is the title (subject line) of the Post. Note that for a Post within a threaded discussion that has no parents, it would detail the topic thread.
 *
 * http://rdfs.org/sioc/ns#title
 */
export const title: string = 'http://rdfs.org/sioc/ns#title';

/**
 * content encoded
 * 
 * The encoded content of the Post, contained in CDATA areas.
 *
 * http://rdfs.org/sioc/ns#content_encoded
 */
export const content_encoded: string = 'http://rdfs.org/sioc/ns#content_encoded';

/**
 * created at
 * 
 * When this was created, in ISO 8601 format.
 *
 * http://rdfs.org/sioc/ns#created_at
 */
export const created_at: string = 'http://rdfs.org/sioc/ns#created_at';

/**
 * description
 * 
 * The content of the Post.
 *
 * http://rdfs.org/sioc/ns#description
 */
export const description: string = 'http://rdfs.org/sioc/ns#description';

/**
 * first name
 * 
 * First (real) name of this User. Synonyms include given name or christian name.
 *
 * http://rdfs.org/sioc/ns#first_name
 */
export const first_name: string = 'http://rdfs.org/sioc/ns#first_name';

/**
 * last name
 * 
 * Last (real) name of this user. Synonyms include surname or family name.
 *
 * http://rdfs.org/sioc/ns#last_name
 */
export const last_name: string = 'http://rdfs.org/sioc/ns#last_name';

/**
 * modified at
 * 
 * When this was modified, in ISO 8601 format.
 *
 * http://rdfs.org/sioc/ns#modified_at
 */
export const modified_at: string = 'http://rdfs.org/sioc/ns#modified_at';

/**
 * subject
 * 
 * Keyword(s) describing subject of the Post.
 *
 * http://rdfs.org/sioc/ns#subject
 */
export const subject: string = 'http://rdfs.org/sioc/ns#subject';
