// rdf-namespaces/dist/skos.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var skos$1 = require('./skos-0db70ed9.js');



exports.Collection = skos$1.Collection;
exports.Concept = skos$1.Concept;
exports.ConceptScheme = skos$1.ConceptScheme;
exports.OrderedCollection = skos$1.OrderedCollection;
exports.altLabel = skos$1.altLabel;
exports.broadMatch = skos$1.broadMatch;
exports.broader = skos$1.broader;
exports.broaderTransitive = skos$1.broaderTransitive;
exports.changeNote = skos$1.changeNote;
exports.closeMatch = skos$1.closeMatch;
exports.definition = skos$1.definition;
exports.editorialNote = skos$1.editorialNote;
exports.exactMatch = skos$1.exactMatch;
exports.example = skos$1.example;
exports.hasTopConcept = skos$1.hasTopConcept;
exports.hiddenLabel = skos$1.hiddenLabel;
exports.historyNote = skos$1.historyNote;
exports.inScheme = skos$1.inScheme;
exports.mappingRelation = skos$1.mappingRelation;
exports.member = skos$1.member;
exports.memberList = skos$1.memberList;
exports.narrowMatch = skos$1.narrowMatch;
exports.narrower = skos$1.narrower;
exports.narrowerTransitive = skos$1.narrowerTransitive;
exports.notation = skos$1.notation;
exports.note = skos$1.note;
exports.prefLabel = skos$1.prefLabel;
exports.related = skos$1.related;
exports.relatedMatch = skos$1.relatedMatch;
exports.scopeNote = skos$1.scopeNote;
exports.semanticRelation = skos$1.semanticRelation;
exports.topConceptOf = skos$1.topConceptOf;
