// rdf-namespaces/dist/skos-d7472b70.js
/**
 * is in scheme
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#inScheme
 */
var inScheme = 'http://www.w3.org/2004/02/skos/core#inScheme';
/**
 * has top concept
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#hasTopConcept
 */
var hasTopConcept = 'http://www.w3.org/2004/02/skos/core#hasTopConcept';
/**
 * is top concept in scheme
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#topConceptOf
 */
var topConceptOf = 'http://www.w3.org/2004/02/skos/core#topConceptOf';
/**
 * preferred label
 *
 * A resource has no more than one value of skos:prefLabel per language tag, and no more than one value of skos:prefLabel without language tag.
 *
 * http://www.w3.org/2004/02/skos/core#prefLabel
 */
var prefLabel = 'http://www.w3.org/2004/02/skos/core#prefLabel';
/**
 * alternative label
 *
 * The range of skos:altLabel is the class of RDF plain literals.
 *
 * http://www.w3.org/2004/02/skos/core#altLabel
 */
var altLabel = 'http://www.w3.org/2004/02/skos/core#altLabel';
/**
 * hidden label
 *
 * The range of skos:hiddenLabel is the class of RDF plain literals.
 *
 * http://www.w3.org/2004/02/skos/core#hiddenLabel
 */
var hiddenLabel = 'http://www.w3.org/2004/02/skos/core#hiddenLabel';
/**
 * notation
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#notation
 */
var notation = 'http://www.w3.org/2004/02/skos/core#notation';
/**
 * note
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#note
 */
var note = 'http://www.w3.org/2004/02/skos/core#note';
/**
 * change note
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#changeNote
 */
var changeNote = 'http://www.w3.org/2004/02/skos/core#changeNote';
/**
 * definition
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#definition
 */
var definition = 'http://www.w3.org/2004/02/skos/core#definition';
/**
 * editorial note
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#editorialNote
 */
var editorialNote = 'http://www.w3.org/2004/02/skos/core#editorialNote';
/**
 * example
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#example
 */
var example = 'http://www.w3.org/2004/02/skos/core#example';
/**
 * history note
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#historyNote
 */
var historyNote = 'http://www.w3.org/2004/02/skos/core#historyNote';
/**
 * scope note
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#scopeNote
 */
var scopeNote = 'http://www.w3.org/2004/02/skos/core#scopeNote';
/**
 * is in semantic relation with
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#semanticRelation
 */
var semanticRelation = 'http://www.w3.org/2004/02/skos/core#semanticRelation';
/**
 * has broader
 *
 * Broader concepts are typically rendered as parents in a concept hierarchy (tree).
 *
 * http://www.w3.org/2004/02/skos/core#broader
 */
var broader = 'http://www.w3.org/2004/02/skos/core#broader';
/**
 * has narrower
 *
 * Narrower concepts are typically rendered as children in a concept hierarchy (tree).
 *
 * http://www.w3.org/2004/02/skos/core#narrower
 */
var narrower = 'http://www.w3.org/2004/02/skos/core#narrower';
/**
 * has related
 *
 * skos:related is disjoint with skos:broaderTransitive
 *
 * http://www.w3.org/2004/02/skos/core#related
 */
var related = 'http://www.w3.org/2004/02/skos/core#related';
/**
 * has broader transitive
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#broaderTransitive
 */
var broaderTransitive = 'http://www.w3.org/2004/02/skos/core#broaderTransitive';
/**
 * has narrower transitive
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#narrowerTransitive
 */
var narrowerTransitive = 'http://www.w3.org/2004/02/skos/core#narrowerTransitive';
/**
 * has member
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#member
 */
var member = 'http://www.w3.org/2004/02/skos/core#member';
/**
 * has member list
 *
 * For any resource, every item in the list given as the value of the
 *       skos:memberList property is also a value of the skos:member property.
 *
 * http://www.w3.org/2004/02/skos/core#memberList
 */
var memberList = 'http://www.w3.org/2004/02/skos/core#memberList';
/**
 * is in mapping relation with
 *
 * These concept mapping relations mirror semantic relations, and the data model defined below is similar (with the exception of skos:exactMatch) to the data model defined for semantic relations. A distinct vocabulary is provided for concept mapping relations, to provide a convenient way to differentiate links within a concept scheme from links between concept schemes. However, this pattern of usage is not a formal requirement of the SKOS data model, and relies on informal definitions of best practice.
 *
 * http://www.w3.org/2004/02/skos/core#mappingRelation
 */
var mappingRelation = 'http://www.w3.org/2004/02/skos/core#mappingRelation';
/**
 * has broader match
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#broadMatch
 */
var broadMatch = 'http://www.w3.org/2004/02/skos/core#broadMatch';
/**
 * has narrower match
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#narrowMatch
 */
var narrowMatch = 'http://www.w3.org/2004/02/skos/core#narrowMatch';
/**
 * has related match
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#relatedMatch
 */
var relatedMatch = 'http://www.w3.org/2004/02/skos/core#relatedMatch';
/**
 * has exact match
 *
 * skos:exactMatch is disjoint with each of the properties skos:broadMatch and skos:relatedMatch.
 *
 * http://www.w3.org/2004/02/skos/core#exactMatch
 */
var exactMatch = 'http://www.w3.org/2004/02/skos/core#exactMatch';
/**
 * has close match
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#closeMatch
 */
var closeMatch = 'http://www.w3.org/2004/02/skos/core#closeMatch';
/**
 * Concept
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#Concept
 */
var Concept = 'http://www.w3.org/2004/02/skos/core#Concept';
/**
 * Concept Scheme
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#ConceptScheme
 */
var ConceptScheme = 'http://www.w3.org/2004/02/skos/core#ConceptScheme';
/**
 * Collection
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#Collection
 */
var Collection = 'http://www.w3.org/2004/02/skos/core#Collection';
/**
 * Ordered Collection
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#OrderedCollection
 */
var OrderedCollection = 'http://www.w3.org/2004/02/skos/core#OrderedCollection';

var skosImport = /*#__PURE__*/Object.freeze({
	inScheme: inScheme,
	hasTopConcept: hasTopConcept,
	topConceptOf: topConceptOf,
	prefLabel: prefLabel,
	altLabel: altLabel,
	hiddenLabel: hiddenLabel,
	notation: notation,
	note: note,
	changeNote: changeNote,
	definition: definition,
	editorialNote: editorialNote,
	example: example,
	historyNote: historyNote,
	scopeNote: scopeNote,
	semanticRelation: semanticRelation,
	broader: broader,
	narrower: narrower,
	related: related,
	broaderTransitive: broaderTransitive,
	narrowerTransitive: narrowerTransitive,
	member: member,
	memberList: memberList,
	mappingRelation: mappingRelation,
	broadMatch: broadMatch,
	narrowMatch: narrowMatch,
	relatedMatch: relatedMatch,
	exactMatch: exactMatch,
	closeMatch: closeMatch,
	Concept: Concept,
	ConceptScheme: ConceptScheme,
	Collection: Collection,
	OrderedCollection: OrderedCollection
});

export { exactMatch as A, closeMatch as B, Concept as C, ConceptScheme as D, Collection as E, skosImport as F, OrderedCollection as O, altLabel as a, hiddenLabel as b, note as c, changeNote as d, definition as e, editorialNote as f, example as g, hasTopConcept as h, inScheme as i, historyNote as j, semanticRelation as k, broader as l, narrower as m, notation as n, broaderTransitive as o, prefLabel as p, narrowerTransitive as q, related as r, scopeNote as s, topConceptOf as t, member as u, memberList as v, mappingRelation as w, broadMatch as x, narrowMatch as y, relatedMatch as z };
