// rdf-namespaces/dist/skos.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * is in scheme
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#inScheme
 */
export const inScheme: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#inScheme';

/**
 * has top concept
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#hasTopConcept
 */
export const hasTopConcept: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#hasTopConcept';

/**
 * is top concept in scheme
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#topConceptOf
 */
export const topConceptOf: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#topConceptOf';

/**
 * preferred label
 * 
 * A resource has no more than one value of skos:prefLabel per language tag, and no more than one value of skos:prefLabel without language tag.
 *
 * http://www.w3.org/2004/02/skos/core#prefLabel
 */
export const prefLabel: string = 'http://www.w3.org/2004/02/skos/core#prefLabel';

/**
 * alternative label
 * 
 * The range of skos:altLabel is the class of RDF plain literals.
 *
 * http://www.w3.org/2004/02/skos/core#altLabel
 */
export const altLabel: string = 'http://www.w3.org/2004/02/skos/core#altLabel';

/**
 * hidden label
 * 
 * The range of skos:hiddenLabel is the class of RDF plain literals.
 *
 * http://www.w3.org/2004/02/skos/core#hiddenLabel
 */
export const hiddenLabel: string = 'http://www.w3.org/2004/02/skos/core#hiddenLabel';

/**
 * notation
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#notation
 */
export const notation: OwlDatatypeProperty = 'http://www.w3.org/2004/02/skos/core#notation';

/**
 * note
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#note
 */
export const note: string = 'http://www.w3.org/2004/02/skos/core#note';

/**
 * change note
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#changeNote
 */
export const changeNote: string = 'http://www.w3.org/2004/02/skos/core#changeNote';

/**
 * definition
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#definition
 */
export const definition: string = 'http://www.w3.org/2004/02/skos/core#definition';

/**
 * editorial note
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#editorialNote
 */
export const editorialNote: string = 'http://www.w3.org/2004/02/skos/core#editorialNote';

/**
 * example
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#example
 */
export const example: string = 'http://www.w3.org/2004/02/skos/core#example';

/**
 * history note
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#historyNote
 */
export const historyNote: string = 'http://www.w3.org/2004/02/skos/core#historyNote';

/**
 * scope note
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#scopeNote
 */
export const scopeNote: string = 'http://www.w3.org/2004/02/skos/core#scopeNote';

/**
 * is in semantic relation with
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#semanticRelation
 */
export const semanticRelation: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#semanticRelation';

/**
 * has broader
 * 
 * Broader concepts are typically rendered as parents in a concept hierarchy (tree).
 *
 * http://www.w3.org/2004/02/skos/core#broader
 */
export const broader: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#broader';

/**
 * has narrower
 * 
 * Narrower concepts are typically rendered as children in a concept hierarchy (tree).
 *
 * http://www.w3.org/2004/02/skos/core#narrower
 */
export const narrower: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#narrower';

/**
 * has related
 * 
 * skos:related is disjoint with skos:broaderTransitive
 *
 * http://www.w3.org/2004/02/skos/core#related
 */
export const related: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#related';

/**
 * has broader transitive
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#broaderTransitive
 */
export const broaderTransitive: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#broaderTransitive';

/**
 * has narrower transitive
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#narrowerTransitive
 */
export const narrowerTransitive: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#narrowerTransitive';

/**
 * has member
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#member
 */
export const member: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#member';

/**
 * has member list
 * 
 * For any resource, every item in the list given as the value of the
 *       skos:memberList property is also a value of the skos:member property.
 *
 * http://www.w3.org/2004/02/skos/core#memberList
 */
export const memberList: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#memberList';

/**
 * is in mapping relation with
 * 
 * These concept mapping relations mirror semantic relations, and the data model defined below is similar (with the exception of skos:exactMatch) to the data model defined for semantic relations. A distinct vocabulary is provided for concept mapping relations, to provide a convenient way to differentiate links within a concept scheme from links between concept schemes. However, this pattern of usage is not a formal requirement of the SKOS data model, and relies on informal definitions of best practice.
 *
 * http://www.w3.org/2004/02/skos/core#mappingRelation
 */
export const mappingRelation: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#mappingRelation';

/**
 * has broader match
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#broadMatch
 */
export const broadMatch: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#broadMatch';

/**
 * has narrower match
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#narrowMatch
 */
export const narrowMatch: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#narrowMatch';

/**
 * has related match
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#relatedMatch
 */
export const relatedMatch: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#relatedMatch';

/**
 * has exact match
 * 
 * skos:exactMatch is disjoint with each of the properties skos:broadMatch and skos:relatedMatch.
 *
 * http://www.w3.org/2004/02/skos/core#exactMatch
 */
export const exactMatch: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#exactMatch';

/**
 * has close match
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#closeMatch
 */
export const closeMatch: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#closeMatch';

/**
 * Concept
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#Concept
 */
export const Concept: OwlClass = 'http://www.w3.org/2004/02/skos/core#Concept';

/**
 * Concept Scheme
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#ConceptScheme
 */
export const ConceptScheme: OwlClass = 'http://www.w3.org/2004/02/skos/core#ConceptScheme';

/**
 * Collection
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#Collection
 */
export const Collection: OwlClass = 'http://www.w3.org/2004/02/skos/core#Collection';

/**
 * Ordered Collection
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#OrderedCollection
 */
export const OrderedCollection: OwlClass = 'http://www.w3.org/2004/02/skos/core#OrderedCollection';
