// rdf-namespaces/dist/solid.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var solid$1 = require('./solid-e10b9413.js');



exports.Account = solid$1.Account;
exports.Inbox = solid$1.Inbox;
exports.ListedDocument = solid$1.ListedDocument;
exports.Notification = solid$1.Notification;
exports.Patch = solid$1.Patch;
exports.Timeline = solid$1.Timeline;
exports.TypeIndex = solid$1.TypeIndex;
exports.TypeRegistration = solid$1.TypeRegistration;
exports.UnlistedDocument = solid$1.UnlistedDocument;
exports.account = solid$1.account;
exports.deletes = solid$1.deletes;
exports.forClass = solid$1.forClass;
exports.inbox = solid$1.inbox;
exports.inserts = solid$1.inserts;
exports.instance = solid$1.instance;
exports.instanceContainer = solid$1.instanceContainer;
exports.loginEndpoint = solid$1.loginEndpoint;
exports.logoutEndpoint = solid$1.logoutEndpoint;
exports.notification = solid$1.notification;
exports.oidcIssuer = solid$1.oidcIssuer;
exports.patches = solid$1.patches;
exports.privateTypeIndex = solid$1.privateTypeIndex;
exports.publicTypeIndex = solid$1.publicTypeIndex;
exports.read = solid$1.read;
exports.storageQuota = solid$1.storageQuota;
exports.storageUsage = solid$1.storageUsage;
exports.timeline = solid$1.timeline;
exports.typeIndex = solid$1.typeIndex;
exports.where = solid$1.where;
