// rdf-namespaces/dist/solid.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * account
 * 
 * A solid account belonging to an Agent.
 *
 * http://www.w3.org/ns/solid/terms#account
 */
export const account: Property = 'http://www.w3.org/ns/solid/terms#account';

/**
 * deletes
 * 
 * The triple patterns this patch removes from the document.
 *
 * http://www.w3.org/ns/solid/terms#deletes
 */
export const deletes: Property = 'http://www.w3.org/ns/solid/terms#deletes';

/**
 * registry class
 * 
 * A class that is used to map an listed or unlisted type index.
 *
 * http://www.w3.org/ns/solid/terms#forClass
 */
export const forClass: Property = 'http://www.w3.org/ns/solid/terms#forClass';

/**
 * inbox (deprecated)
 * 
 * Deprecated pointer to a Linked Data Notifications inbox; please use http://www.w3.org/ns/ldp#inbox instead.
 *
 * http://www.w3.org/ns/solid/terms#inbox
 */
export const inbox: Property = 'http://www.w3.org/ns/solid/terms#inbox';

/**
 * inserts
 * 
 * The triple patterns this patch adds to the document.
 *
 * http://www.w3.org/ns/solid/terms#inserts
 */
export const inserts: Property = 'http://www.w3.org/ns/solid/terms#inserts';

/**
 * instance
 * 
 * Maps a type to an individual resource, typically an index or a directory listing resource.
 *
 * http://www.w3.org/ns/solid/terms#instance
 */
export const instance: Property = 'http://www.w3.org/ns/solid/terms#instance';

/**
 * instance container
 * 
 * Maps a type to a container which the client would have to list to get the instances of that type.
 *
 * http://www.w3.org/ns/solid/terms#instanceContainer
 */
export const instanceContainer: Property = 'http://www.w3.org/ns/solid/terms#instanceContainer';

/**
 * loginEndpoint
 * 
 * The login URI of a given server.
 *
 * http://www.w3.org/ns/solid/terms#loginEndpoint
 */
export const loginEndpoint: Property = 'http://www.w3.org/ns/solid/terms#loginEndpoint';

/**
 * logoutEndpoint
 * 
 * The logout URI of a given server.
 *
 * http://www.w3.org/ns/solid/terms#logoutEndpoint
 */
export const logoutEndpoint: Property = 'http://www.w3.org/ns/solid/terms#logoutEndpoint';

/**
 * notification
 * 
 * Notification resource for an inbox.
 *
 * http://www.w3.org/ns/solid/terms#notification
 */
export const notification: Property = 'http://www.w3.org/ns/solid/terms#notification';

/**
 * private type index
 * 
 * Points to an unlisted type index resource.
 *
 * http://www.w3.org/ns/solid/terms#privateTypeIndex
 */
export const privateTypeIndex: Property = 'http://www.w3.org/ns/solid/terms#privateTypeIndex';

/**
 * public type index
 * 
 * Points to a listed type index resource.
 *
 * http://www.w3.org/ns/solid/terms#publicTypeIndex
 */
export const publicTypeIndex: Property = 'http://www.w3.org/ns/solid/terms#publicTypeIndex';

/**
 * read
 * 
 * Indicates if a message has been read or not. This property should have a boolean datatype.
 *
 * http://www.w3.org/ns/solid/terms#read
 */
export const read: Property = 'http://www.w3.org/ns/solid/terms#read';

/**
 * type index
 * 
 * Points to a TypeIndex resource.
 *
 * http://www.w3.org/ns/solid/terms#typeIndex
 */
export const typeIndex: Property = 'http://www.w3.org/ns/solid/terms#typeIndex';

/**
 * oidcIssuer
 * 
 * The preferred OpenID Connect issuer URI for a given Web ID.
 *
 * http://www.w3.org/ns/solid/terms#oidcIssuer
 */
export const oidcIssuer: Property = 'http://www.w3.org/ns/solid/terms#oidcIssuer';

/**
 * patches
 * 
 * The document to which this patch applies.
 *
 * http://www.w3.org/ns/solid/terms#patches
 */
export const patches: Property = 'http://www.w3.org/ns/solid/terms#patches';

/**
 * timeline
 * 
 * Timeline for a given resource.
 *
 * http://www.w3.org/ns/solid/terms#timeline
 */
export const timeline: Property = 'http://www.w3.org/ns/solid/terms#timeline';

/**
 * where
 * 
 * The conditions the document and the inserted and deleted triple patterns need to satisfy in order for the patch to be applied.
 *
 * http://www.w3.org/ns/solid/terms#where
 */
export const where: Property = 'http://www.w3.org/ns/solid/terms#where';

/**
 * Non-volatile memory quota
 * 
 * The quota of non-volatile memory that is available for the account (in bytes)
 *
 * http://www.w3.org/ns/solid/terms#storageQuota
 */
export const storageQuota: Property = 'http://www.w3.org/ns/solid/terms#storageQuota';

/**
 * Non-volatile memory usage
 * 
 * The amount of non-volatile memory that the account have used (in bytes)
 *
 * http://www.w3.org/ns/solid/terms#storageUsage
 */
export const storageUsage: Property = 'http://www.w3.org/ns/solid/terms#storageUsage';

/**
 * Account
 * 
 * A Solid account.
 *
 * http://www.w3.org/ns/solid/terms#Account
 */
export const Account: Class = 'http://www.w3.org/ns/solid/terms#Account';

/**
 * Inbox
 * 
 * A resource containing notifications.
 *
 * http://www.w3.org/ns/solid/terms#Inbox
 */
export const Inbox: Class = 'http://www.w3.org/ns/solid/terms#Inbox';

/**
 * Listed Type Index
 * 
 * Listed Type Index is a registry of resources that are publicly discoverable by outside users and applications.
 *
 * http://www.w3.org/ns/solid/terms#ListedDocument
 */
export const ListedDocument: Class = 'http://www.w3.org/ns/solid/terms#ListedDocument';

/**
 * Notification
 * 
 * A notification resource.
 *
 * http://www.w3.org/ns/solid/terms#Notification
 */
export const Notification: Class = 'http://www.w3.org/ns/solid/terms#Notification';

/**
 * Patch
 * 
 * A patch expresses conditional modifications to a resource that has an RDF-based representation.
 *
 * http://www.w3.org/ns/solid/terms#Patch
 */
export const Patch: Class = 'http://www.w3.org/ns/solid/terms#Patch';

/**
 * Timeline
 * 
 * A resource containing time ordered items and sub-containers.  Sub-containers may be desirable in file based systems to split the timeline into logical components e.g. /yyyy-mm-dd/ as used in ISO 8061.
 *
 * http://www.w3.org/ns/solid/terms#Timeline
 */
export const Timeline: Class = 'http://www.w3.org/ns/solid/terms#Timeline';

/**
 * Type index
 * 
 * A index of type registries for resources. Applications can register the RDF type they use and list them in the index resource.
 *
 * http://www.w3.org/ns/solid/terms#TypeIndex
 */
export const TypeIndex: Class = 'http://www.w3.org/ns/solid/terms#TypeIndex';

/**
 * Type Registration
 * 
 * The registered types that map a RDF classes/types to their locations using either `instance` or `instanceContainer` property.
 *
 * http://www.w3.org/ns/solid/terms#TypeRegistration
 */
export const TypeRegistration: Class = 'http://www.w3.org/ns/solid/terms#TypeRegistration';

/**
 * Unlisted Type Index
 * 
 * Unlisted Type Index is a registry of resources that are private to the user and their apps, for types that are not publicly discoverable.
 *
 * http://www.w3.org/ns/solid/terms#UnlistedDocument
 */
export const UnlistedDocument: Class = 'http://www.w3.org/ns/solid/terms#UnlistedDocument';
