// rdf-namespaces/dist/space.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * master workspace
 * 
 * 
 *
 * http://www.w3.org/ns/pim/space#masterWorkspace
 */
export const masterWorkspace: Property = 'http://www.w3.org/ns/pim/space#masterWorkspace';

/**
 * preferences file
 * 
 * 
 *
 * http://www.w3.org/ns/pim/space#preferencesFile
 */
export const preferencesFile: Property = 'http://www.w3.org/ns/pim/space#preferencesFile';

/**
 * storage
 * 
 * The storage in which this workspace is
 *
 * http://www.w3.org/ns/pim/space#storage
 */
export const storage: Property = 'http://www.w3.org/ns/pim/space#storage';

/**
 * URI prefix
 * 
 * URIs which start with this string are in this workspace or storage.
 * This may be used for constructing URIs for new storage resources.
 * 
 *
 * http://www.w3.org/ns/pim/space#uriPrefix
 */
export const uriPrefix: Property = 'http://www.w3.org/ns/pim/space#uriPrefix';

/**
 * workspace
 * 
 * 
 *
 * http://www.w3.org/ns/pim/space#workspace
 */
export const workspace: Property = 'http://www.w3.org/ns/pim/space#workspace';

/**
 * access controlled storage
 * 
 * A  storage is a space of URIs in which you can individually control for each resource
 *     who has access to it.
 * 
 *
 * http://www.w3.org/ns/pim/space#ControlledStorage
 */
export const ControlledStorage: Class = 'http://www.w3.org/ns/pim/space#ControlledStorage';

/**
 * personal storage
 * 
 * A personal storage is a space of URIs in which you and only you have access to data,
 *     you cannot give access to anyone else.
 * 
 *
 * http://www.w3.org/ns/pim/space#PersonalStorage
 */
export const PersonalStorage: Class = 'http://www.w3.org/ns/pim/space#PersonalStorage';

/**
 * public storage
 * 
 * A public storage is a space of URIs in which you have access to data,
 *     and all data is accessible to anyone without control.
 * 
 *
 * http://www.w3.org/ns/pim/space#PublicStorage
 */
export const PublicStorage: Class = 'http://www.w3.org/ns/pim/space#PublicStorage';

/**
 * storage
 * 
 * A storage is a space of URIs in which you have access to data.
 * 
 *
 * http://www.w3.org/ns/pim/space#Storage
 */
export const Storage: Class = 'http://www.w3.org/ns/pim/space#Storage';

/**
 * workspace
 * 
 * Workspaces are place where data is stored, and associated polices of privacy.
 * A given application typically stores information in several different
 * workspaces, some being user private, some shared, and some public.
 * 
 *
 * http://www.w3.org/ns/pim/space#Workspace
 */
export const Workspace: Class = 'http://www.w3.org/ns/pim/space#Workspace';
