// rdf-namespaces/dist/space-57bef06f.js
'use strict';

/**
 * master workspace
 *
 *
 *
 * http://www.w3.org/ns/pim/space#masterWorkspace
 */
var masterWorkspace = 'http://www.w3.org/ns/pim/space#masterWorkspace';
/**
 * preferences file
 *
 *
 *
 * http://www.w3.org/ns/pim/space#preferencesFile
 */
var preferencesFile = 'http://www.w3.org/ns/pim/space#preferencesFile';
/**
 * storage
 *
 * The storage in which this workspace is
 *
 * http://www.w3.org/ns/pim/space#storage
 */
var storage = 'http://www.w3.org/ns/pim/space#storage';
/**
 * URI prefix
 *
 * URIs which start with this string are in this workspace or storage.
 * This may be used for constructing URIs for new storage resources.
 *
 *
 * http://www.w3.org/ns/pim/space#uriPrefix
 */
var uriPrefix = 'http://www.w3.org/ns/pim/space#uriPrefix';
/**
 * workspace
 *
 *
 *
 * http://www.w3.org/ns/pim/space#workspace
 */
var workspace = 'http://www.w3.org/ns/pim/space#workspace';
/**
 * access controlled storage
 *
 * A  storage is a space of URIs in which you can individually control for each resource
 *     who has access to it.
 *
 *
 * http://www.w3.org/ns/pim/space#ControlledStorage
 */
var ControlledStorage = 'http://www.w3.org/ns/pim/space#ControlledStorage';
/**
 * personal storage
 *
 * A personal storage is a space of URIs in which you and only you have access to data,
 *     you cannot give access to anyone else.
 *
 *
 * http://www.w3.org/ns/pim/space#PersonalStorage
 */
var PersonalStorage = 'http://www.w3.org/ns/pim/space#PersonalStorage';
/**
 * public storage
 *
 * A public storage is a space of URIs in which you have access to data,
 *     and all data is accessible to anyone without control.
 *
 *
 * http://www.w3.org/ns/pim/space#PublicStorage
 */
var PublicStorage = 'http://www.w3.org/ns/pim/space#PublicStorage';
/**
 * storage
 *
 * A storage is a space of URIs in which you have access to data.
 *
 *
 * http://www.w3.org/ns/pim/space#Storage
 */
var Storage = 'http://www.w3.org/ns/pim/space#Storage';
/**
 * workspace
 *
 * Workspaces are place where data is stored, and associated polices of privacy.
 * A given application typically stores information in several different
 * workspaces, some being user private, some shared, and some public.
 *
 *
 * http://www.w3.org/ns/pim/space#Workspace
 */
var Workspace = 'http://www.w3.org/ns/pim/space#Workspace';

var spaceImport = /*#__PURE__*/Object.freeze({
	masterWorkspace: masterWorkspace,
	preferencesFile: preferencesFile,
	storage: storage,
	uriPrefix: uriPrefix,
	workspace: workspace,
	ControlledStorage: ControlledStorage,
	PersonalStorage: PersonalStorage,
	PublicStorage: PublicStorage,
	Storage: Storage,
	Workspace: Workspace
});

exports.ControlledStorage = ControlledStorage;
exports.PersonalStorage = PersonalStorage;
exports.PublicStorage = PublicStorage;
exports.Storage = Storage;
exports.Workspace = Workspace;
exports.masterWorkspace = masterWorkspace;
exports.preferencesFile = preferencesFile;
exports.spaceImport = spaceImport;
exports.storage = storage;
exports.uriPrefix = uriPrefix;
exports.workspace = workspace;
