// rdf-namespaces/dist/tab.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * URI
 * 
 * The URI requested in this access. 
 *         This is NOT the relationship betwen a URI and a thing t identifiers
 *
 * http://www.w3.org/2007/ont/link#uri
 */
export const uri: Property = 'http://www.w3.org/2007/ont/link#uri';

/**
 * is mentioned in
 * 
 * 
 *       This class is mentioned in the following document
 *     
 *
 * http://www.w3.org/2007/ont/link#isMentionedIn
 */
export const isMentionedIn: Property = 'http://www.w3.org/2007/ont/link#isMentionedIn';

/**
 * mentions
 * 
 * This document mentions the following class
 *
 * http://www.w3.org/2007/ont/link#mentionsClass
 */
export const mentionsClass: Property = 'http://www.w3.org/2007/ont/link#mentionsClass';

/**
 * supports protocol
 * 
 * The application supports this network protocol.
 *
 * http://www.w3.org/2007/ont/link#protocol
 */
export const protocol: Property = 'http://www.w3.org/2007/ont/link#protocol';

/**
 * requested URI
 * 
 * 
 *
 * http://www.w3.org/2007/ont/link#requestedURI
 */
export const requestedURI: Property = 'http://www.w3.org/2007/ont/link#requestedURI';

/**
 * redirected request
 * 
 * 
 *
 * http://www.w3.org/2007/ont/link#redirectedRequest
 */
export const redirectedRequest: Property = 'http://www.w3.org/2007/ont/link#redirectedRequest';

/**
 * response
 * 
 * 
 *
 * http://www.w3.org/2007/ont/link#response
 */
export const response: Property = 'http://www.w3.org/2007/ont/link#response';

/**
 * uses source
 * 
 * A source file accessed in this session.
 *
 * http://www.w3.org/2007/ont/link#source
 */
export const source: Property = 'http://www.w3.org/2007/ont/link#source';

/**
 * status
 * 
 * The status of a network access.
 *
 * http://www.w3.org/2007/ont/link#status
 */
export const status: Property = 'http://www.w3.org/2007/ont/link#status';

/**
 * examples and test data
 * 
 * Soemthing which can be used as an example for testing purposes
 *
 * http://www.w3.org/2007/ont/link#testDataset
 */
export const testDataset: Property = 'http://www.w3.org/2007/ont/link#testDataset';

/**
 * session
 * 
 * 
 *
 * http://www.w3.org/2007/ont/link#session
 */
export const session: Property = 'http://www.w3.org/2007/ont/link#session';

/**
 * referrer
 * 
 * A resource which contains a reference to the subject.
 *          That is, one of the statements in the object file
 *          has the subject URI.
 *          This property is stored by the tabulator library to explain 
 *          why it looked up a given URI. This can help 
 *          trace an error when it was in fact a mis-spelling.
 *
 * http://www.w3.org/2007/ont/link#requestedBy
 */
export const requestedBy: Property = 'http://www.w3.org/2007/ont/link#requestedBy';

/**
 * document
 * 
 * An Information Resource, such as a web page, image,
 *         data file, etc.
 *
 * http://www.w3.org/2007/ont/link#Document
 */
export const Document: Class = 'http://www.w3.org/2007/ont/link#Document';

/**
 * data document
 * 
 * An Information Resource whose meaning which can directly mapped into the RDF model. For example a data file which can be reterived as RDF using HTTP.
 *
 * http://www.w3.org/2007/ont/link#RDFDocument
 */
export const RDFDocument: Class = 'http://www.w3.org/2007/ont/link#RDFDocument';

/**
 * mailbox
 * 
 * An end point in the Internet Mail sytem. 
 *             Anything which has a URI in the malto: scheme is a Mailbox.
 *
 * http://www.w3.org/2007/ont/link#Mailbox
 */
export const Mailbox: Class = 'http://www.w3.org/2007/ont/link#Mailbox';

/**
 * access
 * 
 * An event in which a network access is made.
 *
 * http://www.w3.org/2007/ont/link#Access
 */
export const Access: Class = 'http://www.w3.org/2007/ont/link#Access';

/**
 * access
 * 
 * A network response to a request.
 *
 * http://www.w3.org/2007/ont/link#Response
 */
export const Response: Class = 'http://www.w3.org/2007/ont/link#Response';

/**
 * protocol event
 * 
 * A step in a network protocol, such as a request
 *         or response message.
 *
 * http://www.w3.org/2007/ont/link#ProtocolEvent
 */
export const ProtocolEvent: Class = 'http://www.w3.org/2007/ont/link#ProtocolEvent';

/**
 * session
 * 
 * A run of the application software in which various
 *         network acecssses etc happen.
 *
 * http://www.w3.org/2007/ont/link#Session
 */
export const Session: Class = 'http://www.w3.org/2007/ont/link#Session';
