// rdf-namespaces/dist/trip.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * funder
 * 
 * Who pays for the this? An organization or individual.
 *      Could apply to a trip (or a doap:Project, or a cal:Event, etc).
 *      There may be more than one funder.
 *
 * http://www.w3.org/ns/pim/trip#funder
 */
export const funder: Property = 'http://www.w3.org/ns/pim/trip#funder';

/**
 * location
 * 
 * 
 *
 * http://www.w3.org/ns/pim/trip#location
 */
export const location: Property = 'http://www.w3.org/ns/pim/trip#location';

/**
 * part
 * 
 * 
 *
 * http://www.w3.org/ns/pim/trip#part
 */
export const part: Property = 'http://www.w3.org/ns/pim/trip#part';

/**
 * travel date
 * 
 * When the travel took place, or started. This should be a datetime (or a date).
 *     This may be different fron the date the fare was ivoiced or charged.
 *
 * http://www.w3.org/ns/pim/trip#travelDate
 */
export const travelDate: Property = 'http://www.w3.org/ns/pim/trip#travelDate';

/**
 * trip
 * 
 * The trip associated with an expense (a financial transaction), 
 *     a photo, a meeting, etc.
 *
 * http://www.w3.org/ns/pim/trip#trip
 */
export const trip: Property = 'http://www.w3.org/ns/pim/trip#trip';

/**
 * taxi transaction
 * 
 * A Taxi transaction is a financial trasction related to a ride in
 *     a taxi, car, etc. 
 *
 * http://www.w3.org/ns/pim/trip#TaxiTransaction
 */
export const TaxiTransaction: Class = 'http://www.w3.org/ns/pim/trip#TaxiTransaction';

/**
 * trip
 * 
 * 
 *
 * http://www.w3.org/ns/pim/trip#Trip
 */
export const Trip: Class = 'http://www.w3.org/ns/pim/trip#Trip';
