// rdf-namespaces/dist/trip-f119c3cc.js
'use strict';

/**
 * funder
 *
 * Who pays for the this? An organization or individual.
 *      Could apply to a trip (or a doap:Project, or a cal:Event, etc).
 *      There may be more than one funder.
 *
 * http://www.w3.org/ns/pim/trip#funder
 */
var funder = 'http://www.w3.org/ns/pim/trip#funder';
/**
 * location
 *
 *
 *
 * http://www.w3.org/ns/pim/trip#location
 */
var location = 'http://www.w3.org/ns/pim/trip#location';
/**
 * part
 *
 *
 *
 * http://www.w3.org/ns/pim/trip#part
 */
var part = 'http://www.w3.org/ns/pim/trip#part';
/**
 * travel date
 *
 * When the travel took place, or started. This should be a datetime (or a date).
 *     This may be different fron the date the fare was ivoiced or charged.
 *
 * http://www.w3.org/ns/pim/trip#travelDate
 */
var travelDate = 'http://www.w3.org/ns/pim/trip#travelDate';
/**
 * trip
 *
 * The trip associated with an expense (a financial transaction),
 *     a photo, a meeting, etc.
 *
 * http://www.w3.org/ns/pim/trip#trip
 */
var trip = 'http://www.w3.org/ns/pim/trip#trip';
/**
 * taxi transaction
 *
 * A Taxi transaction is a financial trasction related to a ride in
 *     a taxi, car, etc.
 *
 * http://www.w3.org/ns/pim/trip#TaxiTransaction
 */
var TaxiTransaction = 'http://www.w3.org/ns/pim/trip#TaxiTransaction';
/**
 * trip
 *
 *
 *
 * http://www.w3.org/ns/pim/trip#Trip
 */
var Trip = 'http://www.w3.org/ns/pim/trip#Trip';

var tripImport = /*#__PURE__*/Object.freeze({
	funder: funder,
	location: location,
	part: part,
	travelDate: travelDate,
	trip: trip,
	TaxiTransaction: TaxiTransaction,
	Trip: Trip
});

exports.TaxiTransaction = TaxiTransaction;
exports.Trip = Trip;
exports.funder = funder;
exports.location = location;
exports.part = part;
exports.travelDate = travelDate;
exports.trip = trip;
exports.tripImport = tripImport;
