// rdf-namespaces/dist/vcard.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * Acquaintance
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Acquaintance
 */
export const Acquaintance: OwlClass = 'http://www.w3.org/2006/vcard/ns#Acquaintance';

/**
 * Agent
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Agent
 */
export const Agent: OwlClass = 'http://www.w3.org/2006/vcard/ns#Agent';

/**
 * BBS
 * 
 * This class is deprecated
 *
 * http://www.w3.org/2006/vcard/ns#BBS
 */
export const BBS: OwlClass = 'http://www.w3.org/2006/vcard/ns#BBS';

/**
 * Car
 * 
 * This class is deprecated
 *
 * http://www.w3.org/2006/vcard/ns#Car
 */
export const Car: OwlClass = 'http://www.w3.org/2006/vcard/ns#Car';

/**
 * Cell
 * 
 * Also called mobile telephone
 *
 * http://www.w3.org/2006/vcard/ns#Cell
 */
export const Cell: OwlClass = 'http://www.w3.org/2006/vcard/ns#Cell';

/**
 * Child
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Child
 */
export const Child: OwlClass = 'http://www.w3.org/2006/vcard/ns#Child';

/**
 * Colleague
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Colleague
 */
export const Colleague: OwlClass = 'http://www.w3.org/2006/vcard/ns#Colleague';

/**
 * Contact
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Contact
 */
export const Contact: OwlClass = 'http://www.w3.org/2006/vcard/ns#Contact';

/**
 * Coresident
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Coresident
 */
export const Coresident: OwlClass = 'http://www.w3.org/2006/vcard/ns#Coresident';

/**
 * Coworker
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Coworker
 */
export const Coworker: OwlClass = 'http://www.w3.org/2006/vcard/ns#Coworker';

/**
 * Crush
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Crush
 */
export const Crush: OwlClass = 'http://www.w3.org/2006/vcard/ns#Crush';

/**
 * Date
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Date
 */
export const Date: OwlClass = 'http://www.w3.org/2006/vcard/ns#Date';

/**
 * Dom
 * 
 * This class is deprecated
 *
 * http://www.w3.org/2006/vcard/ns#Dom
 */
export const Dom: OwlClass = 'http://www.w3.org/2006/vcard/ns#Dom';

/**
 * Emergency
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Emergency
 */
export const Emergency: OwlClass = 'http://www.w3.org/2006/vcard/ns#Emergency';

/**
 * Fax
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Fax
 */
export const Fax: OwlClass = 'http://www.w3.org/2006/vcard/ns#Fax';

/**
 * Female
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Female
 */
export const Female: OwlClass = 'http://www.w3.org/2006/vcard/ns#Female';

/**
 * Friend
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Friend
 */
export const Friend: OwlClass = 'http://www.w3.org/2006/vcard/ns#Friend';

/**
 * Home
 * 
 * This implies that the property is related to an individual's personal life
 *
 * http://www.w3.org/2006/vcard/ns#Home
 */
export const Home: OwlClass = 'http://www.w3.org/2006/vcard/ns#Home';

/**
 * ISDN
 * 
 * This class is deprecated
 *
 * http://www.w3.org/2006/vcard/ns#ISDN
 */
export const ISDN: OwlClass = 'http://www.w3.org/2006/vcard/ns#ISDN';

/**
 * Internet
 * 
 * This class is deprecated
 *
 * http://www.w3.org/2006/vcard/ns#Internet
 */
export const Internet: OwlClass = 'http://www.w3.org/2006/vcard/ns#Internet';

/**
 * Intl
 * 
 * This class is deprecated
 *
 * http://www.w3.org/2006/vcard/ns#Intl
 */
export const Intl: OwlClass = 'http://www.w3.org/2006/vcard/ns#Intl';

/**
 * Kin
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Kin
 */
export const Kin: OwlClass = 'http://www.w3.org/2006/vcard/ns#Kin';

/**
 * Label
 * 
 * This class is deprecated
 *
 * http://www.w3.org/2006/vcard/ns#Label
 */
export const Label: OwlClass = 'http://www.w3.org/2006/vcard/ns#Label';

/**
 * Male
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Male
 */
export const Male: OwlClass = 'http://www.w3.org/2006/vcard/ns#Male';

/**
 * Me
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Me
 */
export const Me: OwlClass = 'http://www.w3.org/2006/vcard/ns#Me';

/**
 * Met
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Met
 */
export const Met: OwlClass = 'http://www.w3.org/2006/vcard/ns#Met';

/**
 * Modem
 * 
 * This class is deprecated
 *
 * http://www.w3.org/2006/vcard/ns#Modem
 */
export const Modem: OwlClass = 'http://www.w3.org/2006/vcard/ns#Modem';

/**
 * Msg
 * 
 * This class is deprecated
 *
 * http://www.w3.org/2006/vcard/ns#Msg
 */
export const Msg: OwlClass = 'http://www.w3.org/2006/vcard/ns#Msg';

/**
 * Muse
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Muse
 */
export const Muse: OwlClass = 'http://www.w3.org/2006/vcard/ns#Muse';

/**
 * Neighbor
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Neighbor
 */
export const Neighbor: OwlClass = 'http://www.w3.org/2006/vcard/ns#Neighbor';

/**
 * None
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#None
 */
export const None: OwlClass = 'http://www.w3.org/2006/vcard/ns#None';

/**
 * Other
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Other
 */
export const Other: OwlClass = 'http://www.w3.org/2006/vcard/ns#Other';

/**
 * PCS
 * 
 * This class is deprecated
 *
 * http://www.w3.org/2006/vcard/ns#PCS
 */
export const PCS: OwlClass = 'http://www.w3.org/2006/vcard/ns#PCS';

/**
 * Pager
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Pager
 */
export const Pager: OwlClass = 'http://www.w3.org/2006/vcard/ns#Pager';

/**
 * Parcel
 * 
 * This class is deprecated
 *
 * http://www.w3.org/2006/vcard/ns#Parcel
 */
export const Parcel: OwlClass = 'http://www.w3.org/2006/vcard/ns#Parcel';

/**
 * Parent
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Parent
 */
export const Parent: OwlClass = 'http://www.w3.org/2006/vcard/ns#Parent';

/**
 * Postal
 * 
 * This class is deprecated
 *
 * http://www.w3.org/2006/vcard/ns#Postal
 */
export const Postal: OwlClass = 'http://www.w3.org/2006/vcard/ns#Postal';

/**
 * Pref
 * 
 * This class is deprecated
 *
 * http://www.w3.org/2006/vcard/ns#Pref
 */
export const Pref: OwlClass = 'http://www.w3.org/2006/vcard/ns#Pref';

/**
 * Sibling
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Sibling
 */
export const Sibling: OwlClass = 'http://www.w3.org/2006/vcard/ns#Sibling';

/**
 * Spouse
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Spouse
 */
export const Spouse: OwlClass = 'http://www.w3.org/2006/vcard/ns#Spouse';

/**
 * Sweetheart
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Sweetheart
 */
export const Sweetheart: OwlClass = 'http://www.w3.org/2006/vcard/ns#Sweetheart';

/**
 * Tel
 * 
 * This class is deprecated. Use the hasTelephone object property.
 *
 * http://www.w3.org/2006/vcard/ns#Tel
 */
export const Tel: OwlClass = 'http://www.w3.org/2006/vcard/ns#Tel';

/**
 * Text
 * 
 * Also called sms telephone
 *
 * http://www.w3.org/2006/vcard/ns#Text
 */
export const Text: OwlClass = 'http://www.w3.org/2006/vcard/ns#Text';

/**
 * Text phone
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#TextPhone
 */
export const TextPhone: OwlClass = 'http://www.w3.org/2006/vcard/ns#TextPhone';

/**
 * Unknown
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Unknown
 */
export const Unknown: OwlClass = 'http://www.w3.org/2006/vcard/ns#Unknown';

/**
 * Video
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Video
 */
export const Video: OwlClass = 'http://www.w3.org/2006/vcard/ns#Video';

/**
 * Voice
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Voice
 */
export const Voice: OwlClass = 'http://www.w3.org/2006/vcard/ns#Voice';

/**
 * Work
 * 
 * This implies that the property is related to an individual's work place
 *
 * http://www.w3.org/2006/vcard/ns#Work
 */
export const Work: OwlClass = 'http://www.w3.org/2006/vcard/ns#Work';

/**
 * X400
 * 
 * This class is deprecated
 *
 * http://www.w3.org/2006/vcard/ns#X400
 */
export const X400: OwlClass = 'http://www.w3.org/2006/vcard/ns#X400';

/**
 * Address
 * 
 * To specify the components of the delivery address for the  object
 *
 * http://www.w3.org/2006/vcard/ns#Address
 */
export const Address: OwlClass = 'http://www.w3.org/2006/vcard/ns#Address';

/**
 * Email
 * 
 * To specify the electronic mail address for communication with the object the vCard represents. Use the hasEmail object property.
 *
 * http://www.w3.org/2006/vcard/ns#Email
 */
export const Email: OwlClass = 'http://www.w3.org/2006/vcard/ns#Email';

/**
 * Group
 * 
 * Object representing a group of persons or entities.  A group object will usually contain hasMember properties to specify the members of the group.
 *
 * http://www.w3.org/2006/vcard/ns#Group
 */
export const Group: OwlClass = 'http://www.w3.org/2006/vcard/ns#Group';

/**
 * Individual
 * 
 * An object representing a single person or entity
 *
 * http://www.w3.org/2006/vcard/ns#Individual
 */
export const Individual: OwlClass = 'http://www.w3.org/2006/vcard/ns#Individual';

/**
 * Name
 * 
 * To specify the components of the name of the object
 *
 * http://www.w3.org/2006/vcard/ns#Name
 */
export const Name: OwlClass = 'http://www.w3.org/2006/vcard/ns#Name';

/**
 * VCard
 * 
 * The vCard class is  equivalent to the new Kind class, which is the parent for the four explicit types of vCards (Individual, Organization, Location, Group)
 *
 * http://www.w3.org/2006/vcard/ns#VCard
 */
export const VCard: OwlClass = 'http://www.w3.org/2006/vcard/ns#VCard';

/**
 * Location
 * 
 * An object representing a named geographical place
 *
 * http://www.w3.org/2006/vcard/ns#Location
 */
export const Location: OwlClass = 'http://www.w3.org/2006/vcard/ns#Location';

/**
 * Organization
 * 
 * An object representing an organization.  An organization is a single entity, and might represent a business or government, a department or division within a business or government, a club, an association, or the like.
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Organization
 */
export const Organization: OwlClass = 'http://www.w3.org/2006/vcard/ns#Organization';

/**
 * Gender
 * 
 * Used for gender codes. The URI of the gender code must be used as the value for Gender.
 *
 * http://www.w3.org/2006/vcard/ns#Gender
 */
export const Gender: OwlClass = 'http://www.w3.org/2006/vcard/ns#Gender';

/**
 * Kind
 * 
 * The parent class for all objects
 *
 * http://www.w3.org/2006/vcard/ns#Kind
 */
export const Kind: OwlClass = 'http://www.w3.org/2006/vcard/ns#Kind';

/**
 * Type
 * 
 * Used for type codes. The URI of the type code must be used as the value for Type.
 *
 * http://www.w3.org/2006/vcard/ns#Type
 */
export const Type: OwlClass = 'http://www.w3.org/2006/vcard/ns#Type';

/**
 * Phone
 * 
 * Used for telephone type codes. The URI of the telephone type code must be used as the value for the Telephone Type.
 *
 * http://www.w3.org/2006/vcard/ns#TelephoneType
 */
export const TelephoneType: OwlClass = 'http://www.w3.org/2006/vcard/ns#TelephoneType';

/**
 * Relation Type
 * 
 * Used for relation type codes. The URI of the relation type code must be used as the value for the Relation Type.
 *
 * http://www.w3.org/2006/vcard/ns#RelatedType
 */
export const RelatedType: OwlClass = 'http://www.w3.org/2006/vcard/ns#RelatedType';

/**
 * address
 * 
 * This object property has been mapped
 *
 * http://www.w3.org/2006/vcard/ns#adr
 */
export const adr: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#adr';

/**
 * agent
 * 
 * This object property has been deprecated
 *
 * http://www.w3.org/2006/vcard/ns#agent
 */
export const agent: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#agent';

/**
 * email
 * 
 * This object property has been mapped
 *
 * http://www.w3.org/2006/vcard/ns#email
 */
export const email: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#email';

/**
 * geo
 * 
 * This object property has been mapped
 *
 * http://www.w3.org/2006/vcard/ns#geo
 */
export const geo: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#geo';

/**
 * has additional name
 * 
 * Used to support property parameters for the additional name data property
 *
 * http://www.w3.org/2006/vcard/ns#hasAdditionalName
 */
export const hasAdditionalName: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasAdditionalName';

/**
 * has calendar busy
 * 
 * To specify the busy time associated with the object. (Was called FBURL in RFC6350)
 *
 * http://www.w3.org/2006/vcard/ns#hasCalendarBusy
 */
export const hasCalendarBusy: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasCalendarBusy';

/**
 * has calendar link
 * 
 * To specify the calendar associated with the object. (Was called CALURI in RFC6350)
 *
 * http://www.w3.org/2006/vcard/ns#hasCalendarLink
 */
export const hasCalendarLink: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasCalendarLink';

/**
 * has calendar request
 * 
 * To specify the calendar user address to which a scheduling request be sent for the object. (Was called CALADRURI in RFC6350)
 *
 * http://www.w3.org/2006/vcard/ns#hasCalendarRequest
 */
export const hasCalendarRequest: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasCalendarRequest';

/**
 * has category
 * 
 * Used to support property parameters for the category data property
 *
 * http://www.w3.org/2006/vcard/ns#hasCategory
 */
export const hasCategory: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasCategory';

/**
 * has country name
 * 
 * Used to support property parameters for the country name data property
 *
 * http://www.w3.org/2006/vcard/ns#hasCountryName
 */
export const hasCountryName: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasCountryName';

/**
 * has formatted name
 * 
 * Used to support property parameters for the formatted name data property
 *
 * http://www.w3.org/2006/vcard/ns#hasFN
 */
export const hasFN: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasFN';

/**
 * has family name
 * 
 * Used to support property parameters for the family name data property
 *
 * http://www.w3.org/2006/vcard/ns#hasFamilyName
 */
export const hasFamilyName: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasFamilyName';

/**
 * has gender
 * 
 * To specify  the sex or gender identity of the object. URIs are recommended to enable interoperable sex and gender codes to be used.
 *
 * http://www.w3.org/2006/vcard/ns#hasGender
 */
export const hasGender: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasGender';

/**
 * has given name
 * 
 * Used to support property parameters for the given name data property
 *
 * http://www.w3.org/2006/vcard/ns#hasGivenName
 */
export const hasGivenName: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasGivenName';

/**
 * has honorific prefix
 * 
 * Used to support property parameters for the honorific prefix data property
 *
 * http://www.w3.org/2006/vcard/ns#hasHonorificPrefix
 */
export const hasHonorificPrefix: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasHonorificPrefix';

/**
 * has honorific suffix
 * 
 * Used to support property parameters for the honorific suffix data property
 *
 * http://www.w3.org/2006/vcard/ns#hasHonorificSuffix
 */
export const hasHonorificSuffix: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasHonorificSuffix';

/**
 * has messaging
 * 
 * To specify the instant messaging and presence protocol communications with the object. (Was called IMPP in RFC6350)
 *
 * http://www.w3.org/2006/vcard/ns#hasInstantMessage
 */
export const hasInstantMessage: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasInstantMessage';

/**
 * has language
 * 
 * Used to support property parameters for the language data property
 *
 * http://www.w3.org/2006/vcard/ns#hasLanguage
 */
export const hasLanguage: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasLanguage';

/**
 * has locality
 * 
 * Used to support property parameters for the locality data property
 *
 * http://www.w3.org/2006/vcard/ns#hasLocality
 */
export const hasLocality: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasLocality';

/**
 * has nickname
 * 
 * Used to support property parameters for the nickname data property
 *
 * http://www.w3.org/2006/vcard/ns#hasNickname
 */
export const hasNickname: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasNickname';

/**
 * has note
 * 
 * Used to support property parameters for the note data property
 *
 * http://www.w3.org/2006/vcard/ns#hasNote
 */
export const hasNote: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasNote';

/**
 * has organization name
 * 
 * Used to support property parameters for the organization name data property
 *
 * http://www.w3.org/2006/vcard/ns#hasOrganizationName
 */
export const hasOrganizationName: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasOrganizationName';

/**
 * has organization unit name
 * 
 * Used to support property parameters for the organization unit name data property
 *
 * http://www.w3.org/2006/vcard/ns#hasOrganizationUnit
 */
export const hasOrganizationUnit: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasOrganizationUnit';

/**
 * has postal code
 * 
 * Used to support property parameters for the postal code data property
 *
 * http://www.w3.org/2006/vcard/ns#hasPostalCode
 */
export const hasPostalCode: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasPostalCode';

/**
 * has region
 * 
 * Used to support property parameters for the region data property
 *
 * http://www.w3.org/2006/vcard/ns#hasRegion
 */
export const hasRegion: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasRegion';

/**
 * has related
 * 
 * To specify a relationship between another entity and the entity represented by this object
 *
 * http://www.w3.org/2006/vcard/ns#hasRelated
 */
export const hasRelated: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasRelated';

/**
 * has role
 * 
 * Used to support property parameters for the role data property
 *
 * http://www.w3.org/2006/vcard/ns#hasRole
 */
export const hasRole: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasRole';

/**
 * has source
 * 
 * To identify the source of directory information of the object
 *
 * http://www.w3.org/2006/vcard/ns#hasSource
 */
export const hasSource: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasSource';

/**
 * has street address
 * 
 * Used to support property parameters for the street address data property
 *
 * http://www.w3.org/2006/vcard/ns#hasStreetAddress
 */
export const hasStreetAddress: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasStreetAddress';

/**
 * has title
 * 
 * Used to support property parameters for the title data property
 *
 * http://www.w3.org/2006/vcard/ns#hasTitle
 */
export const hasTitle: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasTitle';

/**
 * has uid
 * 
 * To specify a value that represents a globally unique identifier corresponding to the object
 *
 * http://www.w3.org/2006/vcard/ns#hasUID
 */
export const hasUID: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasUID';

/**
 * has value
 * 
 * Used to indicate the resource value of an object property that requires property parameters
 *
 * http://www.w3.org/2006/vcard/ns#hasValue
 */
export const hasValue: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasValue';

/**
 * organization
 * 
 * This object property has been mapped. Use the organization-name data property.
 *
 * http://www.w3.org/2006/vcard/ns#org
 */
export const org: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#org';

/**
 * has address
 * 
 * To specify the components of the delivery address for the object
 *
 * http://www.w3.org/2006/vcard/ns#hasAddress
 */
export const hasAddress: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasAddress';

/**
 * has email
 * 
 * To specify the electronic mail address for communication with the object
 *
 * http://www.w3.org/2006/vcard/ns#hasEmail
 */
export const hasEmail: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasEmail';

/**
 * has geo
 * 
 * To specify information related to the global positioning of the object. May also be used as a property parameter.
 *
 * http://www.w3.org/2006/vcard/ns#hasGeo
 */
export const hasGeo: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasGeo';

/**
 * has key
 * 
 * To specify a public key or authentication certificate associated with the object
 *
 * http://www.w3.org/2006/vcard/ns#hasKey
 */
export const hasKey: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasKey';

/**
 * has logo
 * 
 * To specify a graphic image of a logo associated with the object 
 *
 * http://www.w3.org/2006/vcard/ns#hasLogo
 */
export const hasLogo: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasLogo';

/**
 * has name
 * 
 * To specify the components of the name of the object
 *
 * http://www.w3.org/2006/vcard/ns#hasName
 */
export const hasName: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasName';

/**
 * has photo
 * 
 * To specify an image or photograph information that annotates some aspect of the object
 *
 * http://www.w3.org/2006/vcard/ns#hasPhoto
 */
export const hasPhoto: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasPhoto';

/**
 * has sound
 * 
 * To specify a digital sound content information that annotates some aspect of the object
 *
 * http://www.w3.org/2006/vcard/ns#hasSound
 */
export const hasSound: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasSound';

/**
 * has telephone
 * 
 * To specify the telephone number for telephony communication with the object
 *
 * http://www.w3.org/2006/vcard/ns#hasTelephone
 */
export const hasTelephone: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasTelephone';

/**
 * has url
 * 
 * To specify a uniform resource locator associated with the object
 *
 * http://www.w3.org/2006/vcard/ns#hasURL
 */
export const hasURL: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasURL';

/**
 * key
 * 
 * This object property has been mapped
 *
 * http://www.w3.org/2006/vcard/ns#key
 */
export const key: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#key';

/**
 * logo
 * 
 * This object property has been mapped
 *
 * http://www.w3.org/2006/vcard/ns#logo
 */
export const logo: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#logo';

/**
 * name
 * 
 * This object property has been mapped
 *
 * http://www.w3.org/2006/vcard/ns#n
 */
export const n: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#n';

/**
 * photo
 * 
 * This object property has been mapped
 *
 * http://www.w3.org/2006/vcard/ns#photo
 */
export const photo: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#photo';

/**
 * sound
 * 
 * This object property has been mapped
 *
 * http://www.w3.org/2006/vcard/ns#sound
 */
export const sound: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#sound';

/**
 * telephone
 * 
 * This object property has been mapped
 *
 * http://www.w3.org/2006/vcard/ns#tel
 */
export const tel: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#tel';

/**
 * url
 * 
 * This object property has been mapped
 *
 * http://www.w3.org/2006/vcard/ns#url
 */
export const url: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#url';

/**
 * has member
 * 
 * To include a member in the group this object represents. (This property can only be used by Group individuals)
 *
 * http://www.w3.org/2006/vcard/ns#hasMember
 */
export const hasMember: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasMember';

/**
 * anniversary
 * 
 * The date of marriage, or equivalent, of the object
 *
 * http://www.w3.org/2006/vcard/ns#anniversary
 */
export const anniversary: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#anniversary';

/**
 * birth date
 * 
 * To specify the birth date of the object
 *
 * http://www.w3.org/2006/vcard/ns#bday
 */
export const bday: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#bday';

/**
 * category
 * 
 * The category information about the object, also known as tags
 *
 * http://www.w3.org/2006/vcard/ns#category
 */
export const category: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#category';

/**
 * class
 * 
 * This data property has been deprecated
 * Note that `class` is a reserved Javascript keyword, and is therefore suffixed by `__workaround`.
 * For a list of reserved keywords, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords.
 *
 * http://www.w3.org/2006/vcard/ns#class
 */
export const class__workaround: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#class';

/**
 * label
 * 
 * This data property has been deprecated
 *
 * http://www.w3.org/2006/vcard/ns#label
 */
export const label: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#label';

/**
 * language
 * 
 * To specify the language that may be used for contacting the object. May also be used as a property parameter.
 *
 * http://www.w3.org/2006/vcard/ns#language
 */
export const language: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#language';

/**
 * latitude
 * 
 * This data property has been deprecated. See hasGeo
 *
 * http://www.w3.org/2006/vcard/ns#latitude
 */
export const latitude: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#latitude';

/**
 * longitude
 * 
 * This data property has been deprecated. See hasGeo
 *
 * http://www.w3.org/2006/vcard/ns#longitude
 */
export const longitude: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#longitude';

/**
 * mailer
 * 
 * This data property has been deprecated
 *
 * http://www.w3.org/2006/vcard/ns#mailer
 */
export const mailer: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#mailer';

/**
 * note
 * 
 * A note associated with the object
 *
 * http://www.w3.org/2006/vcard/ns#note
 */
export const note: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#note';

/**
 * product id
 * 
 * To specify the identifier for the product that created the object
 *
 * http://www.w3.org/2006/vcard/ns#prodid
 */
export const prodid: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#prodid';

/**
 * revision
 * 
 * To specify revision information about the object
 *
 * http://www.w3.org/2006/vcard/ns#rev
 */
export const rev: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#rev';

/**
 * role
 * 
 * To specify the function or part played in a particular situation by the object
 *
 * http://www.w3.org/2006/vcard/ns#role
 */
export const role: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#role';

/**
 * title
 * 
 * To specify the position or job of the object
 *
 * http://www.w3.org/2006/vcard/ns#title
 */
export const title: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#title';

/**
 * time zone
 * 
 * To indicate time zone information that is specific to the object. May also be used as a property parameter.
 *
 * http://www.w3.org/2006/vcard/ns#tz
 */
export const tz: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#tz';

/**
 * value
 * 
 * Used to indicate the literal value of a data property that requires property parameters
 *
 * http://www.w3.org/2006/vcard/ns#value
 */
export const value: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#value';

/**
 * formatted name
 * 
 * The formatted text corresponding to the name of the object
 *
 * http://www.w3.org/2006/vcard/ns#fn
 */
export const fn: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#fn';

/**
 * nickname
 * 
 * The nick name associated with the object
 *
 * http://www.w3.org/2006/vcard/ns#nickname
 */
export const nickname: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#nickname';

/**
 * locality
 * 
 * The locality (e.g. city or town) associated with the address of the object
 *
 * http://www.w3.org/2006/vcard/ns#locality
 */
export const locality: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#locality';

/**
 * region
 * 
 * The region (e.g. state or province) associated with the address of the object
 *
 * http://www.w3.org/2006/vcard/ns#region
 */
export const region: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#region';
