// rdf-namespaces/dist/vcard.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var vcard$1 = require('./vcard-8871dca5.js');



exports.Acquaintance = vcard$1.Acquaintance;
exports.Address = vcard$1.Address;
exports.Agent = vcard$1.Agent;
exports.BBS = vcard$1.BBS;
exports.Car = vcard$1.Car;
exports.Cell = vcard$1.Cell;
exports.Child = vcard$1.Child;
exports.Colleague = vcard$1.Colleague;
exports.Contact = vcard$1.Contact;
exports.Coresident = vcard$1.Coresident;
exports.Coworker = vcard$1.Coworker;
exports.Crush = vcard$1.Crush;
exports.Date = vcard$1.Date;
exports.Dom = vcard$1.Dom;
exports.Email = vcard$1.Email;
exports.Emergency = vcard$1.Emergency;
exports.Fax = vcard$1.Fax;
exports.Female = vcard$1.Female;
exports.Friend = vcard$1.Friend;
exports.Gender = vcard$1.Gender;
exports.Group = vcard$1.Group;
exports.Home = vcard$1.Home;
exports.ISDN = vcard$1.ISDN;
exports.Individual = vcard$1.Individual;
exports.Internet = vcard$1.Internet;
exports.Intl = vcard$1.Intl;
exports.Kin = vcard$1.Kin;
exports.Kind = vcard$1.Kind;
exports.Label = vcard$1.Label;
exports.Location = vcard$1.Location;
exports.Male = vcard$1.Male;
exports.Me = vcard$1.Me;
exports.Met = vcard$1.Met;
exports.Modem = vcard$1.Modem;
exports.Msg = vcard$1.Msg;
exports.Muse = vcard$1.Muse;
exports.Name = vcard$1.Name;
exports.Neighbor = vcard$1.Neighbor;
exports.None = vcard$1.None;
exports.Organization = vcard$1.Organization;
exports.Other = vcard$1.Other;
exports.PCS = vcard$1.PCS;
exports.Pager = vcard$1.Pager;
exports.Parcel = vcard$1.Parcel;
exports.Parent = vcard$1.Parent;
exports.Postal = vcard$1.Postal;
exports.Pref = vcard$1.Pref;
exports.RelatedType = vcard$1.RelatedType;
exports.Sibling = vcard$1.Sibling;
exports.Spouse = vcard$1.Spouse;
exports.Sweetheart = vcard$1.Sweetheart;
exports.Tel = vcard$1.Tel;
exports.TelephoneType = vcard$1.TelephoneType;
exports.Text = vcard$1.Text;
exports.TextPhone = vcard$1.TextPhone;
exports.Type = vcard$1.Type;
exports.Unknown = vcard$1.Unknown;
exports.VCard = vcard$1.VCard;
exports.Video = vcard$1.Video;
exports.Voice = vcard$1.Voice;
exports.Work = vcard$1.Work;
exports.X400 = vcard$1.X400;
exports.adr = vcard$1.adr;
exports.agent = vcard$1.agent;
exports.anniversary = vcard$1.anniversary;
exports.bday = vcard$1.bday;
exports.category = vcard$1.category;
exports.class__workaround = vcard$1.class__workaround;
exports.email = vcard$1.email;
exports.fn = vcard$1.fn;
exports.geo = vcard$1.geo;
exports.hasAdditionalName = vcard$1.hasAdditionalName;
exports.hasAddress = vcard$1.hasAddress;
exports.hasCalendarBusy = vcard$1.hasCalendarBusy;
exports.hasCalendarLink = vcard$1.hasCalendarLink;
exports.hasCalendarRequest = vcard$1.hasCalendarRequest;
exports.hasCategory = vcard$1.hasCategory;
exports.hasCountryName = vcard$1.hasCountryName;
exports.hasEmail = vcard$1.hasEmail;
exports.hasFN = vcard$1.hasFN;
exports.hasFamilyName = vcard$1.hasFamilyName;
exports.hasGender = vcard$1.hasGender;
exports.hasGeo = vcard$1.hasGeo;
exports.hasGivenName = vcard$1.hasGivenName;
exports.hasHonorificPrefix = vcard$1.hasHonorificPrefix;
exports.hasHonorificSuffix = vcard$1.hasHonorificSuffix;
exports.hasInstantMessage = vcard$1.hasInstantMessage;
exports.hasKey = vcard$1.hasKey;
exports.hasLanguage = vcard$1.hasLanguage;
exports.hasLocality = vcard$1.hasLocality;
exports.hasLogo = vcard$1.hasLogo;
exports.hasMember = vcard$1.hasMember;
exports.hasName = vcard$1.hasName;
exports.hasNickname = vcard$1.hasNickname;
exports.hasNote = vcard$1.hasNote;
exports.hasOrganizationName = vcard$1.hasOrganizationName;
exports.hasOrganizationUnit = vcard$1.hasOrganizationUnit;
exports.hasPhoto = vcard$1.hasPhoto;
exports.hasPostalCode = vcard$1.hasPostalCode;
exports.hasRegion = vcard$1.hasRegion;
exports.hasRelated = vcard$1.hasRelated;
exports.hasRole = vcard$1.hasRole;
exports.hasSound = vcard$1.hasSound;
exports.hasSource = vcard$1.hasSource;
exports.hasStreetAddress = vcard$1.hasStreetAddress;
exports.hasTelephone = vcard$1.hasTelephone;
exports.hasTitle = vcard$1.hasTitle;
exports.hasUID = vcard$1.hasUID;
exports.hasURL = vcard$1.hasURL;
exports.hasValue = vcard$1.hasValue;
exports.key = vcard$1.key;
exports.label = vcard$1.label;
exports.language = vcard$1.language;
exports.latitude = vcard$1.latitude;
exports.locality = vcard$1.locality;
exports.logo = vcard$1.logo;
exports.longitude = vcard$1.longitude;
exports.mailer = vcard$1.mailer;
exports.n = vcard$1.n;
exports.nickname = vcard$1.nickname;
exports.note = vcard$1.note;
exports.org = vcard$1.org;
exports.photo = vcard$1.photo;
exports.prodid = vcard$1.prodid;
exports.region = vcard$1.region;
exports.rev = vcard$1.rev;
exports.role = vcard$1.role;
exports.sound = vcard$1.sound;
exports.tel = vcard$1.tel;
exports.title = vcard$1.title;
exports.tz = vcard$1.tz;
exports.url = vcard$1.url;
exports.value = vcard$1.value;
