// rdf-namespaces/dist/wf.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var wf$1 = require('./wf-e7df4562.js');



exports.Change = wf$1.Change;
exports.Closed = wf$1.Closed;
exports.Creation = wf$1.Creation;
exports.Message = wf$1.Message;
exports.NonTerminalState = wf$1.NonTerminalState;
exports.Open = wf$1.Open;
exports.Product = wf$1.Product;
exports.Task = wf$1.Task;
exports.TerminalState = wf$1.TerminalState;
exports.Tracker = wf$1.Tracker;
exports.Transition = wf$1.Transition;
exports.affects = wf$1.affects;
exports.allowedTransitions = wf$1.allowedTransitions;
exports.asigneeClass = wf$1.asigneeClass;
exports.assignee = wf$1.assignee;
exports.attachment = wf$1.attachment;
exports.creates = wf$1.creates;
exports.dateDue = wf$1.dateDue;
exports.deliverable = wf$1.deliverable;
exports.dependent = wf$1.dependent;
exports.description = wf$1.description;
exports.final = wf$1.final;
exports.goalDescription = wf$1.goalDescription;
exports.initialState = wf$1.initialState;
exports.issue = wf$1.issue;
exports.issueCategory = wf$1.issueCategory;
exports.issueClass = wf$1.issueClass;
exports.message = wf$1.message;
exports.modifiedBy = wf$1.modifiedBy;
exports.recipent = wf$1.recipent;
exports.requires = wf$1.requires;
exports.screenShot = wf$1.screenShot;
exports.sender = wf$1.sender;
exports.source = wf$1.source;
exports.stateStore = wf$1.stateStore;
exports.subscriber = wf$1.subscriber;
exports.successCriteria = wf$1.successCriteria;
exports.task = wf$1.task;
exports.terminalOutput = wf$1.terminalOutput;
exports.testData = wf$1.testData;
exports.tracker = wf$1.tracker;
exports.transactionStore = wf$1.transactionStore;
