// ldflex/lib/defaultHandlers.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _AsyncIteratorHandler = _interopRequireDefault(require("./AsyncIteratorHandler"));

var _DataHandler = _interopRequireDefault(require("./DataHandler"));

var _DeleteFunctionHandler = _interopRequireDefault(require("./DeleteFunctionHandler"));

var _ExecuteQueryHandler = _interopRequireDefault(require("./ExecuteQueryHandler"));

var _GetFunctionHandler = _interopRequireDefault(require("./GetFunctionHandler"));

var _InsertFunctionHandler = _interopRequireDefault(require("./InsertFunctionHandler"));

var _MutationExpressionsHandler = _interopRequireDefault(require("./MutationExpressionsHandler"));

var _PathExpressionHandler = _interopRequireDefault(require("./PathExpressionHandler"));

var _PredicateHandler = _interopRequireDefault(require("./PredicateHandler"));

var _PredicatesHandler = _interopRequireDefault(require("./PredicatesHandler"));

var _PreloadHandler = _interopRequireDefault(require("./PreloadHandler"));

var _PropertiesHandler = _interopRequireDefault(require("./PropertiesHandler"));

var _ReplaceFunctionHandler = _interopRequireDefault(require("./ReplaceFunctionHandler"));

var _SetFunctionHandler = _interopRequireDefault(require("./SetFunctionHandler"));

var _SortHandler = _interopRequireDefault(require("./SortHandler"));

var _SparqlHandler = _interopRequireDefault(require("./SparqlHandler"));

var _StringToLDflexHandler = _interopRequireDefault(require("./StringToLDflexHandler"));

var _SubjectHandler = _interopRequireDefault(require("./SubjectHandler"));

var _SubjectsHandler = _interopRequireDefault(require("./SubjectsHandler"));

var _ThenHandler = _interopRequireDefault(require("./ThenHandler"));

var _ToArrayHandler = _interopRequireDefault(require("./ToArrayHandler"));

var _valueUtils = require("./valueUtils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * A map with default property handlers.
 */
var _default = {
  // Flag to loaders that exported paths are not ES6 modules
  __esModule: () => undefined,
  // Add thenable and async iterable behavior
  then: new _ThenHandler.default(),
  [Symbol.asyncIterator]: new _AsyncIteratorHandler.default(),
  // Add read and query functionality
  get: new _GetFunctionHandler.default(),
  subject: new _SubjectHandler.default(),
  predicate: new _PredicateHandler.default(),
  properties: new _PropertiesHandler.default(),
  predicates: new _PredicatesHandler.default(),
  pathExpression: new _PathExpressionHandler.default(),
  sparql: new _SparqlHandler.default(),
  subjects: new _SubjectsHandler.default(),
  results: new _ExecuteQueryHandler.default(),
  sort: new _SortHandler.default('ASC'),
  sortDesc: new _SortHandler.default('DESC'),
  preload: new _PreloadHandler.default(),
  // Add write functionality
  mutationExpressions: new _MutationExpressionsHandler.default(),
  add: new _InsertFunctionHandler.default(),
  set: new _SetFunctionHandler.default(),
  replace: new _ReplaceFunctionHandler.default(),
  delete: new _DeleteFunctionHandler.default(),
  // Add RDFJS term handling
  termType: termPropertyHandler('termType'),
  value: termPropertyHandler('value'),
  datatype: termPropertyHandler('datatype'),
  language: termPropertyHandler('language'),
  canonical: termPropertyHandler('canonical'),
  equals: _DataHandler.default.sync('subject', 'equals'),
  toString: _DataHandler.default.syncFunction('subject', 'value'),
  valueOf: subjectToPrimitiveHandler(),
  toPrimitive: subjectToPrimitiveHandler(),
  // Add iteration helpers
  toArray: new _ToArrayHandler.default(),
  termTypes: handler((_, path) => path.toArray(term => term.termType)),
  values: handler((_, path) => path.toArray(term => term.value)),
  datatypes: handler((_, path) => path.toArray(term => term.datatype)),
  languages: handler((_, path) => path.toArray(term => term.language)),
  // Parse a string into an LDflex object
  resolve: new _StringToLDflexHandler.default()
}; // Creates a handler from the given function

exports.default = _default;

function handler(handle) {
  return {
    handle
  };
} // Creates a handler for the given RDF/JS Term property


function termPropertyHandler(property) {
  // If a resolved subject is present,
  // behave as an RDF/JS term and synchronously expose the property;
  // otherwise, return a promise to the property value
  return handler(({
    subject
  }, path) => subject && property in subject ? subject[property] : path.then && path.then(term => term === null || term === void 0 ? void 0 : term[property]));
} // Creates a handler that converts the subject into a primitive


function subjectToPrimitiveHandler() {
  return handler(({
    subject
  }) => () => typeof (subject === null || subject === void 0 ? void 0 : subject.termType) !== 'string' ? undefined : (0, _valueUtils.termToPrimitive)(subject));
}
